---
title: "circos"
author: "Robin Droit"
date: "22/04/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r data_prep}
set.seed(seed = 42006900)

rna_table <- read.table(file = "../expression/circos_exp.tsv", header = TRUE)
exp_table <- rna_table
rna_table[,5:37] <- log10(rna_table[,5:37] + 1)

table_heatmap_expression <- rna_table[rna_table$GeneName %in% c("CD47", "B4GALNT1"), ]
rownames(table_heatmap_expression) <- table_heatmap_expression$GeneName
table_heatmap_expression <- table_heatmap_expression[,5:37]
table_heatmap_expression <- t(table_heatmap_expression)

pheatmap::pheatmap(table_heatmap_expression, filename = "heatmap_CD47_B4GALNT1_human_part_log.pdf", cluster_rows = F)

table_heatmap_expression <- exp_table[exp_table$GeneName %in% c("CD47", "B4GALNT1"), ]
rownames(table_heatmap_expression) <- table_heatmap_expression$GeneName
table_heatmap_expression <- table_heatmap_expression[,5:37]
table_heatmap_expression <- t(table_heatmap_expression)

pheatmap::pheatmap(table_heatmap_expression, filename = "heatmap_CD47_B4GALNT1_human_part.pdf", cluster_rows = F)

```

```{r circos}
library(RCircos)
library("dplyr")
library("plyr")
library("tidyr")
library("readr")

library(GenomicFeatures)
foo = makeTxDbFromUCSC(genome="hg19", tablename="knownGene")
bar = transcriptsBy(foo, by="gene")

frm = as.data.frame(bar)

write.csv(frm2, file = "all_genes_id.csv")

library(TxDb.Hsapiens.UCSC.hg19.knownGene)
g = genes(TxDb.Hsapiens.UCSC.hg19.knownGene)
head(g)

frm2 = as.data.frame(g)

```

```{r CNV_function}
# function to create the apropriate table for Circos starting from the path of a cncf file
CNV_prep <- function(table_path) {
  
  cnv_table <- read.table(file = table_path, sep = "\t", header = TRUE)
  cnv_table <- cnv_table[, c("chrom", "start", "end", "seg", "tcn.em")]
  colnames(cnv_table) <- c("Chromosome", "chromStart", "chromEnd", "Name", "CNvalue")
  cnv_table$Chromosome <- sub("^", "chr", cnv_table$Chromosome)
  cnv_table[cnv_table$Chromosome == "chr23","Chromosome"] <- "chrX"
  cnv_table[cnv_table$Chromosome == "chr24","Chromosome"] <- "chrY"
  cnv_table[cnv_table$CNvalue == 0, "CNvalue"] <- -2
  cnv_table[cnv_table$CNvalue == 1, "CNvalue"] <- -1
  cnv_table[cnv_table$CNvalue == 2, "CNvalue"] <- 0
  cnv_table[cnv_table$CNvalue >= 7 , "CNvalue"] <- 2
  cnv_table[cnv_table$CNvalue >= 3, "CNvalue"] <- 1
  cnv_table$length <- cnv_table$chromEnd - cnv_table$chromStart
  
  return(cnv_table)
}

```

```{r Fusions_function}
# starting from the paths of all the files we whant to add, creating a fusion file for the circos
fusionsPrep <- function(folder, pattern) { 
  
  data_all <- list.files(path = folder, pattern = pattern, full.names = TRUE) 
  data_all <- lapply(data_all, function(i){read.table(i, header = FALSE)}) %>% bind_rows
  colnames(data_all) <- "full_name"
  data_all <- ddply(data_all,.(full_name),nrow)
  data_all <- separate(data = data_all, col = "full_name", into = c("Genes", "positions"), sep = ":")
  data_all <- separate(data = data_all, col = "positions", into = c("Position1", "Position2"), sep = "\\|")
  data_all <- separate(data = data_all, col = "Position1", into = c("Chromosome", "chromStart"), sep = "-")
  data_all$chromStart <- as.numeric(data_all$chromStart)
  data_all$chromEnd <- data_all$chromStart
  data_all <- separate(data = data_all, col = "Position2", into = c("Chromosome.1", "chromStart.1"), sep = "-")
  data_all$chromStart.1 <- as.numeric(data_all$chromStart.1)
  data_all$chromEnd.1 <- data_all$chromStart.1
  data_all$PlotColor <- "blue"
  data_all[which(data_all$V1 == 3), "PlotColor"] <- "green"
  data_all[which(data_all$V1 == 4), "PlotColor"] <- "red"
  
  return(data_all)
}

```

```{r Mutations_function}
mutPrep <- function(folder, pattern) {
  
  mut_table <- list.files(path = folder, pattern = pattern, full.names = TRUE) 
  mut_table <- lapply(mut_table, function(i){read.table(i, sep = ",", header = TRUE)}) %>% bind_rows
  #mut_table <- separate(data = mut_table, col = 1, into = c("Symbol", "Chromosome", "chromStart", "chromEnd"), sep = " ")
  mut_table <- mut_table[which(mut_table$Symbol != "Intergenic"),]
  mut_table[mut_table$Chromosome == "chr23","Chromosome"] <- "chrX"
  mut_table[mut_table$Chromosome == "chr24","Chromosome"] <- "chrY"
  mut_table$chromStart <- as.numeric(mut_table$chromStart)
  mut_table$chromEnd <- as.numeric(mut_table$chromEnd)
  
  return(mut_table)
}

```

```{r circos_177}
chr.exclude <- NULL
data(UCSC.HG19.Human.CytoBandIdeogram)
cyto.info <- UCSC.HG19.Human.CytoBandIdeogram
tracks.inside <- 13
tracks.outside <- 0
RCircos.Set.Core.Components(cyto.info, chr.exclude ,tracks.inside, tracks.outside)

rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.position <- RCircos.Get.Plot.Positions()

rcircos.params$point.size <- 3
RCircos.Reset.Plot.Parameters(rcircos.params)

RCircos.List.Plot.Parameters()

# Actually making the plot

# pdf(file="RCircos_PDX177.pdf", height=8, width=8, compress=TRUE)
#pdf(file="Rcircos_PDX177.pdf")
png(file="RCircos_PDX177.png", res = 600, width = 4800, height = 4800)
RCircos.Set.Plot.Area()

par(mai=c(0.25, 0.25, 0.25, 0.25))
plot.new()
plot.window(c(-2.5, 2.5), c(-2.5, 2.5))

# scatter
mut_177 <- mutPrep("../CCF", "MAP177*")
mut_177_Diag <- mut_177[which(mut_177$PlotColor == "black"), ]
colnames(mut_177_Diag) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
# rownames(mut_177_Diag) <- paste(mut_177_Diag$Symbol, mut_177_Diag$chromStart)
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_177_Diag, data.col, side, inside.pos = 1.6, outside.pos = 1.65)

mut_177_Relapse <- mut_177[which(mut_177$PlotColor == "red"), ]
colnames(mut_177_Relapse) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
# rownames(mut_177_Relapse) <- paste(mut_177_Relapse$Symbol, mut_177_Relapse$chromStart)
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_177_Relapse, data.col, side, inside.pos = 1.65, outside.pos = 1.7)

# keep_mut <- c(rownames(mut_177_Diag),rownames(mut_177_Relapse))

mut_177_Ortho <- mut_177[which(mut_177$PlotColor == "blue"), ]
colnames(mut_177_Ortho) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_177_Ortho, data.col, side, inside.pos = 1.7, outside.pos = 1.75)

mut_177_Sc <- mut_177[which(mut_177$PlotColor == "green"), ]
colnames(mut_177_Sc) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_177_Sc, data.col, side, inside.pos = 1.75, outside.pos = 1.8)


# DIAG    
# Heatmap
cnv_fc_177 <- CNV_prep("../../facets/MAP177_Diag_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_177, data.col, side, inside.pos = 1.15, outside.pos = 1.25, max.value = 2, min.value = -2)

# META
# Heatmap
cnv_fc_177 <- CNV_prep("../../facets/MAP177_Relapse_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_177, data.col, side, inside.pos = 1.25, outside.pos = 1.35, max.value = 2, min.value = -2)

# ORTHO
# Heatmap
cnv_fc_177 <- CNV_prep("../../facets/MAP177_Ortho_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_177, data.col, side, inside.pos = 1.35, outside.pos = 1.45, max.value = 2, min.value = -2)

# SC
# Heatmap
cnv_fc_177 <- CNV_prep("../../facets/MAP177_Sc_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_177, data.col, side, inside.pos = 1.45, outside.pos = 1.55, max.value = 2, min.value = -2)

# DIAG
# Histogram
rna_diag <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP177_Diag")]
rna_diag$PlotColor <- "black"
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_diag, data.col, side, inside.pos = 0.9, outside.pos = 0.95, min.value=min(rna_diag$MAP177_Diag), max.value=max(rna_diag$MAP177_Diag))

# META
# Histogram
rna_meta <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP177_Relapse")]
rna_meta$PlotColor <- "red"
rna_meta$MAP177_Relapse <- ((rna_meta$MAP177_Relapse - rna_diag$MAP177_Diag) / rna_diag$MAP177_Diag) ^ 1/3
rna_meta <- rna_meta[complete.cases(rna_meta),]
rna_meta <- rna_meta[is.finite(rna_meta$MAP177_Relapse),]
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_meta, data.col, side, inside.pos = 0.95, outside.pos = 1, min.value=min(rna_meta$MAP177_Relapse), max.value=max(rna_meta$MAP177_Relapse))

# ORTHO
# Histogram
rna_ortho <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP177_Ortho")]
rna_ortho$PlotColor <- "blue"
rna_ortho$MAP177_Ortho <- ((rna_ortho$MAP177_Ortho - rna_diag$MAP177_Diag) / rna_diag$MAP177_Diag) ^ 1/3
rna_ortho <- rna_ortho[complete.cases(rna_ortho),]
rna_ortho <- rna_ortho[is.finite(rna_ortho$MAP177_Ortho),]
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_ortho, data.col, side, inside.pos = 1, outside.pos = 1.05, min.value=min(rna_ortho$MAP177_Ortho), max.value=max(rna_ortho$MAP177_Ortho))

# SC
# Histogram
rna_sc <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP177_Sc")]
rna_sc$PlotColor <- "green"
rna_sc$MAP177_Sc <- ((rna_sc$MAP177_Sc - rna_diag$MAP177_Diag) / rna_diag$MAP177_Diag) ^ 1/3
rna_sc <- rna_sc[complete.cases(rna_sc),]
rna_sc <- rna_sc[is.finite(rna_sc$MAP177_Sc),]
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_sc, data.col, side, inside.pos = 1.05, outside.pos = 1.1, min.value=min(rna_sc$MAP177_Sc), max.value=max(rna_sc$MAP177_Sc))


# Fusions data 177
data(RCircos.Link.Data)
fusions_177 <- fusionsPrep("../Fusions/data_Fusion_Inspector", "MAP177*")
fusions_177 <- fusions_177[which(fusions_177$PlotColor != "blue"),]
fusions_177 <- fusions_177[!(fusions_177$Genes %in% c("SDHAF2--PPP1R32", "COL1A1--COL3A1", "COL1A1--COL5A1")),]
fusions_177_plot <- fusions_177[c("Chromosome", "chromStart","chromEnd", "Chromosome.1", "chromStart.1", "chromEnd.1", "PlotColor")]
track.num <- 13
RCircos.Link.Plot(fusions_177_plot, track.num)


RCircos.Chromosome.Ideogram.Plot()

fus_gene_list <- fusions_177[which(fusions_177$PlotColor == "green" | fusions_177$PlotColor == "red"), ]
fus_gene_list <- separate(data = fus_gene_list, col = "Genes", into = c("Symbol", "Symbol.1"), sep = "--")
left_gene_list <- fus_gene_list[c("Chromosome", "chromStart", "chromEnd", "Symbol")]
right_gene_list <- fus_gene_list[c("Chromosome.1", "chromStart.1", "chromEnd.1", "Symbol.1")]
colnames(right_gene_list) <- c("Chromosome", "chromStart", "chromEnd", "Symbol")
fus_gene_list <- bind_rows(left_gene_list, right_gene_list)
fus_gene_list$PlotColor <- "blue"

mut_gene_list <- mut_177[,c("Chromosome", "chromStart", "chromEnd", "Symbol")]
mut_gene_list <-  ddply(mut_gene_list,.(Chromosome,chromStart,chromEnd,Symbol),nrow)
mut_gene_list <- mut_gene_list[which(mut_gene_list$V1 >= 3), ]
mut_gene_list$V1 <- NULL
mut_gene_list$PlotColor <- "red"

New_gene_list <- bind_rows(fus_gene_list, mut_gene_list)
#New_gene_list <- ddply(New_gene_list,.(Chromosome,chromStart,chromEnd,Symbol),nrow)
#New_gene_list$PlotColor <- "black"
#New_gene_list[which(New_gene_list$V1 == 2), "PlotColor"] <- "red"

data(RCircos.Gene.Label.Data)
name.col <- 4
side <- "in"
track.num <- 1
list_connector <- New_gene_list
RCircos.Gene.Connector.Plot(list_connector, track.num, side)

track.num <- 2
New_gene_list$PlotColor <- "black"
RCircos.Gene.Name.Plot(New_gene_list, name.col, track.num, side)

dev.off()

```

```{r circos_194}
chr.exclude <- NULL
data(UCSC.HG19.Human.CytoBandIdeogram)
cyto.info <- UCSC.HG19.Human.CytoBandIdeogram
tracks.inside <- 13
tracks.outside <- 0
RCircos.Set.Core.Components(cyto.info, chr.exclude ,tracks.inside, tracks.outside)

rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.position <- RCircos.Get.Plot.Positions()

rcircos.params$point.size <- 3
RCircos.Reset.Plot.Parameters(rcircos.params)

RCircos.List.Plot.Parameters()

# Actually making the plot

# pdf(file="RCircos_PDX194.pdf", height=8, width=8, compress=TRUE)
# pdf(file="Rcircos_PDX194.pdf")
png(file="RCircos_PDX194.png", res = 600, width = 4800, height = 4800)
RCircos.Set.Plot.Area()

par(mai=c(0.25, 0.25, 0.25, 0.25))
plot.new()
plot.window(c(-2.5, 2.5), c(-2.5, 2.5))

# scatter
mut_194 <- mutPrep("../CCF", "MAP194*")
mut_194_Diag <- mut_194[which(mut_194$PlotColor == "black"), ]
colnames(mut_194_Diag) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
# rownames(mut_194_Diag) <- paste(mut_194_Diag$Symbol, mut_194_Diag$chromStart)
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_194_Diag, data.col, side, inside.pos = 1.6, outside.pos = 1.65)

mut_194_Relapse <- mut_194[which(mut_194$PlotColor == "red"), ]
colnames(mut_194_Relapse) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
# rownames(mut_194_Relapse) <- paste(mut_194_Relapse$Symbol, mut_194_Relapse$chromStart)
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_194_Relapse, data.col, side, inside.pos = 1.65, outside.pos = 1.7)

# keep_mut <- c(rownames(mut_194_Diag),rownames(mut_194_Relapse))

mut_194_Ortho <- mut_194[which(mut_194$PlotColor == "blue"), ]
colnames(mut_194_Ortho) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_194_Ortho, data.col, side, inside.pos = 1.7, outside.pos = 1.75)

mut_194_Sc <- mut_194[which(mut_194$PlotColor == "green"), ]
colnames(mut_194_Sc) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_194_Sc, data.col, side, inside.pos = 1.75, outside.pos = 1.8)


# DIAG    
# Heatmap
cnv_fc_194 <- CNV_prep("../../facets/MAP194_Diag_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_194, data.col, side, inside.pos = 1.15, outside.pos = 1.25, max.value = 2, min.value = -2)

# META
# Heatmap
cnv_fc_194 <- CNV_prep("../../facets/MAP194_Relapse_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_194, data.col, side, inside.pos = 1.25, outside.pos = 1.35, max.value = 2, min.value = -2)

# ORTHO
# Heatmap
cnv_fc_194 <- CNV_prep("../../facets/MAP194_Ortho_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_194, data.col, side, inside.pos = 1.35, outside.pos = 1.45, max.value = 2, min.value = -2)

# SC
# Heatmap
cnv_fc_194 <- CNV_prep("../../facets/MAP194_Sc_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_194, data.col, side, inside.pos = 1.45, outside.pos = 1.55, max.value = 2, min.value = -2)

# DIAG
# Histogram
rna_diag <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP194_Diag")]
rna_diag$PlotColor <- "black"
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_diag, data.col, side, inside.pos = 0.9, outside.pos = 0.95, min.value=min(rna_diag$MAP194_Diag), max.value=max(rna_diag$MAP194_Diag))

# META
# Histogram
rna_meta <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP194_Relapse")]
rna_meta$PlotColor <- "red"
rna_meta$MAP194_Relapse <- ((rna_meta$MAP194_Relapse - rna_diag$MAP194_Diag) / rna_diag$MAP194_Diag) ^ 1/3
rna_meta <- rna_meta[complete.cases(rna_meta),]
rna_meta <- rna_meta[is.finite(rna_meta$MAP194_Relapse),]
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_meta, data.col, side, inside.pos = 0.95, outside.pos = 1, min.value=min(rna_meta$MAP194_Relapse), max.value=max(rna_meta$MAP194_Relapse))

# ORTHO
# Histogram
rna_ortho <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP194_Ortho")]
rna_ortho$PlotColor <- "blue"
rna_ortho$MAP194_Ortho <- ((rna_ortho$MAP194_Ortho - rna_diag$MAP194_Diag) / rna_diag$MAP194_Diag) ^ 1/3
rna_ortho <- rna_ortho[complete.cases(rna_ortho),]
rna_ortho <- rna_ortho[is.finite(rna_ortho$MAP194_Ortho),]
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_ortho, data.col, side, inside.pos = 1, outside.pos = 1.05, min.value=min(rna_ortho$MAP194_Ortho), max.value=max(rna_ortho$MAP194_Ortho))

# SC
# Histogram
rna_sc <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP194_Sc")]
rna_sc$PlotColor <- "green"
rna_sc$MAP194_Sc <- ((rna_sc$MAP194_Sc - rna_diag$MAP194_Diag) / rna_diag$MAP194_Diag) ^ 1/3
rna_sc <- rna_sc[complete.cases(rna_sc),]
rna_sc <- rna_sc[is.finite(rna_sc$MAP194_Sc),]
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_sc, data.col, side, inside.pos = 1.05, outside.pos = 1.1, min.value=min(rna_sc$MAP194_Sc), max.value=max(rna_sc$MAP194_Sc))


# Fusions data 194
data(RCircos.Link.Data)
fusions_194 <- fusionsPrep("../Fusions/data_Fusion_Inspector", "MAP194*")
fusions_194 <- fusions_194[which(fusions_194$PlotColor != "blue"),]
fusions_194 <- fusions_194[!(fusions_194$Genes %in% c("SDHAF2--PPP1R32", "COL1A1--COL3A1", "COL1A1--COL5A1")),]
fusions_194_plot <- fusions_194[c("Chromosome", "chromStart","chromEnd", "Chromosome.1", "chromStart.1", "chromEnd.1", "PlotColor")]
track.num <- 13
RCircos.Link.Plot(fusions_194_plot, track.num)


RCircos.Chromosome.Ideogram.Plot()

fus_gene_list <- fusions_194[which(fusions_194$PlotColor == "green" | fusions_194$PlotColor == "red"), ]
fus_gene_list <- separate(data = fus_gene_list, col = "Genes", into = c("Symbol", "Symbol.1"), sep = "--")
left_gene_list <- fus_gene_list[c("Chromosome", "chromStart", "chromEnd", "Symbol")]
right_gene_list <- fus_gene_list[c("Chromosome.1", "chromStart.1", "chromEnd.1", "Symbol.1")]
colnames(right_gene_list) <- c("Chromosome", "chromStart", "chromEnd", "Symbol")
fus_gene_list <- bind_rows(left_gene_list, right_gene_list)
fus_gene_list$PlotColor <- "blue"

mut_gene_list <- mut_194[,c("Chromosome", "chromStart", "chromEnd", "Symbol")]
mut_gene_list <-  ddply(mut_gene_list,.(Chromosome,chromStart,chromEnd,Symbol),nrow)
mut_gene_list <- mut_gene_list[which(mut_gene_list$V1 >= 3), ]
mut_gene_list$V1 <- NULL
mut_gene_list$PlotColor <- "red"

New_gene_list <- bind_rows(fus_gene_list, mut_gene_list)
#New_gene_list <- ddply(New_gene_list,.(Chromosome,chromStart,chromEnd,Symbol),nrow)
#New_gene_list$PlotColor <- "black"
#New_gene_list[which(New_gene_list$V1 == 2), "PlotColor"] <- "red"

data(RCircos.Gene.Label.Data)
name.col <- 4
side <- "in"
track.num <- 1
list_connector <- New_gene_list
RCircos.Gene.Connector.Plot(list_connector, track.num, side)

track.num <- 2
New_gene_list$PlotColor <- "black"
RCircos.Gene.Name.Plot(New_gene_list, name.col, track.num, side)

dev.off()
```

```{r circos_194}
chr.exclude <- NULL
data(UCSC.HG19.Human.CytoBandIdeogram)
cyto.info <- UCSC.HG19.Human.CytoBandIdeogram
tracks.inside <- 13
tracks.outside <- 0
RCircos.Set.Core.Components(cyto.info, chr.exclude ,tracks.inside, tracks.outside)

rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.position <- RCircos.Get.Plot.Positions()

rcircos.params$point.size <- 3
RCircos.Reset.Plot.Parameters(rcircos.params)

RCircos.List.Plot.Parameters()

# Actually making the plot

# pdf(file="RCircos_PDX194.pdf", height=8, width=8, compress=TRUE)
# pdf(file="Rcircos_PDX194.pdf")
png(file="RCircos_PDX194.png", res = 600, width = 4800, height = 4800)
RCircos.Set.Plot.Area()

par(mai=c(0.25, 0.25, 0.25, 0.25))
plot.new()
plot.window(c(-2.5, 2.5), c(-2.5, 2.5))

# scatter
mut_194 <- mutPrep("../Mutations", "MAP194*")
mut_194_Diag <- mut_194[which(mut_194$col == "black"), ]
mut_194_Diag <- mut_194_Diag[c("Chromosome", "chromStart", "chromEnd", "ccf_diag", "col","Symbol")]
colnames(mut_194_Diag) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
# rownames(mut_194_Diag) <- paste(mut_194_Diag$Symbol, mut_194_Diag$chromStart)
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_194_Diag, data.col, side, inside.pos = 1.6, outside.pos = 1.65)

mut_194_Relapse <- mut_194[which(mut_194$col == "red"), ]
mut_194_Relapse <- mut_194_Relapse[c("Chromosome", "chromStart", "chromEnd", "ccf_meta", "col","Symbol")]
colnames(mut_194_Relapse) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
# rownames(mut_194_Relapse) <- paste(mut_194_Relapse$Symbol, mut_194_Relapse$chromStart)
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_194_Relapse, data.col, side, inside.pos = 1.65, outside.pos = 1.7)

# keep_mut <- c(rownames(mut_194_Diag),rownames(mut_194_Relapse))

mut_194_Ortho <- mut_194[which(mut_194$col == "blue"), ]
mut_194_Ortho <- mut_194_Ortho[c("Chromosome", "chromStart", "chromEnd", "ccf_ortho", "col","Symbol")]
colnames(mut_194_Ortho) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_194_Ortho, data.col, side, inside.pos = 1.7, outside.pos = 1.75)

mut_194_Sc <- mut_194[which(mut_194$col == "green"), ]
mut_194_Sc <- mut_194_Sc[c("Chromosome", "chromStart", "chromEnd", "ccf_sc", "col","Symbol")]
colnames(mut_194_Sc) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_194_Sc, data.col, side, inside.pos = 1.75, outside.pos = 1.8)


# DIAG    
# Heatmap
cnv_fc_194 <- CNV_prep("../../facets/MAP194_Diag_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_194, data.col, side, inside.pos = 1.15, outside.pos = 1.25, max.value = 2, min.value = -2)

# META
# Heatmap
cnv_fc_194 <- CNV_prep("../../facets/MAP194_Relapse_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_194, data.col, side, inside.pos = 1.25, outside.pos = 1.35, max.value = 2, min.value = -2)

# ORTHO
# Heatmap
cnv_fc_194 <- CNV_prep("../../facets/MAP194_Ortho_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_194, data.col, side, inside.pos = 1.35, outside.pos = 1.45, max.value = 2, min.value = -2)

# SC
# Heatmap
cnv_fc_194 <- CNV_prep("../../facets/MAP194_Sc_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_194, data.col, side, inside.pos = 1.45, outside.pos = 1.55, max.value = 2, min.value = -2)

# DIAG
# Histogram
rna_diag <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP194_Diag")]
rna_diag$PlotColor <- "black"
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_diag, data.col, side, inside.pos = 0.9, outside.pos = 0.95, min.value=min(rna_diag$MAP194_Diag), max.value=max(rna_diag$MAP194_Diag))

# META
# Histogram
rna_meta <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP194_Relapse")]
rna_meta$PlotColor <- "red"
rna_meta$MAP194_Relapse <- ((rna_meta$MAP194_Relapse - rna_diag$MAP194_Diag) / rna_diag$MAP194_Diag) ^ 1/3
rna_meta <- rna_meta[complete.cases(rna_meta),]
rna_meta <- rna_meta[is.finite(rna_meta$MAP194_Relapse),]
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_meta, data.col, side, inside.pos = 0.95, outside.pos = 1, min.value=min(rna_meta$MAP194_Relapse), max.value=max(rna_meta$MAP194_Relapse))

# ORTHO
# Histogram
rna_ortho <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP194_Ortho")]
rna_ortho$PlotColor <- "blue"
rna_ortho$MAP194_Ortho <- ((rna_ortho$MAP194_Ortho - rna_diag$MAP194_Diag) / rna_diag$MAP194_Diag) ^ 1/3
rna_ortho <- rna_ortho[complete.cases(rna_ortho),]
rna_ortho <- rna_ortho[is.finite(rna_ortho$MAP194_Ortho),]
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_ortho, data.col, side, inside.pos = 1, outside.pos = 1.05, min.value=min(rna_ortho$MAP194_Ortho), max.value=max(rna_ortho$MAP194_Ortho))

# SC
# Histogram
rna_sc <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP194_Sc")]
rna_sc$PlotColor <- "green"
rna_sc$MAP194_Sc <- ((rna_sc$MAP194_Sc - rna_diag$MAP194_Diag) / rna_diag$MAP194_Diag) ^ 1/3
rna_sc <- rna_sc[complete.cases(rna_sc),]
rna_sc <- rna_sc[is.finite(rna_sc$MAP194_Sc),]
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_sc, data.col, side, inside.pos = 1.05, outside.pos = 1.1, min.value=min(rna_sc$MAP194_Sc), max.value=max(rna_sc$MAP194_Sc))


# Fusions data 194
data(RCircos.Link.Data)
fusions_194 <- fusionsPrep("../Fusions/data_Fusion_Inspector", "MAP194*")
fusions_194 <- fusions_194[which(fusions_194$PlotColor != "blue"),]
fusions_194 <- fusions_194[!(fusions_194$Genes %in% c("SDHAF2--PPP1R32", "COL1A1--COL3A1", "COL1A1--COL5A1")),]
fusions_194_plot <- fusions_194[c("Chromosome", "chromStart","chromEnd", "Chromosome.1", "chromStart.1", "chromEnd.1", "PlotColor")]
track.num <- 13
RCircos.Link.Plot(fusions_194_plot, track.num)


RCircos.Chromosome.Ideogram.Plot()

fus_gene_list <- fusions_194[which(fusions_194$PlotColor == "green" | fusions_194$PlotColor == "red"), ]
fus_gene_list <- separate(data = fus_gene_list, col = "Genes", into = c("Symbol", "Symbol.1"), sep = "--")
left_gene_list <- fus_gene_list[c("Chromosome", "chromStart", "chromEnd", "Symbol")]
right_gene_list <- fus_gene_list[c("Chromosome.1", "chromStart.1", "chromEnd.1", "Symbol.1")]
colnames(right_gene_list) <- c("Chromosome", "chromStart", "chromEnd", "Symbol")
fus_gene_list <- bind_rows(left_gene_list, right_gene_list)
fus_gene_list$PlotColor <- "blue"

mut_gene_list <- mut_194[,c("Chromosome", "chromStart", "chromEnd", "Symbol")]
mut_gene_list <-  ddply(mut_gene_list,.(Chromosome,chromStart,chromEnd,Symbol),nrow)
mut_gene_list <- mut_gene_list[which(mut_gene_list$V1 >= 3), ]
mut_gene_list$V1 <- NULL
mut_gene_list$PlotColor <- "red"

New_gene_list <- bind_rows(fus_gene_list, mut_gene_list)
#New_gene_list <- ddply(New_gene_list,.(Chromosome,chromStart,chromEnd,Symbol),nrow)
#New_gene_list$PlotColor <- "black"
#New_gene_list[which(New_gene_list$V1 == 2), "PlotColor"] <- "red"

data(RCircos.Gene.Label.Data)
name.col <- 4
side <- "in"
track.num <- 1
list_connector <- New_gene_list
RCircos.Gene.Connector.Plot(list_connector, track.num, side)

track.num <- 2
New_gene_list$PlotColor <- "black"
RCircos.Gene.Name.Plot(New_gene_list, name.col, track.num, side)

dev.off()
```


```{r circos_217}
chr.exclude <- NULL
data(UCSC.HG19.Human.CytoBandIdeogram)
cyto.info <- UCSC.HG19.Human.CytoBandIdeogram
tracks.inside <- 13
tracks.outside <- 0
RCircos.Set.Core.Components(cyto.info, chr.exclude ,tracks.inside, tracks.outside)

rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.position <- RCircos.Get.Plot.Positions()

rcircos.params$point.size <- 3
RCircos.Reset.Plot.Parameters(rcircos.params)

RCircos.List.Plot.Parameters()

# Actually making the plot

# pdf(file="RCircos_PDX217.pdf", height=8, width=8, compress=TRUE)
# pdf(file="Rcircos_PDX217.pdf")
png(file="RCircos_PDX217.png", res = 600, width = 4800, height = 4800)
RCircos.Set.Plot.Area()

par(mai=c(0.25, 0.25, 0.25, 0.25))
plot.new()
plot.window(c(-2.5, 2.5), c(-2.5, 2.5))

# scatter
mut_217 <- mutPrep("../CCF", "MAP217*")
mut_217_Diag <- mut_217[which(mut_217$PlotColor == "black"), ]
colnames(mut_217_Diag) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
# rownames(mut_217_Diag) <- paste(mut_217_Diag$Symbol, mut_217_Diag$chromStart)
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_217_Diag, data.col, side, inside.pos = 1.6, outside.pos = 1.65)

mut_217_Relapse <- mut_217[which(mut_217$PlotColor == "red"), ]
colnames(mut_217_Relapse) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
# rownames(mut_217_Relapse) <- paste(mut_217_Relapse$Symbol, mut_217_Relapse$chromStart)
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_217_Relapse, data.col, side, inside.pos = 1.65, outside.pos = 1.7)

# keep_mut <- c(rownames(mut_217_Diag),rownames(mut_217_Relapse))

mut_217_Ortho <- mut_217[which(mut_217$PlotColor == "blue"), ]
colnames(mut_217_Ortho) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_217_Ortho, data.col, side, inside.pos = 1.7, outside.pos = 1.75)

mut_217_Sc <- mut_217[which(mut_217$PlotColor == "green"), ]
colnames(mut_217_Sc) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_217_Sc, data.col, side, inside.pos = 1.75, outside.pos = 1.8)


# DIAG    
# Heatmap
cnv_fc_217 <- CNV_prep("../../facets/MAP217_Diag_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_217, data.col, side, inside.pos = 1.15, outside.pos = 1.25, max.value = 2, min.value = -2)

# META
# Heatmap
cnv_fc_217 <- CNV_prep("../../facets/MAP217_Relapse_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_217, data.col, side, inside.pos = 1.25, outside.pos = 1.35, max.value = 2, min.value = -2)

# ORTHO
# Heatmap
cnv_fc_217 <- CNV_prep("../../facets/MAP217_Ortho_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_217, data.col, side, inside.pos = 1.35, outside.pos = 1.45, max.value = 2, min.value = -2)

# SC
# Heatmap
cnv_fc_217 <- CNV_prep("../../facets/MAP217_Sc_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_217, data.col, side, inside.pos = 1.45, outside.pos = 1.55, max.value = 2, min.value = -2)

# DIAG
# Histogram
rna_diag <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP217_Diag")]
rna_diag$PlotColor <- "black"
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_diag, data.col, side, inside.pos = 0.9, outside.pos = 0.95, min.value=min(rna_diag$MAP217_Diag), max.value=max(rna_diag$MAP217_Diag))

# META
# Histogram
rna_meta <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP217_Relapse")]
rna_meta$PlotColor <- "red"
rna_meta$MAP217_Relapse <- ((rna_meta$MAP217_Relapse - rna_diag$MAP217_Diag) / rna_diag$MAP217_Diag) ^ 1/3
rna_meta <- rna_meta[complete.cases(rna_meta),]
rna_meta <- rna_meta[is.finite(rna_meta$MAP217_Relapse),]
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_meta, data.col, side, inside.pos = 0.95, outside.pos = 1, min.value=min(rna_meta$MAP217_Relapse), max.value=max(rna_meta$MAP217_Relapse))

# ORTHO
# Histogram
rna_ortho <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP217_Ortho")]
rna_ortho$PlotColor <- "blue"
rna_ortho$MAP217_Ortho <- ((rna_ortho$MAP217_Ortho - rna_diag$MAP217_Diag) / rna_diag$MAP217_Diag) ^ 1/3
rna_ortho <- rna_ortho[complete.cases(rna_ortho),]
rna_ortho <- rna_ortho[is.finite(rna_ortho$MAP217_Ortho),]
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_ortho, data.col, side, inside.pos = 1, outside.pos = 1.05, min.value=min(rna_ortho$MAP217_Ortho), max.value=max(rna_ortho$MAP217_Ortho))

# SC
# Histogram
rna_sc <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP217_Sc")]
rna_sc$PlotColor <- "green"
rna_sc$MAP217_Sc <- ((rna_sc$MAP217_Sc - rna_diag$MAP217_Diag) / rna_diag$MAP217_Diag) ^ 1/3
rna_sc <- rna_sc[complete.cases(rna_sc),]
rna_sc <- rna_sc[is.finite(rna_sc$MAP217_Sc),]
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_sc, data.col, side, inside.pos = 1.05, outside.pos = 1.1, min.value=min(rna_sc$MAP217_Sc), max.value=max(rna_sc$MAP217_Sc))


# Fusions data 217
data(RCircos.Link.Data)
fusions_217 <- fusionsPrep("../Fusions/data_Fusion_Inspector", "MAP217*")
fusions_217 <- fusions_217[which(fusions_217$PlotColor != "blue"),]
fusions_217 <- fusions_217[!(fusions_217$Genes %in% c("SDHAF2--PPP1R32", "COL1A1--COL3A1", "COL1A1--COL5A1")),]
fusions_217_plot <- fusions_217[c("Chromosome", "chromStart","chromEnd", "Chromosome.1", "chromStart.1", "chromEnd.1", "PlotColor")]
track.num <- 13
RCircos.Link.Plot(fusions_217_plot, track.num)


RCircos.Chromosome.Ideogram.Plot()

fus_gene_list <- fusions_217[which(fusions_217$PlotColor == "green" | fusions_217$PlotColor == "red"), ]
fus_gene_list <- separate(data = fus_gene_list, col = "Genes", into = c("Symbol", "Symbol.1"), sep = "--")
left_gene_list <- fus_gene_list[c("Chromosome", "chromStart", "chromEnd", "Symbol")]
right_gene_list <- fus_gene_list[c("Chromosome.1", "chromStart.1", "chromEnd.1", "Symbol.1")]
colnames(right_gene_list) <- c("Chromosome", "chromStart", "chromEnd", "Symbol")
fus_gene_list <- bind_rows(left_gene_list, right_gene_list)
fus_gene_list$PlotColor <- "blue"

mut_gene_list <- mut_217[,c("Chromosome", "chromStart", "chromEnd", "Symbol")]
mut_gene_list <-  ddply(mut_gene_list,.(Chromosome,chromStart,chromEnd,Symbol),nrow)
mut_gene_list <- mut_gene_list[which(mut_gene_list$V1 >= 3), ]
mut_gene_list$V1 <- NULL
mut_gene_list$PlotColor <- "red"

New_gene_list <- bind_rows(fus_gene_list, mut_gene_list)
#New_gene_list <- ddply(New_gene_list,.(Chromosome,chromStart,chromEnd,Symbol),nrow)
#New_gene_list$PlotColor <- "black"
#New_gene_list[which(New_gene_list$V1 == 2), "PlotColor"] <- "red"

data(RCircos.Gene.Label.Data)
name.col <- 4
side <- "in"
track.num <- 1
list_connector <- New_gene_list
RCircos.Gene.Connector.Plot(list_connector, track.num, side)

track.num <- 2
New_gene_list$PlotColor <- "black"
RCircos.Gene.Name.Plot(New_gene_list, name.col, track.num, side)

dev.off()
```


```{r circos_342}
chr.exclude <- NULL
data(UCSC.HG19.Human.CytoBandIdeogram)
cyto.info <- UCSC.HG19.Human.CytoBandIdeogram
tracks.inside <- 13
tracks.outside <- 0
RCircos.Set.Core.Components(cyto.info, chr.exclude ,tracks.inside, tracks.outside)

rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.position <- RCircos.Get.Plot.Positions()

rcircos.params$point.size <- 3
RCircos.Reset.Plot.Parameters(rcircos.params)

RCircos.List.Plot.Parameters()

# Actually making the plot

# pdf(file="RCircos_PDX342.pdf", height=8, width=8, compress=TRUE)
# pdf(file="Rcircos_PDX342.pdf")
png(file="RCircos_PDX342.png", res = 600, width = 4800, height = 4800)
RCircos.Set.Plot.Area()

par(mai=c(0.25, 0.25, 0.25, 0.25))
plot.new()
plot.window(c(-2.5, 2.5), c(-2.5, 2.5))

# scatter
mut_342 <- mutPrep("../CCF", "MAP342*")
mut_342_Diag <- mut_342[which(mut_342$PlotColor == "black"), ]
colnames(mut_342_Diag) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
# rownames(mut_342_Diag) <- paste(mut_342_Diag$Symbol, mut_342_Diag$chromStart)
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_342_Diag, data.col, side, inside.pos = 1.6, outside.pos = 1.65)

mut_342_Relapse <- mut_342[which(mut_342$PlotColor == "red"), ]
colnames(mut_342_Relapse) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
# rownames(mut_342_Relapse) <- paste(mut_342_Relapse$Symbol, mut_342_Relapse$chromStart)
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_342_Relapse, data.col, side, inside.pos = 1.65, outside.pos = 1.7)

# keep_mut <- c(rownames(mut_342_Diag),rownames(mut_342_Relapse))

mut_342_Ortho <- mut_342[which(mut_342$PlotColor == "blue"), ]
colnames(mut_342_Ortho) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_342_Ortho, data.col, side, inside.pos = 1.7, outside.pos = 1.75)

mut_342_Sc <- mut_342[which(mut_342$PlotColor == "green"), ]
colnames(mut_342_Sc) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_342_Sc, data.col, side, inside.pos = 1.75, outside.pos = 1.8)


# DIAG    
# Heatmap
cnv_fc_342 <- CNV_prep("../../facets/MAP342_Diag_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_342, data.col, side, inside.pos = 1.15, outside.pos = 1.25, max.value = 2, min.value = -2)

# META
# Heatmap
cnv_fc_342 <- CNV_prep("../../facets/MAP342_Relapse_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_342, data.col, side, inside.pos = 1.25, outside.pos = 1.35, max.value = 2, min.value = -2)

# ORTHO
# Heatmap
cnv_fc_342 <- CNV_prep("../../facets/MAP342_Ortho_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_342, data.col, side, inside.pos = 1.35, outside.pos = 1.45, max.value = 2, min.value = -2)

# SC
# Heatmap
cnv_fc_342 <- CNV_prep("../../facets/MAP342_Sc_Ortho-meta_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_342, data.col, side, inside.pos = 1.45, outside.pos = 1.55, max.value = 2, min.value = -2)

# DIAG
# Histogram
rna_diag <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP342_Diag")]
rna_diag$PlotColor <- "black"
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_diag, data.col, side, inside.pos = 0.9, outside.pos = 0.95, min.value=min(rna_diag$MAP342_Diag), max.value=max(rna_diag$MAP342_Diag))

# META
# Histogram
rna_meta <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP342_Relapse")]
rna_meta$PlotColor <- "red"
rna_meta$MAP342_Relapse <- ((rna_meta$MAP342_Relapse - rna_diag$MAP342_Diag) / rna_diag$MAP342_Diag) ^ 1/3
rna_meta <- rna_meta[complete.cases(rna_meta),]
rna_meta <- rna_meta[is.finite(rna_meta$MAP342_Relapse),]
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_meta, data.col, side, inside.pos = 0.95, outside.pos = 1, min.value=min(rna_meta$MAP342_Relapse), max.value=max(rna_meta$MAP342_Relapse))

# ORTHO
# Histogram
rna_ortho <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP342_Ortho")]
rna_ortho$PlotColor <- "blue"
rna_ortho$MAP342_Ortho <- ((rna_ortho$MAP342_Ortho - rna_diag$MAP342_Diag) / rna_diag$MAP342_Diag) ^ 1/3
rna_ortho <- rna_ortho[complete.cases(rna_ortho),]
rna_ortho <- rna_ortho[is.finite(rna_ortho$MAP342_Ortho),]
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_ortho, data.col, side, inside.pos = 1, outside.pos = 1.05, min.value=min(rna_ortho$MAP342_Ortho), max.value=max(rna_ortho$MAP342_Ortho))

# SC
# Histogram
rna_sc <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP342_Ortho.meta")]
rna_sc$PlotColor <- "green"
rna_sc$MAP342_Ortho.meta <- ((rna_sc$MAP342_Ortho.meta - rna_diag$MAP342_Diag) / rna_diag$MAP342_Diag) ^ 1/3
rna_sc <- rna_sc[complete.cases(rna_sc),]
rna_sc <- rna_sc[is.finite(rna_sc$MAP342_Ortho.meta),]
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_sc, data.col, side, inside.pos = 1.05, outside.pos = 1.1, min.value=min(rna_sc$MAP342_Ortho.meta), max.value=max(rna_sc$MAP342_Ortho.meta))


# Fusions data 342
data(RCircos.Link.Data)
fusions_342 <- fusionsPrep("../Fusions/data_Fusion_Inspector", "MAP342*")
fusions_342 <- fusions_342[which(fusions_342$PlotColor != "blue"),]
fusions_342 <- fusions_342[!(fusions_342$Genes %in% c("SDHAF2--PPP1R32", "COL1A1--COL3A1", "COL1A1--COL5A1")),]
fusions_342_plot <- fusions_342[c("Chromosome", "chromStart","chromEnd", "Chromosome.1", "chromStart.1", "chromEnd.1", "PlotColor")]
track.num <- 13
RCircos.Link.Plot(fusions_342_plot, track.num)


RCircos.Chromosome.Ideogram.Plot()

fus_gene_list <- fusions_342[which(fusions_342$PlotColor == "green" | fusions_342$PlotColor == "red"), ]
fus_gene_list <- separate(data = fus_gene_list, col = "Genes", into = c("Symbol", "Symbol.1"), sep = "--")
left_gene_list <- fus_gene_list[c("Chromosome", "chromStart", "chromEnd", "Symbol")]
right_gene_list <- fus_gene_list[c("Chromosome.1", "chromStart.1", "chromEnd.1", "Symbol.1")]
colnames(right_gene_list) <- c("Chromosome", "chromStart", "chromEnd", "Symbol")
fus_gene_list <- bind_rows(left_gene_list, right_gene_list)
fus_gene_list$PlotColor <- "blue"

mut_gene_list <- mut_342[,c("Chromosome", "chromStart", "chromEnd", "Symbol")]
mut_gene_list <-  ddply(mut_gene_list,.(Chromosome,chromStart,chromEnd,Symbol),nrow)
mut_gene_list <- mut_gene_list[which(mut_gene_list$V1 >= 3), ]
mut_gene_list$V1 <- NULL
mut_gene_list$PlotColor <- "red"

New_gene_list <- bind_rows(fus_gene_list, mut_gene_list)
#New_gene_list <- ddply(New_gene_list,.(Chromosome,chromStart,chromEnd,Symbol),nrow)
#New_gene_list$PlotColor <- "black"
#New_gene_list[which(New_gene_list$V1 == 2), "PlotColor"] <- "red"

data(RCircos.Gene.Label.Data)
name.col <- 4
side <- "in"
track.num <- 1
list_connector <- New_gene_list
RCircos.Gene.Connector.Plot(list_connector, track.num, side)

track.num <- 2
New_gene_list$PlotColor <- "black"
RCircos.Gene.Name.Plot(New_gene_list, name.col, track.num, side)

dev.off()
```

```{r circos_469}
chr.exclude <- NULL
data(UCSC.HG19.Human.CytoBandIdeogram)
cyto.info <- UCSC.HG19.Human.CytoBandIdeogram
tracks.inside <- 13
tracks.outside <- 0
RCircos.Set.Core.Components(cyto.info, chr.exclude ,tracks.inside, tracks.outside)

rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.position <- RCircos.Get.Plot.Positions()

rcircos.params$point.size <- 3
RCircos.Reset.Plot.Parameters(rcircos.params)

RCircos.List.Plot.Parameters()

# Actually making the plot

# pdf(file="RCircos_PDX469.pdf", height=8, width=8, compress=TRUE)
# pdf(file="Rcircos_PDX469.pdf")
png(file="RCircos_PDX469.png", res = 600, width = 4800, height = 4800)
RCircos.Set.Plot.Area()

par(mai=c(0.25, 0.25, 0.25, 0.25))
plot.new()
plot.window(c(-2.5, 2.5), c(-2.5, 2.5))

# scatter
mut_469 <- mutPrep("../CCF", "MAP469*")
mut_469_Diag <- mut_469[which(mut_469$PlotColor == "black"), ]
colnames(mut_469_Diag) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
# rownames(mut_469_Diag) <- paste(mut_469_Diag$Symbol, mut_469_Diag$chromStart)
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_469_Diag, data.col, side, inside.pos = 1.6, outside.pos = 1.65)

mut_469_Relapse <- mut_469[which(mut_469$PlotColor == "red"), ]
colnames(mut_469_Relapse) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
# rownames(mut_469_Relapse) <- paste(mut_469_Relapse$Symbol, mut_469_Relapse$chromStart)
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_469_Relapse, data.col, side, inside.pos = 1.65, outside.pos = 1.7)

# keep_mut <- c(rownames(mut_469_Diag),rownames(mut_469_Relapse))

mut_469_Ortho <- mut_469[which(mut_469$PlotColor == "blue"), ]
colnames(mut_469_Ortho) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_469_Ortho, data.col, side, inside.pos = 1.7, outside.pos = 1.75)

mut_469_Sc <- mut_469[which(mut_469$PlotColor == "green"), ]
colnames(mut_469_Sc) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_469_Sc, data.col, side, inside.pos = 1.75, outside.pos = 1.8)


# DIAG    
# Heatmap
cnv_fc_469 <- CNV_prep("../../facets/MAP469_Diag_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_469, data.col, side, inside.pos = 1.15, outside.pos = 1.25, max.value = 2, min.value = -2)

# META
# Heatmap
cnv_fc_469 <- CNV_prep("../../facets/MAP469_Relapse_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_469, data.col, side, inside.pos = 1.25, outside.pos = 1.35, max.value = 2, min.value = -2)

# ORTHO
# Heatmap
cnv_fc_469 <- CNV_prep("../../facets/MAP469_Ortho_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_469, data.col, side, inside.pos = 1.35, outside.pos = 1.45, max.value = 2, min.value = -2)

# SC
# Heatmap
cnv_fc_469 <- CNV_prep("../../facets/MAP469_Sc_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_469, data.col, side, inside.pos = 1.45, outside.pos = 1.55, max.value = 2, min.value = -2)

# DIAG
# Histogram
rna_diag <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP469_Diag")]
rna_diag$PlotColor <- "black"
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_diag, data.col, side, inside.pos = 0.9, outside.pos = 0.95, min.value=min(rna_diag$MAP469_Diag), max.value=max(rna_diag$MAP469_Diag))

# META
# Histogram
rna_meta <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP469_Relapse")]
rna_meta$PlotColor <- "red"
rna_meta$MAP469_Relapse <- ((rna_meta$MAP469_Relapse - rna_diag$MAP469_Diag) / rna_diag$MAP469_Diag) ^ 1/3
rna_meta <- rna_meta[complete.cases(rna_meta),]
rna_meta <- rna_meta[is.finite(rna_meta$MAP469_Relapse),]
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_meta, data.col, side, inside.pos = 0.95, outside.pos = 1, min.value=min(rna_meta$MAP469_Relapse), max.value=max(rna_meta$MAP469_Relapse))

# ORTHO
# Histogram
rna_ortho <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP469_Ortho")]
rna_ortho$PlotColor <- "blue"
rna_ortho$MAP469_Ortho <- ((rna_ortho$MAP469_Ortho - rna_diag$MAP469_Diag) / rna_diag$MAP469_Diag) ^ 1/3
rna_ortho <- rna_ortho[complete.cases(rna_ortho),]
rna_ortho <- rna_ortho[is.finite(rna_ortho$MAP469_Ortho),]
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_ortho, data.col, side, inside.pos = 1, outside.pos = 1.05, min.value=min(rna_ortho$MAP469_Ortho), max.value=max(rna_ortho$MAP469_Ortho))

# SC
# Histogram
rna_sc <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP469_Sc")]
rna_sc$PlotColor <- "green"
rna_sc$MAP469_Sc <- ((rna_sc$MAP469_Sc - rna_diag$MAP469_Diag) / rna_diag$MAP469_Diag) ^ 1/3
rna_sc <- rna_sc[complete.cases(rna_sc),]
rna_sc <- rna_sc[is.finite(rna_sc$MAP469_Sc),]
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_sc, data.col, side, inside.pos = 1.05, outside.pos = 1.1, min.value=min(rna_sc$MAP469_Sc), max.value=max(rna_sc$MAP469_Sc))


# Fusions data 469
data(RCircos.Link.Data)
fusions_469 <- fusionsPrep("../Fusions/data_Fusion_Inspector", "MAP469*")
fusions_469 <- fusions_469[which(fusions_469$PlotColor != "blue"),]
fusions_469 <- fusions_469[!(fusions_469$Genes %in% c("SDHAF2--PPP1R32", "COL1A1--COL3A1", "COL1A1--COL5A1")),]
fusions_469_plot <- fusions_469[c("Chromosome", "chromStart","chromEnd", "Chromosome.1", "chromStart.1", "chromEnd.1", "PlotColor")]
track.num <- 13
RCircos.Link.Plot(fusions_469_plot, track.num)


RCircos.Chromosome.Ideogram.Plot()

fus_gene_list <- fusions_469[which(fusions_469$PlotColor == "green" | fusions_469$PlotColor == "red"), ]
fus_gene_list <- separate(data = fus_gene_list, col = "Genes", into = c("Symbol", "Symbol.1"), sep = "--")
left_gene_list <- fus_gene_list[c("Chromosome", "chromStart", "chromEnd", "Symbol")]
right_gene_list <- fus_gene_list[c("Chromosome.1", "chromStart.1", "chromEnd.1", "Symbol.1")]
colnames(right_gene_list) <- c("Chromosome", "chromStart", "chromEnd", "Symbol")
fus_gene_list <- bind_rows(left_gene_list, right_gene_list)
fus_gene_list$PlotColor <- "blue"

mut_gene_list <- mut_469[,c("Chromosome", "chromStart", "chromEnd", "Symbol")]
mut_gene_list <-  ddply(mut_gene_list,.(Chromosome,chromStart,chromEnd,Symbol),nrow)
mut_gene_list <- mut_gene_list[which(mut_gene_list$V1 >= 3), ]
mut_gene_list$V1 <- NULL
mut_gene_list$PlotColor <- "red"

New_gene_list <- bind_rows(fus_gene_list, mut_gene_list)
#New_gene_list <- ddply(New_gene_list,.(Chromosome,chromStart,chromEnd,Symbol),nrow)
#New_gene_list$PlotColor <- "black"
#New_gene_list[which(New_gene_list$V1 == 2), "PlotColor"] <- "red"

data(RCircos.Gene.Label.Data)
name.col <- 4
side <- "in"
track.num <- 1
list_connector <- New_gene_list
RCircos.Gene.Connector.Plot(list_connector, track.num, side)

track.num <- 2
New_gene_list$PlotColor <- "black"
RCircos.Gene.Name.Plot(New_gene_list, name.col, track.num, side)

dev.off()
```


```{r circos_559}
chr.exclude <- NULL
data(UCSC.HG19.Human.CytoBandIdeogram)
cyto.info <- UCSC.HG19.Human.CytoBandIdeogram
tracks.inside <- 13
tracks.outside <- 0
RCircos.Set.Core.Components(cyto.info, chr.exclude ,tracks.inside, tracks.outside)

rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.position <- RCircos.Get.Plot.Positions()

rcircos.params$point.size <- 3
RCircos.Reset.Plot.Parameters(rcircos.params)

RCircos.List.Plot.Parameters()

# Actually making the plot

# pdf(file="RCircos_PDX559.pdf", height=8, width=8, compress=TRUE)
# pdf(file="Rcircos_PDX559.pdf")
png(file="RCircos_PDX559.png", res = 600, width = 4800, height = 4800)
RCircos.Set.Plot.Area()

par(mai=c(0.25, 0.25, 0.25, 0.25))
plot.new()
plot.window(c(-2.5, 2.5), c(-2.5, 2.5))

# scatter
mut_559 <- mutPrep("../CCF", "MAP559*")

mut_559_Relapse <- mut_559[which(mut_559$PlotColor == "red"), ]
colnames(mut_559_Relapse) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
# rownames(mut_559_Relapse) <- paste(mut_559_Relapse$Symbol, mut_559_Relapse$chromStart)
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_559_Relapse, data.col, side, inside.pos = 1.65, outside.pos = 1.7)

# keep_mut <- c(rownames(mut_559_Diag),rownames(mut_559_Relapse))

mut_559_Ortho <- mut_559[which(mut_559$PlotColor == "blue"), ]
colnames(mut_559_Ortho) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_559_Ortho, data.col, side, inside.pos = 1.7, outside.pos = 1.75)

mut_559_Sc <- mut_559[which(mut_559$PlotColor == "green"), ]
colnames(mut_559_Sc) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_559_Sc, data.col, side, inside.pos = 1.75, outside.pos = 1.8)

# META
# Heatmap
cnv_fc_559 <- CNV_prep("../../facets/MAP559_Relapse_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_559, data.col, side, inside.pos = 1.25, outside.pos = 1.35, max.value = 2, min.value = -2)

# ORTHO
# Heatmap
cnv_fc_559 <- CNV_prep("../../facets/MAP559_Ortho_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_559, data.col, side, inside.pos = 1.35, outside.pos = 1.45, max.value = 2, min.value = -2)

# SC
# Heatmap
cnv_fc_559 <- CNV_prep("../../facets/MAP559_Sc_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_559, data.col, side, inside.pos = 1.45, outside.pos = 1.55, max.value = 2, min.value = -2)

# META
# Histogram
rna_meta <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP559_Relapse")]
rna_meta$PlotColor <- "red"
rna_meta <- rna_meta[complete.cases(rna_meta),]
rna_meta <- rna_meta[is.finite(rna_meta$MAP559_Relapse),]
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_meta, data.col, side, inside.pos = 0.95, outside.pos = 1, min.value=min(rna_meta$MAP559_Relapse), max.value=max(rna_meta$MAP559_Relapse))

# ORTHO
# Histogram
rna_ortho <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP559_Ortho")]
rna_ortho$PlotColor <- "blue"
rna_ortho$MAP559_Ortho <- ((rna_ortho$MAP559_Ortho - rna_meta$MAP559_Relapse) / rna_meta$MAP559_Relapse) ^ 1/3
rna_ortho <- rna_ortho[complete.cases(rna_ortho),]
rna_ortho <- rna_ortho[is.finite(rna_ortho$MAP559_Ortho),]
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_ortho, data.col, side, inside.pos = 1, outside.pos = 1.05, min.value=min(rna_ortho$MAP559_Ortho), max.value=max(rna_ortho$MAP559_Ortho))

# SC
# Histogram
rna_sc <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP559_Sc")]
rna_sc$PlotColor <- "green"
rna_sc$MAP559_Sc <- ((rna_sc$MAP559_Sc - rna_meta$MAP559_Relapse) / rna_meta$MAP559_Relapse) ^ 1/3
rna_sc <- rna_sc[complete.cases(rna_sc),]
rna_sc <- rna_sc[is.finite(rna_sc$MAP559_Sc),]
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_sc, data.col, side, inside.pos = 1.05, outside.pos = 1.1, min.value=min(rna_sc$MAP559_Sc), max.value=max(rna_sc$MAP559_Sc))


# Fusions data 559
data(RCircos.Link.Data)
fusions_559 <- fusionsPrep("../Fusions/data_Fusion_Inspector", "MAP559*")
fusions_559 <- fusions_559[which(fusions_559$PlotColor != "blue"),]
fusions_559 <- fusions_559[!(fusions_559$Genes %in% c("SDHAF2--PPP1R32", "COL1A1--COL3A1", "COL1A1--COL5A1")),]
fusions_559_plot <- fusions_559[c("Chromosome", "chromStart","chromEnd", "Chromosome.1", "chromStart.1", "chromEnd.1", "PlotColor")]
track.num <- 13
RCircos.Link.Plot(fusions_559_plot, track.num)


RCircos.Chromosome.Ideogram.Plot()

fus_gene_list <- fusions_559[which(fusions_559$PlotColor == "green" | fusions_559$PlotColor == "red"), ]
fus_gene_list <- separate(data = fus_gene_list, col = "Genes", into = c("Symbol", "Symbol.1"), sep = "--")
left_gene_list <- fus_gene_list[c("Chromosome", "chromStart", "chromEnd", "Symbol")]
right_gene_list <- fus_gene_list[c("Chromosome.1", "chromStart.1", "chromEnd.1", "Symbol.1")]
colnames(right_gene_list) <- c("Chromosome", "chromStart", "chromEnd", "Symbol")
fus_gene_list <- bind_rows(left_gene_list, right_gene_list)
fus_gene_list$PlotColor <- "blue"

mut_gene_list <- mut_559[,c("Chromosome", "chromStart", "chromEnd", "Symbol")]
mut_gene_list <-  ddply(mut_gene_list,.(Chromosome,chromStart,chromEnd,Symbol),nrow)
mut_gene_list <- mut_gene_list[which(mut_gene_list$V1 >= 3), ]
mut_gene_list$V1 <- NULL
mut_gene_list$PlotColor <- "red"

New_gene_list <- bind_rows(fus_gene_list, mut_gene_list)
#New_gene_list <- ddply(New_gene_list,.(Chromosome,chromStart,chromEnd,Symbol),nrow)
#New_gene_list$PlotColor <- "black"
#New_gene_list[which(New_gene_list$V1 == 2), "PlotColor"] <- "red"

data(RCircos.Gene.Label.Data)
name.col <- 4
side <- "in"
track.num <- 1
list_connector <- New_gene_list
RCircos.Gene.Connector.Plot(list_connector, track.num, side)

track.num <- 2
New_gene_list$PlotColor <- "black"
RCircos.Gene.Name.Plot(New_gene_list, name.col, track.num, side)

dev.off()
```


```{r circos_636}
chr.exclude <- NULL
data(UCSC.HG19.Human.CytoBandIdeogram)
cyto.info <- UCSC.HG19.Human.CytoBandIdeogram
tracks.inside <- 13
tracks.outside <- 0
RCircos.Set.Core.Components(cyto.info, chr.exclude ,tracks.inside, tracks.outside)

rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.position <- RCircos.Get.Plot.Positions()

rcircos.params$point.size <- 3
RCircos.Reset.Plot.Parameters(rcircos.params)

RCircos.List.Plot.Parameters()

# Actually making the plot

# pdf(file="RCircos_PDX636.pdf", height=8, width=8, compress=TRUE)
# pdf(file="Rcircos_PDX636.pdf")
png(file="RCircos_PDX636.png", res = 600, width = 4800, height = 4800)
RCircos.Set.Plot.Area()

par(mai=c(0.25, 0.25, 0.25, 0.25))
plot.new()
plot.window(c(-2.5, 2.5), c(-2.5, 2.5))

# scatter
mut_636 <- mutPrep("../CCF", "MAP636*")
mut_636_Diag <- mut_636[which(mut_636$PlotColor == "black"), ]
colnames(mut_636_Diag) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
# rownames(mut_636_Diag) <- paste(mut_636_Diag$Symbol, mut_636_Diag$chromStart)
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_636_Diag, data.col, side, inside.pos = 1.6, outside.pos = 1.65)

mut_636_Relapse <- mut_636[which(mut_636$PlotColor == "red"), ]
colnames(mut_636_Relapse) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
# rownames(mut_636_Relapse) <- paste(mut_636_Relapse$Symbol, mut_636_Relapse$chromStart)
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_636_Relapse, data.col, side, inside.pos = 1.65, outside.pos = 1.7)

# keep_mut <- c(rownames(mut_636_Diag),rownames(mut_636_Relapse))

mut_636_Ortho <- mut_636[which(mut_636$PlotColor == "blue"), ]
colnames(mut_636_Ortho) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_636_Ortho, data.col, side, inside.pos = 1.7, outside.pos = 1.75)

mut_636_Sc <- mut_636[which(mut_636$PlotColor == "green"), ]
colnames(mut_636_Sc) <- c("Chromosome", "chromStart", "chromEnd", "Data", "PlotColor","Symbol")
data.col <- 4
side <- "in"
RCircos.Scatter.Plot(mut_636_Sc, data.col, side, inside.pos = 1.75, outside.pos = 1.8)


# DIAG        
# Heatmap
cnv_fc_636 <- CNV_prep("../../facets/MAP636_Diag_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_636, data.col, side, inside.pos = 1.15, outside.pos = 1.25, max.value = 2, min.value = -2)

# META
# Heatmap
cnv_fc_636 <- CNV_prep("../../facets/MAP636_Relapse_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_636, data.col, side, inside.pos = 1.25, outside.pos = 1.35, max.value = 2, min.value = -2)

# ORTHO
# Heatmap
cnv_fc_636 <- CNV_prep("../../facets/MAP636_Ortho_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_636, data.col, side, inside.pos = 1.35, outside.pos = 1.45, max.value = 2, min.value = -2)

# SC
# Heatmap
cnv_fc_636 <- CNV_prep("../../facets/MAP636_Sc_cncf.txt")
data.col <- 5
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(cnv_fc_636, data.col, side, inside.pos = 1.45, outside.pos = 1.55, max.value = 2, min.value = -2)

# DIAG
# Histogram
rna_diag <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP636_Diag")]
rna_diag$PlotColor <- "black"
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_diag, data.col, side, inside.pos = 0.9, outside.pos = 0.95, min.value=min(rna_diag$MAP636_Diag), max.value=max(rna_diag$MAP636_Diag))

# META
# Histogram
rna_meta <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP636_Relapse")]
rna_meta$PlotColor <- "red"
rna_meta$MAP636_Relapse <- ((rna_meta$MAP636_Relapse - rna_diag$MAP636_Diag) / rna_diag$MAP636_Diag) ^ 1/3
rna_meta <- rna_meta[complete.cases(rna_meta),]
rna_meta <- rna_meta[is.finite(rna_meta$MAP636_Relapse),]
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_meta, data.col, side, inside.pos = 0.95, outside.pos = 1, min.value=min(rna_meta$MAP636_Relapse), max.value=max(rna_meta$MAP636_Relapse))

# ORTHO
# Histogram
rna_ortho <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP636_Ortho")]
rna_ortho$PlotColor <- "blue"
rna_ortho$MAP636_Ortho <- ((rna_ortho$MAP636_Ortho - rna_diag$MAP636_Diag) / rna_diag$MAP636_Diag) ^ 1/3
rna_ortho <- rna_ortho[complete.cases(rna_ortho),]
rna_ortho <- rna_ortho[is.finite(rna_ortho$MAP636_Ortho),]
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_ortho, data.col, side, inside.pos = 1, outside.pos = 1.05, min.value=min(rna_ortho$MAP636_Ortho), max.value=max(rna_ortho$MAP636_Ortho))

# SC
# Histogram
rna_sc <- rna_table[,c("Chromosome", "chromStart", "chromEnd", "GeneName", "MAP636_Sc")]
rna_sc$PlotColor <- "green"
rna_sc$MAP636_Sc <- ((rna_sc$MAP636_Sc - rna_diag$MAP636_Diag) / rna_diag$MAP636_Diag) ^ 1/3
rna_sc <- rna_sc[complete.cases(rna_sc),]
rna_sc <- rna_sc[is.finite(rna_sc$MAP636_Sc),]
data.col <- 5
side <- "in"
RCircos.Line.Plot(rna_sc, data.col, side, inside.pos = 1.05, outside.pos = 1.1, min.value=min(rna_sc$MAP636_Sc), max.value=max(rna_sc$MAP636_Sc))


# Fusions data 636
data(RCircos.Link.Data)
fusions_636 <- fusionsPrep("../Fusions/data_Fusion_Inspector", "MAP636*")
fusions_636 <- fusions_636[which(fusions_636$PlotColor != "blue"),]
fusions_636 <- fusions_636[!(fusions_636$Genes %in% c("SDHAF2--PPP1R32", "COL1A1--COL3A1", "COL1A1--COL5A1")),]
fusions_636_plot <- fusions_636[c("Chromosome", "chromStart","chromEnd", "Chromosome.1", "chromStart.1", "chromEnd.1", "PlotColor")]
track.num <- 13
RCircos.Link.Plot(fusions_636_plot, track.num)


RCircos.Chromosome.Ideogram.Plot()

fus_gene_list <- fusions_636[which(fusions_636$PlotColor == "green" | fusions_636$PlotColor == "red"), ]
fus_gene_list <- separate(data = fus_gene_list, col = "Genes", into = c("Symbol", "Symbol.1"), sep = "--")
left_gene_list <- fus_gene_list[c("Chromosome", "chromStart", "chromEnd", "Symbol")]
right_gene_list <- fus_gene_list[c("Chromosome.1", "chromStart.1", "chromEnd.1", "Symbol.1")]
colnames(right_gene_list) <- c("Chromosome", "chromStart", "chromEnd", "Symbol")
fus_gene_list <- bind_rows(left_gene_list, right_gene_list)
fus_gene_list$PlotColor <- "blue"

mut_gene_list <- mut_636[,c("Chromosome", "chromStart", "chromEnd", "Symbol")]
mut_gene_list <-  ddply(mut_gene_list,.(Chromosome,chromStart,chromEnd,Symbol),nrow)
mut_gene_list <- mut_gene_list[which(mut_gene_list$V1 >= 3), ]
mut_gene_list$V1 <- NULL
mut_gene_list$PlotColor <- "red"

New_gene_list <- bind_rows(fus_gene_list, mut_gene_list)
#New_gene_list <- ddply(New_gene_list,.(Chromosome,chromStart,chromEnd,Symbol),nrow)
#New_gene_list$PlotColor <- "black"
#New_gene_list[which(New_gene_list$V1 == 2), "PlotColor"] <- "red"

data(RCircos.Gene.Label.Data)
name.col <- 4
side <- "in"
track.num <- 1
list_connector <- New_gene_list
RCircos.Gene.Connector.Plot(list_connector, track.num, side)

track.num <- 2
New_gene_list$PlotColor <- "black"
RCircos.Gene.Name.Plot(New_gene_list, name.col, track.num, side)

dev.off()
```

```{r cnv_genes}
library(biomaRt)
mart <- useMart(biomart="ENSEMBL_MART_ENSEMBL", host="grch37.ensembl.org", path="/biomart/martservice", dataset="hsapiens_gene_ensembl")
attributes <- c("hgnc_symbol")
filters <- c("chromosome_name","start","end")
values <- list(chromosome="1",start="154062041",end="154238920")
all.genes <- getBM(attributes=attributes, filters=filters, values=values, mart=mart, uniqueRows = T)

CGC_ref <- read.table("../reference_data/Census_allFri_Mar_19_02_02_52_2021.tsv", sep = "\t", header = T, fill = T)

genes_cgc <- all.genes[which(all.genes$hgnc_symbol %in% CGC_ref$Gene.Symbol),]
```

##############
Mutations
##############



```{r mutations_stats_data_prep_177}
library(ggplot2)

# MAP177
mut_177_Diag[, c("PlotColor", "Data")] <- NULL
mut_177_Diag[, "Patient"] <- "B77"
mut_177_Diag[, "Time"] <- "Diag"

mut_177_Relapse[, c("PlotColor", "Data")] <- NULL
mut_177_Relapse[, "Patient"] <- "B77"
mut_177_Relapse[, "Time"] <- "Relapse"

mut_177_Ortho[, c("PlotColor", "Data")] <- NULL
mut_177_Ortho[, "Patient"] <- "B77"
mut_177_Ortho[, "Time"] <- "Ortho"

mut_177_Sc[, c("PlotColor", "Data")] <- NULL
mut_177_Sc[, "Patient"] <- "B77"
mut_177_Sc[, "Time"] <- "Sc"

count_mut_177 <- bind_rows(mut_177_Diag, mut_177_Relapse, mut_177_Ortho, mut_177_Sc)

write.csv(count_mut_177, file = "Table_mutations_B77.tsv", sep = "\t", row.names = FALSE)

count_diag_177 <- ddply(count_mut_177,.(Patient,Chromosome, chromStart, chromEnd, Symbol),nrow)
count_diag_177 <- count_diag_177[which(count_diag_177$V1 == 4),]
count_diag_177$V1 <- NULL
write.csv(count_diag_177, file = "Table_mutations_conserved_from_diag_B77.tsv", sep = "\t", row.names = FALSE)

count_relapse_177 <- count_mut_177[which(count_mut_177$Time != "Diag"),]
count_relapse_177 <- ddply(count_relapse_177,.(Patient,Chromosome, chromStart, chromEnd, Symbol),nrow)
count_relapse_177 <- count_relapse_177[which(count_relapse_177$V1 == 3),]
count_relapse_177$V1 <- NULL
write.csv(count_relapse_177, file = "Table_mutations_conserved_from_Relapse_B77.tsv", sep = "\t", row.names = FALSE)

```


```{r mutations_stats_data_prep_194}

# MAP194
mut_194_Diag[, c("PlotColor", "Data")] <- NULL
mut_194_Diag[, "Patient"] <- "B94"
mut_194_Diag[, "Time"] <- "Diag"

mut_194_Relapse[, c("PlotColor", "Data")] <- NULL
mut_194_Relapse[, "Patient"] <- "B94"
mut_194_Relapse[, "Time"] <- "Relapse"

mut_194_Ortho[, c("PlotColor", "Data")] <- NULL
mut_194_Ortho[, "Patient"] <- "B94"
mut_194_Ortho[, "Time"] <- "Ortho"

mut_194_Sc[, c("PlotColor", "Data")] <- NULL
mut_194_Sc[, "Patient"] <- "B94"
mut_194_Sc[, "Time"] <- "Sc"

count_mut_194 <- bind_rows(mut_194_Diag, mut_194_Relapse, mut_194_Ortho, mut_194_Sc)

write.csv(count_mut_194, file = "Table_mutations_B94.tsv", sep = "\t", row.names = FALSE)

count_diag_194 <- ddply(count_mut_194,.(Patient,Chromosome, chromStart, chromEnd, Symbol),nrow)
count_diag_194 <- count_diag_194[which(count_diag_194$V1 == 4),]
count_diag_194$V1 <- NULL
write.csv(count_diag_194, file = "Table_mutations_conserved_from_diag_B94.tsv", sep = "\t", row.names = FALSE)

count_relapse_194 <- count_mut_194[which(count_mut_194$Time != "Diag"),]
count_relapse_194 <- ddply(count_relapse_194,.(Patient,Chromosome, chromStart, chromEnd, Symbol),nrow)
count_relapse_194 <- count_relapse_194[which(count_relapse_194$V1 == 3),]
count_relapse_194$V1 <- NULL
write.csv(count_relapse_194, file = "Table_mutations_conserved_from_Relapse_B94.tsv", sep = "\t", row.names = FALSE)

```

```{r mutations_stats_data_prep_217}

# MAP217
mut_217_Diag[, c("PlotColor", "Data")] <- NULL
mut_217_Diag[, "Patient"] <- "C17"
mut_217_Diag[, "Time"] <- "Diag"

mut_217_Relapse[, c("PlotColor", "Data")] <- NULL
mut_217_Relapse[, "Patient"] <- "C17"
mut_217_Relapse[, "Time"] <- "Relapse"

mut_217_Ortho[, c("PlotColor", "Data")] <- NULL
mut_217_Ortho[, "Patient"] <- "C17"
mut_217_Ortho[, "Time"] <- "Ortho"

mut_217_Sc[, c("PlotColor", "Data")] <- NULL
mut_217_Sc[, "Patient"] <- "C17"
mut_217_Sc[, "Time"] <- "Sc"

count_mut_217 <- bind_rows(mut_217_Diag, mut_217_Relapse, mut_217_Ortho, mut_217_Sc)

write.csv(count_mut_217, file = "Table_mutations_C17.tsv", sep = "\t", row.names = FALSE)

count_diag_217 <- ddply(count_mut_217,.(Patient,Chromosome, chromStart, chromEnd, Symbol),nrow)
count_diag_217 <- count_diag_217[which(count_diag_217$V1 == 4),]
count_diag_217$V1 <- NULL
write.csv(count_diag_217, file = "Table_mutations_conserved_from_diag_C17.tsv", sep = "\t", row.names = FALSE)

count_relapse_217 <- count_mut_217[which(count_mut_217$Time != "Diag"),]
count_relapse_217 <- ddply(count_relapse_217,.(Patient,Chromosome, chromStart, chromEnd, Symbol),nrow)
count_relapse_217 <- count_relapse_217[which(count_relapse_217$V1 == 3),]
count_relapse_217$V1 <- NULL
write.csv(count_relapse_217, file = "Table_mutations_conserved_from_Relapse_C17.tsv", sep = "\t", row.names = FALSE)

```


```{r mutations_stats_data_prep_342}
# MAP342
mut_342_Diag[, c("PlotColor", "Data")] <- NULL
mut_342_Diag[, "Patient"] <- "D42"
mut_342_Diag[, "Time"] <- "Diag"

mut_342_Relapse[, c("PlotColor", "Data")] <- NULL
mut_342_Relapse[, "Patient"] <- "D42"
mut_342_Relapse[, "Time"] <- "Relapse"

mut_342_Ortho[, c("PlotColor", "Data")] <- NULL
mut_342_Ortho[, "Patient"] <- "D42"
mut_342_Ortho[, "Time"] <- "Ortho"

mut_342_Sc[, c("PlotColor", "Data")] <- NULL
mut_342_Sc[, "Patient"] <- "D42"
mut_342_Sc[, "Time"] <- "Sc"

count_mut_342 <- bind_rows(mut_342_Diag, mut_342_Relapse, mut_342_Ortho, mut_342_Sc)

write.csv(count_mut_342, file = "Table_mutations_D42.tsv", sep = "\t", row.names = FALSE)

count_diag_342 <- ddply(count_mut_342,.(Patient,Chromosome, chromStart, chromEnd, Symbol),nrow)
count_diag_342 <- count_diag_342[which(count_diag_342$V1 == 4),]
count_diag_342$V1 <- NULL
write.csv(count_diag_342, file = "Table_mutations_conserved_from_diag_D42.tsv", sep = "\t", row.names = FALSE)

count_relapse_342 <- count_mut_342[which(count_mut_342$Time != "Diag"),]
count_relapse_342 <- ddply(count_relapse_342,.(Patient,Chromosome, chromStart, chromEnd, Symbol),nrow)
count_relapse_342 <- count_relapse_342[which(count_relapse_342$V1 == 3),]
count_relapse_342$V1 <- NULL
write.csv(count_relapse_342, file = "Table_mutations_conserved_from_Relapse_D42.tsv", sep = "\t", row.names = FALSE)

```

```{r mutations_stats_data_prep_469}

# MAP469
mut_469_Diag[, c("PlotColor", "Data")] <- NULL
mut_469_Diag[, "Patient"] <- "E69"
mut_469_Diag[, "Time"] <- "Diag"

mut_469_Relapse[, c("PlotColor", "Data")] <- NULL
mut_469_Relapse[, "Patient"] <- "E69"
mut_469_Relapse[, "Time"] <- "Relapse"

mut_469_Ortho[, c("PlotColor", "Data")] <- NULL
mut_469_Ortho[, "Patient"] <- "E69"
mut_469_Ortho[, "Time"] <- "Ortho"

mut_469_Sc[, c("PlotColor", "Data")] <- NULL
mut_469_Sc[, "Patient"] <- "E69"
mut_469_Sc[, "Time"] <- "Sc"

count_mut_469 <- bind_rows(mut_469_Diag, mut_469_Relapse, mut_469_Ortho, mut_469_Sc)

write.csv(count_mut_469, file = "Table_mutations_E69.tsv", sep = "\t", row.names = FALSE)

count_diag_469 <- ddply(count_mut_469,.(Patient,Chromosome, chromStart, chromEnd, Symbol),nrow)
count_diag_469 <- count_diag_469[which(count_diag_469$V1 == 4),]
count_diag_469$V1 <- NULL
write.csv(count_diag_469, file = "Table_mutations_conserved_from_diag_E69.tsv", sep = "\t", row.names = FALSE)

count_relapse_469 <- count_mut_469[which(count_mut_469$Time != "Diag"),]
count_relapse_469 <- ddply(count_relapse_469,.(Patient,Chromosome, chromStart, chromEnd, Symbol),nrow)
count_relapse_469 <- count_relapse_469[which(count_relapse_469$V1 == 3),]
count_relapse_469$V1 <- NULL
write.csv(count_relapse_469, file = "Table_mutations_conserved_from_Relapse_E69.tsv", sep = "\t", row.names = FALSE)

```

```{r mutations_stats_data_prep_559}

# MAP559
mut_559_Relapse[, c("PlotColor", "Data")] <- NULL
mut_559_Relapse[, "Patient"] <- "F59"
mut_559_Relapse[, "Time"] <- "Relapse"

mut_559_Ortho[, c("PlotColor", "Data")] <- NULL
mut_559_Ortho[, "Patient"] <- "F59"
mut_559_Ortho[, "Time"] <- "Ortho"

mut_559_Sc[, c("PlotColor", "Data")] <- NULL
mut_559_Sc[, "Patient"] <- "F59"
mut_559_Sc[, "Time"] <- "Sc"

count_mut_559 <- bind_rows(mut_559_Relapse, mut_559_Ortho, mut_559_Sc)

write.csv(count_mut_559, file = "Table_mutations_F59.tsv", sep = "\t", row.names = FALSE)

count_relapse_559 <- count_mut_559[which(count_mut_559$Time != "Diag"),]
count_relapse_559 <- ddply(count_relapse_559,.(Patient,Chromosome, chromStart, chromEnd, Symbol),nrow)
count_relapse_559 <- count_relapse_559[which(count_relapse_559$V1 == 3),]
count_relapse_559$V1 <- NULL
write.csv(count_relapse_559, file = "Table_mutations_conserved_from_Relapse_F59.tsv", sep = "\t", row.names = FALSE)

```

```{r mutations_stats_data_prep_636}

# MAP636
mut_636_Diag[, c("PlotColor", "Data")] <- NULL
mut_636_Diag[, "Patient"] <- "G36"
mut_636_Diag[, "Time"] <- "Diag"

mut_636_Relapse[, c("PlotColor", "Data")] <- NULL
mut_636_Relapse[, "Patient"] <- "G36"
mut_636_Relapse[, "Time"] <- "Relapse"

mut_636_Ortho[, c("PlotColor", "Data")] <- NULL
mut_636_Ortho[, "Patient"] <- "G36"
mut_636_Ortho[, "Time"] <- "Ortho"

mut_636_Sc[, c("PlotColor", "Data")] <- NULL
mut_636_Sc[, "Patient"] <- "G36"
mut_636_Sc[, "Time"] <- "Sc"

count_mut_636 <- bind_rows(mut_636_Diag, mut_636_Relapse, mut_636_Ortho, mut_636_Sc)

write.csv(count_mut_636, file = "Table_mutations_G36.tsv", sep = "\t", row.names = FALSE)

count_diag_636 <- ddply(count_mut_636,.(Patient,Chromosome, chromStart, chromEnd, Symbol),nrow)
count_diag_636 <- count_diag_636[which(count_diag_636$V1 == 4),]
count_diag_636$V1 <- NULL
write.csv(count_diag_636, file = "Table_mutations_conserved_from_diag_G36.tsv", sep = "\t", row.names = FALSE)

count_relapse_636 <- count_mut_636[which(count_mut_636$Time != "Diag"),]
count_relapse_636 <- ddply(count_relapse_636,.(Patient,Chromosome, chromStart, chromEnd, Symbol),nrow)
count_relapse_636 <- count_relapse_636[which(count_relapse_636$V1 == 3),]
count_relapse_636$V1 <- NULL
write.csv(count_relapse_636, file = "Table_mutations_conserved_from_Relapse_G36.tsv", sep = "\t", row.names = FALSE)

```

```{r mutations_stats_data_prep_all_samples}

count_mut_all <- bind_rows(count_mut_177, count_mut_194, count_mut_217, count_mut_342, count_mut_469, count_mut_559, count_mut_636)
write.csv(count_mut_all, file = "Table_mutations_all_samples.tsv", sep = "\t", row.names = FALSE)

count_diag_all <- bind_rows(count_diag_177, count_diag_194, count_diag_217, count_diag_342, count_diag_469, count_diag_636)
write.csv(count_diag_all, file = "Table_mutations_conserved_from_diag_all_samples.tsv", sep = "\t", row.names = FALSE)

count_Relapse_all <- bind_rows(count_relapse_177, count_relapse_194, count_relapse_217, count_relapse_342, count_relapse_469, count_relapse_559, count_relapse_636)
write.csv(count_Relapse_all, file = "Table_mutations_conserved_from_relapse_all_samples.tsv", sep = "\t", row.names = FALSE)
```

```{r function_plot_enrichr}
plot_enrichr <- function(enrichr_object, dbs, file_name){
  pdf(file = file_name)
for (i in 1:length(dbs)){
  print(i)
  table_to_plot <- enrichr_object[[i]]
  table_to_plot <- table_to_plot[which(table_to_plot$Adjusted.P.value <= 0.05),]
  if (nrow(table_to_plot) >= 1) {
    unique_plot <- plotEnrich(table_to_plot, showTerms = 20, numChar = 50, y = "Count", orderBy = "Adjusted.P.value", title = dbs[i])
    unique_plot + theme(axis.text=element_text(size=12), axis.title=element_text(size=14,face="bold"))
    print(unique_plot)
  }
}
dev.off()
return(table_to_plot)
}

plot_enrichr_no_file <- function(enrichr_object, dbs){
for (i in 1:length(dbs)){
  print(i)
  table_to_plot <- enrichr_object[[i]]
  table_to_plot <- table_to_plot[which(table_to_plot$Adjusted.P.value <= 0.05),]
  if (nrow(table_to_plot) >= 1) {
    unique_plot <- plotEnrich(table_to_plot, showTerms = 20, numChar = 50, y = "Count", orderBy = "Adjusted.P.value", title = dbs[i])
    unique_plot + theme(axis.text=element_text(size=12), axis.title=element_text(size=14,face="bold"))
    print(unique_plot)
  }
}
return(table_to_plot)
}
```

```{r functionnal_enrichment_CGC_diag}
library(enrichR)
library("ggpubr")
library(JLutils)

dbs_all <- listEnrichrDbs()
dbs_all <- as.vector(dbs_all$libraryName)

dbs_all <- c("Reactome_2013", "WikiPathways_2013", "Disease_Signatures_from_GEO_up_2014", "KEGG_2013", "GO_Molecular_Function_2021", "GO_Cellular_Component_2021","GO_Biological_Process_2021" ,"Chromosome_Location", "Human_Gene_Atlas")

dbs_selection <- c("BioCarta_2013", "Reactome_2013", "WikiPathways_2013", "Disease_Signatures_from_GEO_up_2014", "KEGG_2013", "GO_Molecular_Function_2021", "GO_Cellular_Component_2021","GO_Biological_Process_2021" ,"Chromosome_Location", "Human_Gene_Atlas")


CGC_ref <- read.table("../reference_data/Census_allFri_Mar_19_02_02_52_2021.tsv", sep = "\t", header = T, fill = T)

table_enrich_diag <- CGC_ref[which(CGC_ref$Gene.Symbol %in% count_diag_all$Symbol),]
table_enrich_diag <- ddply(table_enrich_diag,.(HALLMARK),nrow)
table_enrich_diag <- table_enrich_diag[order(table_enrich_diag$V1),]

pdf(file = "barplot_functional_enrichment_genes_conserved_from_diag_all_samples_CGC.pdf")
ggplot(table_enrich_diag, aes(x=reorder(HALLMARK, V1), y=V1)) + geom_bar(stat="identity", fill="steelblue") + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + labs(title = "Count of CGC Hallmark in diagnostic conserved genes", x = "Hallmark", y = "Count")
dev.off()

table_enrichr_diag <- CGC_ref[which(CGC_ref$Gene.Symbol %in% count_diag_all$Symbol),]
list_enrichr_diag <- as.vector(unique(table_enrichr_diag$Gene.Symbol))
enrichment_diag <- enrichr(list_enrichr_diag, dbs_all)
enrichment_diag_select_dbs <- enrichr(list_enrichr_diag, dbs_selection)

plot_enrichr(enrichment_diag, dbs_all, "plot_enrichment_CGC_mut_diagnostic_conserved.pdf")
plot_enrichr(enrichment_diag_select_dbs, dbs_selection, "plot_enrichment_CGC_mut_diagnostic_conserved_selected_dbs.pdf")
```

```{r functionnal_enrichment_diagnostic_conserved_mutations}

table_enrichr_diag_all <- count_diag_all$Symbol
list_enrichr_diag_all <- as.vector(unique(table_enrichr_diag_all))
enrichment_diag_all <- enrichr(list_enrichr_diag_all, dbs_all)
enrichment_diag_all_select_dbs <- enrichr(list_enrichr_diag_all, dbs_selection)

plot_enrichr(enrichment_diag_all, dbs_all, "plot_enrichment_mut_all_diagnostic_conserved.pdf")
plot_enrichr(enrichment_diag_all_select_dbs, dbs_selection, "plot_enrichment_mut_all_diagnostic_conserved_selected_dbs.pdf")
```

```{r functionnal_enrichment_CGC_relapse}

table_enrich_relapse <- CGC_ref[which(CGC_ref$Gene.Symbol %in% count_Relapse_all$Symbol),]
table_enrich_relapse <- ddply(table_enrich_relapse,.(HALLMARK),nrow)
table_enrich_relapse <- table_enrich_relapse[order(table_enrich_relapse$V1),]

pdf(file = "barplot_functional_enrichment_genes_conserved_from_relapse_all_samples_CGC.pdf")
ggplot(table_enrich_relapse, aes(x=reorder(HALLMARK, V1), y=V1)) + geom_bar(stat="identity", fill="steelblue") + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + labs(title = "Count of CGC Hallmark in relapse conserved genes", x = "Hallmark", y = "Count")
dev.off()


table_enrichr_relapse <- count_Relapse_all[which(count_Relapse_all$Symbol %in% unique(CGC_ref$Gene.Symbol)),]
list_enrichr_relapse <- as.vector(unique(table_enrichr_relapse$Symbol))
enrichment_relapse <- enrichr(list_enrichr_relapse, dbs_all)
enrichment_relapse_select_dbs <- enrichr(list_enrichr_relapse, dbs_selection)

plot_enrichr(enrichment_relapse, dbs_all, "plot_enrichment_CGC_mut_relapse_conserved.pdf")
plot_enrichr(enrichment_relapse_select_dbs, dbs_selection, "plot_enrichment_CGC_mut_relapse_conserved_selected_dbs.pdf")

```

```{r functionnal_enrichment_relapse_conserved_mutations}

table_enrichr_relapse_all <- count_Relapse_all$Symbol
list_enrichr_relapse_all <- as.vector(unique(table_enrichr_relapse_all))
enrichment_relapse_all <- enrichr(list_enrichr_relapse_all, dbs_all)
enrichment_relapse_all_select_dbs <- enrichr(list_enrichr_relapse_all, dbs_selection)

plot_enrichr(enrichment_relapse_all, dbs_all, "plot_enrichment_mut_all_relapse_conserved.pdf")
plot_enrichr(enrichment_relapse_all_select_dbs, dbs_selection, "plot_enrichment_mut_all_relapse_conserved_selected_dbs.pdf")
```

```{r functionnal_enrichment_CGC_all_mutations}

table_enrich_all <- CGC_ref[which(CGC_ref$Gene.Symbol %in% count_mut_all$Symbol),]
table_enrich_all <- ddply(table_enrich_all,.(HALLMARK),nrow)
table_enrich_all <- table_enrich_relapse[order(table_enrich_all$V1),]
table_enrich_all <- table_enrich_all[complete.cases(table_enrich_all),]

pdf(file = "barplot_functional_enrichment_all_mutations_all_samples_CGC.pdf")
ggplot(table_enrich_all, aes(x=reorder(HALLMARK, V1), y=V1)) + geom_bar(stat="identity", fill="steelblue") + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + labs(title = "Count of CGC Hallmark in all mutations", x = "Hallmark", y = "Count")
dev.off()

table_enrichr <- CGC_ref[which(CGC_ref$Gene.Symbol %in% count_mut_all$Symbol),]
list_enrichr <- as.vector(unique(table_enrichr$Gene.Symbol))
enrichment <- enrichr(list_enrichr, dbs_all)
enrichment_select_dbs <- enrichr(list_enrichr, dbs_selection)

plot_enrichr(enrichment, dbs_all, "plot_enrichment_CGC_mut_all.pdf")
plot_enrichr(enrichment_select_dbs, dbs_selection, "plot_enrichment_CGC_mut_all_selected_dbs.pdf")
```

```{r functionnal_enrichment_all_mutations}

table_enrichr <- count_mut_all$Symbol
list_enrichr <- as.vector(unique(table_enrichr))
enrichment <- enrichr(list_enrichr, dbs_all)
enrichment_select_dbs <- enrichr(list_enrichr, dbs_selection)

plot_enrichr(enrichment, dbs_all, "plot_enrichment_mut_all.pdf")
plot_enrichr(enrichment_select_dbs, dbs_selection, "plot_enrichment_mut_all_selected_dbs.pdf")
```

###############
FUSIONS
###############


```{r fusions_data_prep_function}
fusionsTablePrep <- function(file) { 
  
  data_all <- read.table(file, header = FALSE)
  colnames(data_all) <- "full_name"
  data_all <- separate(data = data_all, col = "full_name", into = c("Genes", "positions"), sep = ":")
  data_all <- separate(data = data_all, col = "Genes", into = c("Gene1", "Gene2"), sep = "--")
  data_all <- separate(data = data_all, col = "positions", into = c("Position1", "Position2"), sep = "\\|")
  data_all <- separate(data = data_all, col = "Position1", into = c("Chromosome", "chromStart"), sep = "-")
  data_all$chromStart <- as.numeric(data_all$chromStart)
  data_all$chromEnd <- data_all$chromStart
  data_all <- separate(data = data_all, col = "Position2", into = c("Chromosome.1", "chromStart.1"), sep = "-")
  data_all$chromStart.1 <- as.numeric(data_all$chromStart.1)
  data_all$chromEnd.1 <- data_all$chromStart.1
  
  return(data_all)
}
```

```{r fusions_data_prep_177}
fus_B77_Diag <- fusionsTablePrep("../Fusions/data_Fusion_Inspector/MAP177_Diag_Oncoprint_Fusions.tsv")
fus_B77_Diag$Patient <- "B77"
fus_B77_Diag$Time <- "Diag"

fus_B77_Relapse <- fusionsTablePrep("../Fusions/data_Fusion_Inspector/MAP177_Relapse_Oncoprint_Fusions.tsv")
fus_B77_Relapse$Patient <- "B77"
fus_B77_Relapse$Time <- "Relapse"

fus_B77_Ortho <- fusionsTablePrep("../Fusions/data_Fusion_Inspector/MAP177_Ortho_Oncoprint_Fusions.tsv")
fus_B77_Ortho$Patient <- "B77"
fus_B77_Ortho$Time <- "Ortho"

fus_B77_Sc <- fusionsTablePrep("../Fusions/data_Fusion_Inspector/MAP177_Sc_Oncoprint_Fusions.tsv")
fus_B77_Sc$Patient <- "B77"
fus_B77_Sc$Time <- "Sc"

fus_B77 <- bind_rows(fus_B77_Diag, fus_B77_Relapse, fus_B77_Ortho, fus_B77_Sc)
```

```{r fusions_data_prep_194}
fus_B94_Diag <- fusionsTablePrep("../Fusions/data_Fusion_Inspector/MAP194_Diag_Oncoprint_Fusions.tsv")
fus_B94_Diag$Patient <- "B94"
fus_B94_Diag$Time <- "Diag"

fus_B94_Relapse <- fusionsTablePrep("../Fusions/data_Fusion_Inspector/MAP194_Relapse_Oncoprint_Fusions.tsv")
fus_B94_Relapse$Patient <- "B94"
fus_B94_Relapse$Time <- "Relapse"

fus_B94_Ortho <- fusionsTablePrep("../Fusions/data_Fusion_Inspector/MAP194_Ortho_Oncoprint_Fusions.tsv")
fus_B94_Ortho$Patient <- "B94"
fus_B94_Ortho$Time <- "Ortho"

fus_B94_Sc <- fusionsTablePrep("../Fusions/data_Fusion_Inspector/MAP194_Sc_Oncoprint_Fusions.tsv")
fus_B94_Sc$Patient <- "B94"
fus_B94_Sc$Time <- "Sc"

fus_B94 <- bind_rows(fus_B94_Diag, fus_B94_Relapse, fus_B94_Ortho, fus_B94_Sc)
```

```{r fusions_data_prep_217}
fus_C17_Diag <- fusionsTablePrep("../Fusions/data_Fusion_Inspector/MAP217_Diag_Oncoprint_Fusions.tsv")
fus_C17_Diag$Patient <- "C17"
fus_C17_Diag$Time <- "Diag"

fus_C17_Relapse <- fusionsTablePrep("../Fusions/data_Fusion_Inspector/MAP217_Relapse_Oncoprint_Fusions.tsv")
fus_C17_Relapse$Patient <- "C17"
fus_C17_Relapse$Time <- "Relapse"

fus_C17_Ortho <- fusionsTablePrep("../Fusions/data_Fusion_Inspector/MAP217_Ortho_Oncoprint_Fusions.tsv")
fus_C17_Ortho$Patient <- "C17"
fus_C17_Ortho$Time <- "Ortho"

fus_C17_Sc <- fusionsTablePrep("../Fusions/data_Fusion_Inspector/MAP217_Sc_Oncoprint_Fusions.tsv")
fus_C17_Sc$Patient <- "C17"
fus_C17_Sc$Time <- "Sc"

fus_C17 <- bind_rows(fus_C17_Diag, fus_C17_Relapse, fus_C17_Ortho, fus_C17_Sc)
```

```{r fusions_data_prep_342}
fus_D42_Diag <- fusionsTablePrep("../Fusions/data_Fusion_Inspector/MAP342_Diag_Oncoprint_Fusions.tsv")
fus_D42_Diag$Patient <- "D42"
fus_D42_Diag$Time <- "Diag"

fus_D42_Relapse <- fusionsTablePrep("../Fusions/data_Fusion_Inspector/MAP342_Relapse_Oncoprint_Fusions.tsv")
fus_D42_Relapse$Patient <- "D42"
fus_D42_Relapse$Time <- "Relapse"

fus_D42_Ortho <- fusionsTablePrep("../Fusions/data_Fusion_Inspector/MAP342_Ortho_Oncoprint_Fusions.tsv")
fus_D42_Ortho$Patient <- "D42"
fus_D42_Ortho$Time <- "Ortho"

fus_D42_Sc <- fusionsTablePrep("../Fusions/data_Fusion_Inspector/MAP342_Ortho-meta_Oncoprint_Fusions.tsv")
fus_D42_Sc$Patient <- "D42"
fus_D42_Sc$Time <- "Sc"

fus_D42 <- bind_rows(fus_D42_Diag, fus_D42_Relapse, fus_D42_Ortho, fus_D42_Sc)
```

```{r fusions_data_prep_469}
fus_E69_Diag <- fusionsTablePrep("../Fusions/data_Fusion_Inspector/MAP469_Diag_Oncoprint_Fusions.tsv")
fus_E69_Diag$Patient <- "E69"
fus_E69_Diag$Time <- "Diag"

fus_E69_Relapse <- fusionsTablePrep("../Fusions/data_Fusion_Inspector/MAP469_Relapse_Oncoprint_Fusions.tsv")
fus_E69_Relapse$Patient <- "E69"
fus_E69_Relapse$Time <- "Relapse"

fus_E69_Ortho <- fusionsTablePrep("../Fusions/data_Fusion_Inspector/MAP469_Ortho_Oncoprint_Fusions.tsv")
fus_E69_Ortho$Patient <- "E69"
fus_E69_Ortho$Time <- "Ortho"

fus_E69_Sc <- fusionsTablePrep("../Fusions/data_Fusion_Inspector/MAP469_Sc_Oncoprint_Fusions.tsv")
fus_E69_Sc$Patient <- "E69"
fus_E69_Sc$Time <- "Sc"

fus_E69 <- bind_rows(fus_E69_Diag, fus_E69_Relapse, fus_E69_Ortho, fus_E69_Sc)
```

```{r fusions_data_prep_559}
fus_F59_Diag <- fusionsTablePrep("../Fusions/data_Fusion_Inspector/MAP559_Diag_Oncoprint_Fusions.tsv")
fus_F59_Diag$Patient <- "F59"
fus_F59_Diag$Time <- "Diag"

fus_F59_Relapse <- fusionsTablePrep("../Fusions/data_Fusion_Inspector/MAP559_Relapse_Oncoprint_Fusions.tsv")
fus_F59_Relapse$Patient <- "F59"
fus_F59_Relapse$Time <- "Relapse"

fus_F59_Ortho <- fusionsTablePrep("../Fusions/data_Fusion_Inspector/MAP559_Ortho_Oncoprint_Fusions.tsv")
fus_F59_Ortho$Patient <- "F59"
fus_F59_Ortho$Time <- "Ortho"

fus_F59 <- bind_rows(fus_F59_Diag, fus_F59_Relapse, fus_F59_Ortho)
```

```{r fusions_data_prep_636}
fus_G36_Diag <- fusionsTablePrep("../Fusions/data_Fusion_Inspector/MAP636_Diag_Oncoprint_Fusions.tsv")
fus_G36_Diag$Patient <- "G36"
fus_G36_Diag$Time <- "Diag"

fus_G36_Relapse <- fusionsTablePrep("../Fusions/data_Fusion_Inspector/MAP636_Relapse_Oncoprint_Fusions.tsv")
fus_G36_Relapse$Patient <- "G36"
fus_G36_Relapse$Time <- "Relapse"

fus_G36_Ortho <- fusionsTablePrep("../Fusions/data_Fusion_Inspector/MAP636_Ortho_Oncoprint_Fusions.tsv")
fus_G36_Ortho$Patient <- "G36"
fus_G36_Ortho$Time <- "Ortho"

fus_G36_Sc <- fusionsTablePrep("../Fusions/data_Fusion_Inspector/MAP636_Sc_Oncoprint_Fusions.tsv")
fus_G36_Sc$Patient <- "G36"
fus_G36_Sc$Time <- "Sc"

fus_G36 <- bind_rows(fus_G36_Diag, fus_G36_Relapse, fus_G36_Ortho, fus_G36_Sc)
```

```{r fusions_data_prep_all}
library("stringr")
fus_all <- bind_rows(fus_B77, fus_B94, fus_C17, fus_D42, fus_E69, fus_F59, fus_G36)
write.csv(fus_all, file = "Table_all_fusions.tsv", sep = "\t", row.names = FALSE)

# diag conserved
fus_diag_conserved <- ddply(fus_all,.(Gene1,Gene2,chromStart,chromStart.1,Patient),nrow)
fus_diag_conserved <- fus_diag_conserved[which(fus_diag_conserved$V1 == 4),]
list_fus_diag_conserved <- unique(c(fus_diag_conserved$Gene1, fus_diag_conserved$Gene2))
enrichment <- enrichr(list_fus_diag_conserved, dbs_all)
enrichment_select_dbs <- enrichr(list_fus_diag_conserved, dbs_selection)
plot_enrichr(enrichment, dbs_all, "plot_enrichment_fus_diag_conserved.pdf")
plot_enrichr(enrichment_select_dbs, dbs_selection, "plot_enrichment_fus_diag_conserved_selected_dbs.pdf")

# diag conserved CGC
list_fus_diag_cgc <- CGC_ref[which(CGC_ref$Gene.Symbol %in% list_fus_diag_conserved),]
enrichment <- enrichr(as.vector(list_fus_diag_cgc$Gene.Symbol), dbs_all)
enrichment_select_dbs <- enrichr(as.vector(list_fus_diag_cgc$Gene.Symbol), dbs_selection)
plot_enrichr(enrichment, dbs_all, "plot_enrichment_CGC_fus_diag_conserved.pdf")
plot_enrichr(enrichment_select_dbs, dbs_selection, "plot_enrichment_CGC_fus_diag_conserved_selected_dbs.pdf")

# relapse conserved
fus_relapse_conserved <- ddply(fus_all[which(fus_all$Time != "Diag"),],.(Gene1,Gene2,chromStart1,chromStart2,Patient),nrow)
fus_relapse_conserved <- fus_relapse_conserved[which(fus_relapse_conserved$V1 == 3),]
list_fus_relapse_conserved <- unique(c(fus_relapse_conserved$Gene1, fus_relapse_conserved$Gene2))
enrichment <- enrichr(list_fus_relapse_conserved, dbs_all)
enrichment_select_dbs <- enrichr(list_fus_relapse_conserved, dbs_selection)
plot_enrichr(enrichment, dbs_all, "plot_enrichment_fus_relapse_conserved.pdf")
plot_enrichr(enrichment_select_dbs, dbs_selection, "plot_enrichment_fus_relapse_conserved_selected_dbs.pdf")

# relapse conserved CGC
list_fus_relapse_cgc <- CGC_ref[which(CGC_ref$Gene.Symbol %in% list_fus_relapse_conserved),]
enrichment <- enrichr(as.vector(list_fus_relapse_cgc$Gene.Symbol), dbs_all)
enrichment_select_dbs <- enrichr(as.vector(list_fus_relapse_cgc$Gene.Symbol), dbs_selection)
plot_enrichr(enrichment, dbs_all, "plot_enrichment_CGC_fus_relapse_conserved.pdf")
plot_enrichr(enrichment_select_dbs, dbs_selection, "plot_enrichment_CGC_fus_relapse_conserved_selected_dbs.pdf")


# all fus
list_fus_all <- unique(c(fus_all$Gene1, fus_all$Gene2))
enrichment <- enrichr(list_fus_all, dbs_all)
enrichment_select_dbs <- enrichr(list_fus_all, dbs_selection)
plot_enrichr(enrichment, dbs_all, "plot_enrichment_fus_all.pdf")
plot_enrichr(enrichment_select_dbs, dbs_selection, "plot_enrichment_fus_all_selected_dbs.pdf")

# all fus CGC
list_fus_all_cgc <- CGC_ref[which(CGC_ref$Gene.Symbol %in% list_fus_all),]
enrichment <- enrichr(as.vector(list_fus_all_cgc$Gene.Symbol), dbs_all)
enrichment_select_dbs <- enrichr(as.vector(list_fus_all_cgc$Gene.Symbol), dbs_selection)
plot_enrichr(enrichment, dbs_all, "plot_enrichment_CGC_fus_all.pdf")
plot_enrichr(enrichment_select_dbs, dbs_selection, "plot_enrichment_CGC_fus_all_selected_dbs.pdf")

data_for_heatmap_matrix <- fus_all
data_for_heatmap_matrix$namefusion <- str_c(data_for_heatmap_matrix$Gene1, '--',data_for_heatmap_matrix$Gene2, ':',data_for_heatmap_matrix$chromStart1, '-', data_for_heatmap_matrix$chromStart2)
data_for_heatmap_matrix$sample <- str_c(data_for_heatmap_matrix$Patient, '_', data_for_heatmap_matrix$Time)
data_for_heatmap_matrix <- data_for_heatmap_matrix[,c("namefusion", "sample")]

matrix_fus_heatmap <- table(data_for_heatmap_matrix)
matrix_fus_heatmap <- as.table(matrix_fus_heatmap)

list_order <- c("B77_Diag", "B77_Relapse", "B77_Ortho", "B77_Sc","B94_Diag", "B94_Relapse", "B94_Ortho", "B94_Sc","C17_Diag", "C17_Relapse", "C17_Ortho", "C17_Sc","D42_Diag", "D42_Relapse", "D42_Ortho", "D42_Sc","E69_Diag", "E69_Relapse", "E69_Ortho", "E69_Sc", "F59_Diag", "F59_Relapse", "F59_Ortho","G36_Diag", "G36_Relapse", "G36_Ortho", "G36_Sc")

patient_annot <- c("B77", "B77", "B77", "B77","B94", "B94", "B94", "B94","C17", "C17", "C17", "C17","D42", "D42", "D42", "D42","E69", "E69", "E69", "E69", "F59", "F59", "F59","G36", "G36", "G36", "G36")

table_col_annot <- data.frame(list_order, patient_annot)
rownames(table_col_annot) <- table_col_annot$list_order
table_col_annot$list_order <- NULL

more_fusions <- ddply(fus_all,.(Gene1, Gene2, chromStart, chromStart.1),nrow)
more_fusions <- more_fusions[which(more_fusions$V1 >= 3),]
list_more_fusions <- str_c(more_fusions$Gene1, '--', more_fusions$Gene2, ':', more_fusions$chromStart1, '-', more_fusions$chromStart2)

# list_fusions <- unique(c(str_c(fus_diag_conserved$Gene1, '--', fus_diag_conserved$Gene2, ':', fus_diag_conserved$chromStart1, '-', fus_diag_conserved$chromStart2), str_c(fus_relapse_conserved$Gene1, '--', fus_relapse_conserved$Gene2, ':', fus_relapse_conserved$chromStart1, '-', fus_relapse_conserved$chromStart2)))

matrix_fus_heatmap <- matrix_fus_heatmap[list_more_fusions, list_order]
rownames(matrix_fus_heatmap) <- sapply(strsplit(rownames(matrix_fus_heatmap),":"), `[`, 1)

list_remove_fus <- c("NDUFV3--PKNOX1", "SLC43A3--PRG2", "SDHAF2--PPP1R32", "COL1A1--COL5A1", "COL1A1--COL3A1", "ZNF713--MRPS17")
matrix_fus_heatmap <- matrix_fus_heatmap[!rownames(matrix_fus_heatmap) %in% list_remove_fus,]

table_annot_fusions <- read.table("fusion_classification.csv", header = T, sep = "\t")
rownames(table_annot_fusions) <- gsub("..","--",table_annot_fusions$Fusion, fixed = T)
list_keep <- gsub("..","--",rownames(table_annot_fusions), fixed = T)
matrix_fus_heatmap <- matrix_fus_heatmap[which(rownames(matrix_fus_heatmap) %in% list_keep),]
#matrix_fus_heatmap <- as.data.frame(matrix_fus_heatmap)
table_annot_fusions$Fusion <- NULL
table_annot_fusions$pdf <- NULL

list_fusions <- rownames(matrix_fus_heatmap)

pheatmap::pheatmap(matrix_fus_heatmap, clustering_method = "ward.D", filename = "heatmap_Fusion_Inspector_samples.pdf", cellheight = 8, cluster_cols = F, annotation_row = table_annot_fusions, annotation_col = table_col_annot)
```


####################
Mutations + Fusions
####################

```{r Mutations_and_fusions_data_table}
table_mut <- count_mut_all
colnames(table_mut) <- c("Chromosome1", "chromStart1", "chromEnd1", "Gene1", "Patient", "Time")
table_mut$Alteration <- "Somatic_Mutation"

colnames(fus_all) <- c("Gene1", "Gene2", "Chromosome1", "chromStart1", "Chromosome2", "chromStart2", "chromEnd1", "chromEnd2", "Patient", "Time")
fus_all$Alteration <- "Fusion"

table_Fus_Mut <- bind_rows(fus_all, table_mut)

write.csv(table_Fus_Mut, file = "Table_all_fusions_and_mutations.tsv", sep = "\t", row.names = FALSE, quote = F)
```

####################
Expression
####################

```{r RNA_seq_analyse_diff_Human_Mouse}
library(DESeq2)
library(factoextra)
library(plyr)
library(tidyr)
library(enrichR)
library(viridis)
# using exp_table tpm no log
list_exp_order <- c("MAP177_Diag", "MAP177_Relapse", "MAP177_Ortho", "MAP177_Sc","MAP194_Diag", "MAP194_Relapse", "MAP194_Ortho", "MAP194_Sc","MAP204_Diag", "MAP204_Relapse", "MAP204_Ortho", "MAP204_Sc","MAP217_Diag", "MAP217_Relapse", "MAP217_Ortho", "MAP217_Sc","MAP342_Diag", "MAP342_Relapse", "MAP342_Ortho", "MAP342_Ortho.meta","MAP469_Diag", "MAP469_Relapse", "MAP469_Ortho", "MAP469_Sc", "MAP559_Diag", "MAP559_Relapse", "MAP559_Ortho", "MAP559_Sc", "MAP636_Diag", "MAP636_Relapse", "MAP636_Ortho", "MAP636_Sc")

list_exp_order_rename <- c("B77_Diag", "B77_Relapse", "B77_Ortho", "B77_Sc","B94_Diag", "B94_Relapse", "B94_Ortho", "B94_Sc","C04_Diag", "C04_Relapse", "C04_Ortho", "C04_Sc","C17_Diag", "C17_Relapse", "C17_Ortho", "C17_Sc","D42_Diag", "D42_Relapse", "D42_Ortho", "D42_Sc","E69_Diag", "E69_Relapse", "E69_Ortho", "E69_Sc", "F59_Diag", "F59_Relapse", "F59_Ortho", "F59_Sc","G36_Diag", "G36_Relapse", "G36_Ortho", "G36_Sc")

rownames(exp_table) <- exp_table$GeneName
exp_count <- exp_table[,list_exp_order]
colnames(exp_count) <- list_exp_order_rename

#rownames(rna_table) <- rna_table$GeneName
#exp_count <- rna_table[,list_exp_order]
#colnames(exp_count) <- list_exp_order_rename


####### data matrix
table_meta_data <- ddply(table_Fus_Mut, .(Patient, Time), nrow)
table_meta_data$V1 <- NULL
Patient <- c("C04", "C04", "C04", "C04")
Time <- c("Diag", "Relapse", "Ortho", "Sc")
add_204 <- data.frame(Patient, Time)
table_meta_data <- rbind(table_meta_data, add_204)

table_meta_data$Hum_Mou <- "Hum"
table_meta_data[which(table_meta_data$Time %in% c("Ortho", "Sc")), "Hum_Mou"] <- "Mou"
table_meta_data$sample <- str_c(table_meta_data$Patient, '_', table_meta_data$Time)
rownames(table_meta_data) <- table_meta_data$sample
table_meta_data <- table_meta_data[list_exp_order_rename,]


####### differential expression
ddsMat <- DESeqDataSetFromMatrix(countData = round(exp_count),
                                 colData = table_meta_data,
                                 design = ~ Hum_Mou)

keep <- rowSums(counts(ddsMat)) > 1
dds <- ddsMat[keep,]
vsd <- vst(dds, blind = FALSE)
matrix_vsd <- assay(vsd)

sampleDists <- dist(t(assay(vsd)))

sampleDistMatrix <- as.matrix( sampleDists )
pheatmap::pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         filename = "sample_distance_matrix.pdf")

pcaData <- plotPCA(vsd, intgroup = c("Hum_Mou"), returnData = TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))

dds <- DESeq(dds)
res <- results(dds, contrast = c("Hum_Mou", "Hum", "Mou"))


####volcano
alpha <- 0.05 # Threshold on the adjusted p-value
cols <- densCols(res$log2FoldChange, -log10(res$pvalue))

png("volcano_plot_human_mou.png")
plot(res$log2FoldChange, -log10(res$padj), col=cols, panel.first=grid(),
     main="Volcano plot", xlab="Effect size: log2(fold-change)", ylab="-log10(adjusted p-value)",
     pch=20, cex=0.6)
abline(v=0)
abline(v=c(-1.5,1.5), col="brown")
abline(h=-log10(alpha), col="brown")

dev.off()

res_select <- res[-which(abs(res$log2FoldChange) > 1.5 & res$padj < alpha),]
res_removed <- res[which(abs(res$log2FoldChange) > 1.5 & res$padj < alpha),]
res_removed_data <- as.data.frame(res_removed[,c("log2FoldChange", "padj", "baseMean")])
res_removed_data <- res_removed_data[order(abs(res_removed_data$log2FoldChange), decreasing = T),]
write.table(res_removed_data, file = "table_foldChange_removed_genes_expression.csv", sep = "\t", quote = F)

png("volcano_plot_human_mou_after_selection.png")
plot(res_select$log2FoldChange, -log10(res_select$padj), col=cols, panel.first=grid(),
     main="Volcano plot", xlab="Effect size: log2(fold-change)", ylab="-log10(adjusted p-value)",
     pch=20, cex=0.6)
abline(v=0)
abline(v=c(-1.5,1.5), col="brown")
abline(h=-log10(alpha), col="brown")
dev.off()

list_genes_keep <- rownames(res_select)

table_counts_removed <- matrix_vsd[which(!rownames(res) %in% list_genes_keep),]
table_counts_after_diff <- matrix_vsd[list_genes_keep,]

genes_removed <- rownames(table_counts_removed)

enrichment_removed <- enrichr(genes_removed, dbs_all)
plot_enrichr(enrichment_removed, dbs_all, "plot_enrichment_removed_genes_expression.pdf")

project_removed.pca <- prcomp(t(table_counts_removed))
project_after_diff.pca <- prcomp(t(table_counts_after_diff))

table_explained_variance_after_diff <- summary(project_after_diff.pca)
table_explained_variance_after_diff <- table_explained_variance_after_diff$importance
write.table(table_explained_variance_after_diff, file = "table_explained_variance_after_diff.csv", sep = "\t", quote = F, row.names = T)

table_explained_variance_removed <- summary(project_removed.pca)
table_explained_variance_removed <- table_explained_variance_removed$importance
write.table(table_explained_variance_removed, file = "table_explained_variance_removed.csv", sep = "\t", quote = F, row.names = T)


pdf(file = "explained_variance_after_diff_exp.pdf")
fviz_eig(project_after_diff.pca)
dev.off()

pdf(file = "explained_variance_removed_exp.pdf")
fviz_eig(project_removed.pca)
dev.off()

#umap without selection
library(umap)
umap_after_diff <- umap(project_after_diff.pca$x[,c(1,3,4,5,6,7)])
#umap_after_diff <- umap(project_after_diff.pca$x)
umap_plot_df <- data.frame(umap_after_diff$layout) %>%
  # Turn sample IDs stored as row names into a column
  tibble::rownames_to_column("sample") %>%
  # Add the metadata into this data frame; match by sample IDs
  dplyr::inner_join(table_meta_data, by = "sample")

umap_plot_df_pca <- umap_plot_df
library(ggforce)
cairo_pdf(file = "umap_all_genes_after_diff_exp.pdf")
g <- ggplot(
  umap_plot_df,
  aes(
    x = X1,
    y = X2,
    color = Patient,# label points with different colors for each `subgroup`
    shape = Time,
    size = 3
  )
) +
  geom_point() +
  geom_mark_ellipse(aes(x=X1, y=X2,color=Patient, group=Patient, size = 0.2), linetype = 2)

plot(g)
dev.off()


umap_removed <- umap(project_removed.pca$x[,1:10])

umap_removed_plot_df <- data.frame(umap_removed$layout) %>%
  # Turn sample IDs stored as row names into a column
  tibble::rownames_to_column("sample") %>%
  # Add the metadata into this data frame; match by sample IDs
  dplyr::inner_join(table_meta_data, by = "sample")

library(ggforce)
cairo_pdf(file = "umap_removed_after_diff_exp.pdf")
ggplot(
  umap_removed_plot_df,
  aes(
    x = X1,
    y = X2,
    color = Patient,# label points with different colors for each `subgroup`
    shape = Time,
    size = 3
  )
) +
  geom_point() +
  geom_mark_ellipse(aes(x=X1, y=X2, group=Hum_Mou, size = 0.2), linetype = 2)
  #stat_ellipse(aes(x=X1, y=X2,color=Patient, group=Patient),type = "euclid")
dev.off()

# heatmap correlation ACP
cor_acp_mat <- cor(t(project_after_diff.pca$x[,c(1,2,3,4,5,6,7)]))

pheatmap::pheatmap(cor_acp_mat, clustering_method = "ward.D", filename = "heatmap_ACP_correlation.pdf", cellheight = 8, clustering_distance_rows = amap::Dist(cor_acp_mat, method="euclidean"), clustering_distance_cols = amap::Dist(t(cor_acp_mat), method = "kendall"))

pheatmap::pheatmap(cor_acp_mat, clustering_method = "ward.D", filename = "heatmap_ACP_correlation_not_clustered.pdf", cellheight = 8, cluster_rows = F, cluster_cols = F, method = "kendall")

cor_acp_mat_removed <- cor(t(project_removed.pca$x))

pheatmap::pheatmap(cor_acp_mat_removed, clustering_method = "ward.D", filename = "heatmap_ACP_removed_genes_correlation.pdf", cellheight = 8, clustering_distance_rows = amap::Dist(cor_acp_mat_removed, method="euclidean"), clustering_distance_cols = amap::Dist(t(cor_acp_mat_removed), method = "kendall"))

# GSEA enrichment
library(dplyr)

# explaining why we remove the second component
table_components <- as.data.frame(abs(project_removed.pca$x[,"PC2"]))
colnames(table_components) <- "PC2"
table_components$sample <- rownames(table_components)
table_components <- separate(table_components, col = "sample", into = c("Patient", "Time"))
table_components$Hum_Mou <- "Human"
table_components[which(table_components$Time %in% c("Ortho", "Sc")), "Hum_Mou"] <- "Mouse"

# boxplot
cairo_pdf(file = "second_component_boxplot_Human_Mouse.pdf")
table_components %>%
  ggplot(aes(x = Hum_Mou,y = PC2)) +
  geom_boxplot() +
  geom_point(aes(fill=Patient,group=Patient),size=4,shape=21, position = position_dodge(0.2)) +
  theme(legend.position = "right") +
  labs(title = "PC2 values between human and mouse")
dev.off()

# wilcox test
res_wilcox_PC2 <- wilcox.test(x = as.vector(table_components[which(table_components$Hum_Mou == "Human"),"PC2"]), y = as.vector(table_components[which(table_components$Hum_Mou == "Mouse"),"PC2"]), alternative = "two.sided")


# enrichr figure by Component
genes_enrichment_PC1 <- names(head(sort(abs(as.table(project_removed.pca$rotation)[,1]), decreasing = T),100))
enrichr_PC1 <- enrichr(genes_enrichment_PC1, dbs_selection)
test_PC1 <- enrichr_PC1$Disease_Signatures_from_GEO_up_2014
test_PC1$PCA <- "PC1"

genes_enrichment_PC2 <- names(head(sort(abs(as.table(project_removed.pca$rotation)[,2]), decreasing = T),100))
enrichr_PC2 <- enrichr(genes_enrichment_PC2, dbs_selection)
test_PC2 <- enrichr_PC2$Disease_Signatures_from_GEO_up_2014
test_PC2$PCA <- "PC2"

test_table <- bind_rows(test_PC1, test_PC2)
test_table <- test_table[which(test_table$Adjusted.P.value <= 0.05),]
test_table <- separate(test_table, col = "Overlap", sep = "/", into = c("nb_genes", "nb_genes_class"), remove = F)
test_table$nb_genes <- as.numeric(test_table$nb_genes)

plot_test_enrich_pca <- ggplot(test_table, aes(x = PCA, y = Term)) + geom_point(aes(colour=Adjusted.P.value, size=nb_genes))
pdf(file = "test_PCA_enrich_plot.pdf")
plot(plot_test_enrich_pca)
dev.off()

# function based on test
PCA_enrichment <- function(prcomp_object, dbs, components, name_file){
  
  cairo_pdf(file = name_file, width = 10, height = 12)
  for(database in dbs){
    first <- T
    for(component in components){
      if(first == T){
        first <- F
        table_enrich_PCA <- names(head(sort(abs(as.table(prcomp_object$rotation)[,component]), decreasing = T),20))
        enrichr_PCA <- enrichr(table_enrich_PCA, database)
        table_enrich_PCA <- enrichr_PCA[1]
        table_enrich_PCA$PCA <- str_c("PC", as.character(component))
      }
      else{
        table_component_sup <- names(head(sort(abs(as.table(prcomp_object$rotation)[,component]), decreasing = T),20))
        enrichr_comp <- enrichr(table_component_sup, database)
        table_component_sup <- enrichr_comp[1]
        table_component_sup$PCA <- str_c("PC", as.character(component))
        table_enrich_PCA <- bind_rows(table_enrich_PCA, table_component_sup)
      }
      
    }
    # print(table_enrich_PCA)
    table_enrich_PCA <- as.data.frame(table_enrich_PCA)
    table_temp <- table_enrich_PCA[,database]
    table_temp$PCA <- table_enrich_PCA[,"PCA"]
    table_enrich_PCA <- table_temp
    #print(table_enrich_PCA[,database])
    print(colnames(table_enrich_PCA))
    #print(str_c(database, '.', 'Adjusted.P.value'))
    table_enrich_PCA <- table_enrich_PCA[which(table_enrich_PCA[,'Adjusted.P.value'] <= 0.05),]
    table_enrich_PCA$temp <- str_count(table_enrich_PCA$Genes, ";")
    table_enrich_PCA <- table_enrich_PCA[which(table_enrich_PCA[,'temp'] > 1),]
    table_enrich_PCA$temp <- NULL
    print(table_enrich_PCA)
    # print(table_enrich_PCA)
    if(nrow(table_enrich_PCA) > 0){
      table_enrich_PCA <- separate(table_enrich_PCA, col = "Overlap", sep = "/", into = c("nb_genes", "nb_genes_class"), remove = F)
      # print(table_enrich_PCA)
      table_enrich_PCA$nb_genes <- as.numeric(table_enrich_PCA$nb_genes)
      plot_enrich_pca <- ggplot(table_enrich_PCA, aes(x = PCA, y = Term)) + geom_point(aes(colour=Adjusted.P.value, size=nb_genes)) + labs(title = database, x = "Component", y = "Term") + scale_color_gradient(high = "blue", low = "red")
      plot(ggplotGrob(plot_enrich_pca))
    }
    
  }
  dev.off()
}

library(stringr)
PCA_enrichment(project_after_diff.pca, dbs_all, c(1,2,3,4,5,6,7), "all_components_enrichment_expression.pdf")
PCA_enrichment(acp_all, "GO_Biological_Process_2021", c(1,2,3,4), "all_components_enrichment_me_separate_scale_GO_biological.pdf")

table_PC1 <- as.data.frame(project_after_diff.pca$rotation[head(names(sort(abs(project_after_diff.pca$rotation[,"PC1"]), decreasing = T)),100),"PC1"])
colnames(table_PC1) <- "PC1"
write.table(table_PC1, file = "Genes_PC1_umap_after_diff.csv", sep = "\t", quote = F, col.names = F, row.names = T)

table_PC3 <- as.data.frame(project_after_diff.pca$rotation[head(names(sort(abs(project_after_diff.pca$rotation[,"PC3"]), decreasing = T)),100),"PC3"])
colnames(table_PC3) <- "PC3"
write.table(table_PC3, file = "Genes_PC3_umap_after_diff.csv", sep = "\t", quote = F, col.names = F, row.names = T)

table_PC4 <- as.data.frame(project_after_diff.pca$rotation[head(names(sort(abs(project_after_diff.pca$rotation[,"PC4"]), decreasing = T)),100),"PC4"])
colnames(table_PC4) <- "PC4"
write.table(table_PC4, file = "Genes_PC4_umap_after_diff.csv", sep = "\t", quote = F, col.names = F, row.names = T)

table_PC5 <- as.data.frame(project_after_diff.pca$rotation[head(names(sort(abs(project_after_diff.pca$rotation[,"PC5"]), decreasing = T)),100),"PC5"])
colnames(table_PC5) <- "PC5"
write.table(table_PC5, file = "Genes_PC5_umap_after_diff.csv", sep = "\t", quote = F, col.names = F, row.names = T)

table_PC6 <- as.data.frame(project_after_diff.pca$rotation[head(names(sort(abs(project_after_diff.pca$rotation[,"PC6"]), decreasing = T)),100),"PC6"])
colnames(table_PC6) <- "PC6"
write.table(table_PC6, file = "Genes_PC6_umap_after_diff.csv", sep = "\t", quote = F, col.names = F, row.names = T)

table_PC7 <- as.data.frame(project_after_diff.pca$rotation[head(names(sort(abs(project_after_diff.pca$rotation[,"PC7"]), decreasing = T)),100),"PC7"])
colnames(table_PC7) <- "PC7"
write.table(table_PC7, file = "Genes_PC7_umap_after_diff.csv", sep = "\t", quote = F, col.names = F, row.names = T)
dev.off()


# plot PCA contribution

# function table prep
pca_sample_contrib <- function(prcomp_x_matrix_per_cent, list_composant){ 
  return_table <- data.frame()
  
  table_sample_contribution <- as.data.frame(abs(t(prcomp_x_matrix_per_cent)))
  table_sample_contribution <- table_sample_contribution[list_composant,]
  #print(table_sample_contribution)
  table_sample_contribution <- apply(table_sample_contribution,2,function(x){x*100/sum(x)})
  #print(rowSums(table_sample_contribution))
  table_sample_contribution <- as.data.frame(t(table_sample_contribution))
  # print(rowSums(table_sample_contribution))

  for(composant in list_composant){
    table_comp <- as.data.frame(table_sample_contribution[,composant])
    rownames(table_comp) <- rownames(table_sample_contribution)
    colnames(table_comp) <- "value"
    table_comp$sample <- rownames(table_comp)
    #print(table_comp)
    table_comp <- separate(table_comp, col = sample, sep = "_", into = c("Patient", "Time"))
    table_comp$PC <- composant
    
    return_table <- bind_rows(return_table, table_comp)
  }
  return_table$Time <- factor(return_table$Time , levels=unique(return_table$Time))
  return_table$Patient <- factor(return_table$Patient , levels=unique(return_table$Patient))
  return(return_table)
}

pca_sample_values <- function(prcomp_x_matrix_per_cent, list_composant){ 
  return_table <- data.frame()
  
  table_sample_contribution <- as.data.frame(prcomp_x_matrix_per_cent)
  table_sample_contribution <- table_sample_contribution[,list_composant]

  for(composant in list_composant){
    table_comp <- as.data.frame(table_sample_contribution[,composant])
    rownames(table_comp) <- rownames(table_sample_contribution)
    colnames(table_comp) <- "value"
    table_comp$sample <- rownames(table_comp)
    #print(table_comp)
    table_comp <- separate(table_comp, col = sample, sep = "_", into = c("Patient", "Time"))
    table_comp$PC <- composant
    
    return_table <- bind_rows(return_table, table_comp)
  }
  return_table$Time <- factor(return_table$Time , levels=unique(return_table$Time))
  return_table$Patient <- factor(return_table$Patient , levels=unique(c(table_comp$Patient)))
  return(return_table)
}

table_values_pca <- pca_sample_values(as.data.frame(project_after_diff.pca$x), c("PC1","PC3", "PC4", "PC5", "PC6", "PC7"))

pdf(file = "sample_values_participation_PCA.pdf", width = 14, height = 10)
g <- ggplot(table_values_pca, aes(x = Time,y = value ,fill = PC)) +
  geom_bar(stat = "identity") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))

g + facet_wrap(~ PC + Patient, ncol=8, labeller = label_wrap_gen(multi_line=FALSE))
plot(g)
dev.off()

table_values_pca_umap <- merge(table_values_pca, umap_plot_df_pca, by=c("Patient", "Time"))

cairo_pdf(file = "umap_all_genes_after_diff_exp_PCA.pdf", height = 12, width = 14)
g <- ggplot(
  table_values_pca_umap,
  aes(
    x = X1,
    y = X2,
    shape = Time,
    size = 1
  )
) +
  geom_point(aes(color = scale(value))) +
  geom_mark_ellipse(aes(x=X1, y=X2, group=Patient, size = 0.02), linetype = 2) +
  scale_color_gradient2(low="blue", mid="white", high="red") +
  theme_dark()
g + facet_wrap(~ PC, ncol = 3)
dev.off()

table_sample_contribution <- pca_sample_contrib(as.data.frame(abs(project_after_diff.pca$x)), c("PC1", "PC2","PC3", "PC4", "PC5", "PC6", "PC7"))

pdf(file = "sample_componant_participation_PCA.pdf", width = 10)
g <- ggplot(table_sample_contribution, aes(x = Time,y = value ,fill = PC)) +
  geom_col(position = "stack", width = 0.6)
g + facet_wrap(~ Patient, ncol=4)
plot(g)
dev.off()

# removed part
table_sample_contribution <- pca_sample_contrib(as.data.frame(abs(project_removed.pca$x)), c("PC1", "PC2", "PC3", "PC4", "PC5", "PC6", "PC7"))

pdf(file = "sample_componant_participation_PCA_removed_genes.pdf", width = 10)
g <- ggplot(table_sample_contribution, aes(x = Time,y = value ,fill = PC)) +
  geom_col(position = "stack", width = 0.6)
g + facet_wrap(~ Patient, ncol=4)
plot(g)
dev.off()
```

```{r enrichissement par PC}
read_tsv("Genes_PC1_umap_after_diff.csv",col_names=c("Genes","Contrib"))  %>% arrange(desc(Contrib)) %>%as.data.frame %>%remove_rownames %>% column_to_rownames("Genes")-> PC1 ; tmp=PC1[,1];   names(tmp) <- rownames(PC1) ; PC1=tmp

read_tsv("Genes_PC3_umap_after_diff.csv",col_names=c("Genes","Contrib"))  %>% arrange(desc(Contrib)) %>%as.data.frame %>%remove_rownames %>% column_to_rownames("Genes")-> PC3 ; tmp=PC3[,1];   names(tmp) <- rownames(PC3) ; PC3=tmp

read_tsv("Genes_PC4_umap_after_diff.csv",col_names=c("Genes","Contrib"))  %>% arrange(desc(Contrib)) %>%as.data.frame %>%remove_rownames %>% column_to_rownames("Genes")-> PC4 ; tmp=PC4[,1];   names(tmp) <- rownames(PC4) ; PC4=tmp

read_tsv("Genes_PC5_umap_after_diff.csv",col_names=c("Genes","Contrib"))  %>% arrange(desc(Contrib)) %>%as.data.frame %>%remove_rownames %>% column_to_rownames("Genes")-> PC5 ; tmp=PC5[,1];   names(tmp) <- rownames(PC5) ; PC5=tmp

read_tsv("Genes_PC6_umap_after_diff.csv",col_names=c("Genes","Contrib"))  %>% arrange(desc(Contrib)) %>%as.data.frame %>%remove_rownames %>% column_to_rownames("Genes")-> PC6 ; tmp=PC6[,1];   names(tmp) <- rownames(PC6) ; PC6=tmp

read_tsv("Genes_PC7_umap_after_diff.csv",col_names=c("Genes","Contrib"))  %>% arrange(desc(Contrib)) %>%as.data.frame %>%remove_rownames %>% column_to_rownames("Genes")-> PC7 ; tmp=PC7[,1];   names(tmp) <- rownames(PC7) ; PC7=tmp


Gost_by_sign=function(x,y,Type="GO:BP"){
  flag1=0;
  flag2=0;
  
  if(length(names(x[which(x>0)])) > 0){
  gost(names(x[which(x>0)]),organism = "hsapiens", significant = FALSE) ->res
  res$result %>% filter(p_value<0.05 & source==Type) %>% mutate(Score=-log(p_value))%>% add_column(PC=y) ->res_pos
  }else{
  flag1=1
};
  
  if(length(names(x[which(x<0)])) > 0){
  gost(names(x[which(x<0)]),organism = "hsapiens", significant = FALSE) ->res
  res$result %>% filter(p_value<0.05 & source==Type) %>% mutate(Score=log(p_value))%>% add_column(PC=y) ->res_neg
  }else{
  flag2=1
  };
  
if(flag1==0 & flag2==0){
rbind(res_pos,res_neg) -> res
}else{
  if(flag1==0){
    res_pos -> res
  }else{
if(flag2==0){
res_neg -> res
}
  }
}
return(res)
}
library(gprofiler2)

list("PC1"=PC1,"PC3"=PC3,"PC4"=PC4,"PC5"=PC5,"PC6"=PC6,"PC7"=PC7) ->LPC

map2(LPC,names(LPC),~Gost_by_sign(.x,.y,Type="GO:BP")) -> All_5GOBP

All_5GOBP %>% bind_rows %>% filter((PC %in% c("PC1","PC3","PC4","PC5","PC6","PC7")) )  %>% mutate(sign=ifelse(Score>0,"P","M")) %>% group_by(PC,sign) %>% top_n(20,wt=abs(Score))%>% ungroup %>%  count((term_name)) %>%  arrange(desc(n)) %>% filter(n>2) %>% dplyr::select(Name=1)-> Num_All_5GOBP

pdf("enrichment_all_components_by_sign_ME.pdf", width = 12, height = 13)
All_5GOBP_to_plot <- All_5GOBP %>% bind_rows %>% filter((PC %in% c("PC1","PC3","PC4","PC5","PC6","PC7"))) %>% filter(!(term_name %in% Num_All_5GOBP$Name)) %>% mutate(sign=ifelse(Score>0,"P","M")) %>% group_by(PC,sign) %>% top_n(20,wt=abs(Score))%>% ungroup %>% mutate(term_PC=fct_reorder(fct_reorder(paste0(term_name,"",PC),desc(Score)),PC))
  ggplot(All_5GOBP_to_plot[order(All_5GOBP_to_plot$PC, decreasing = T),],aes(term_PC,Score,fill=PC)) + geom_bar(stat="identity") + coord_flip() + geom_hline(yintercept = 0,colour="black", linetype = "longdash") + scale_x_discrete(limits = rev(levels(All_5GOBP_to_plot$term_PC)))
dev.off()
```

```{r PCA_plot_function}

PCA_plot <- function(startFile_Name, prcompobject, color_vector, groups, number_PCA){
  
  for (i in 1:number_PCA){
    start = i
    end = i+1
    #file_name = str_c(startFile_Name, "_CP_", as.character(start), "_", as.character(end), ".pdf")
    #pdf(file = file_name)
    p <-fviz_pca_ind(prcompobject,
            axes = c(start,end), 
            col.ind = groups, # colorer par groupes
            palette = color_vector,
            addEllipses = TRUE, # Ellipse de concentration
            ellipse.type = "confidence",
            legend.title = "Groups",
            repel = TRUE
            )
    plot(p)
    #dev.off()
  }
}


PCA_gene_select <- function(table_count, startFile_Name, prcompobject, color_vector, groups, number_PCA, number_genes){
  table_PC_genes <- data.frame()
  all_genes <- c()
  
  for (i in 1:number_PCA){
    list_genes_first_comp <- names(head(sort(abs(as.table(prcompobject$rotation)[,i]), decreasing = T),number_genes))
    # print(as.table(prcompobject$rotation[list_genes_first_comp, str_c("PC", as.character(i))]))
    list_genes_second_comp <- names(head(sort(abs(as.table(prcompobject$rotation)[,i+1]), decreasing = T),number_genes))
    list_genes <- c(list_genes_first_comp, list_genes_second_comp)
    all_genes <- append(all_genes, list_genes)
    # print(all_genes)
    # print(list_genes)
    start = i
    end = i+1
    
    prcompobject2 <- prcomp(t(table_count[unique(list_genes),]))
    p <-fviz_pca_ind(prcompobject2,
            axes = c(start,end), 
            col.ind = groups, # colorer par groupes
            palette = color_vector,
            addEllipses = TRUE, # Ellipse de concentration
            ellipse.type = "confidence",
            legend.title = "Groups",
            repel = TRUE
            )
    plot(p)
    
    table_to_plot <- table_count[list_genes_first_comp,]
    title = str_c(startFile_Name, "_PC_", i, "_enrichment.pdf")
    table_genes_file = str_c(startFile_Name, "_PC_", i, "_enrichment.tsv")
    
    p <- pheatmap::pheatmap(table_to_plot, clustering_method = "ward.D", cellheight = 8, clustering_distance_rows = amap::Dist(table_to_plot, method="euclidean"), clustering_distance_cols = amap::Dist(t(table_to_plot), method = "kendall"), main = title, legend = T)
    enrichment_select_dbs <- enrichr(list_genes_first_comp, dbs_selection)
    plot_enrichr(enrichment_select_dbs, dbs_selection, title)
    
    write.table(as.table(prcompobject$rotation[list_genes_first_comp, str_c("PC", as.character(i))]), file = table_genes_file, sep = "\t", row.names = F, col.names = F)

    
    table_to_plot <- table_count[list_genes_second_comp,]
    # print(table_to_plot[,str_c()])
    title = str_c(startFile_Name, "_PC_", i+1, "_enrichment.pdf")
    table_genes_file = str_c(startFile_Name, "_PC_", i+1, "_enrichment.tsv")
    p <- pheatmap::pheatmap(table_to_plot, clustering_method = "ward.D", cellheight = 8, clustering_distance_rows = amap::Dist(table_to_plot, method="euclidean"), clustering_distance_cols = amap::Dist(t(table_to_plot), method = "kendall"), main = title, legend = T)
    enrichment_select_dbs <- enrichr(list_genes_second_comp, dbs_selection)
    plot_enrichr(enrichment_select_dbs, dbs_selection, title)
    
    write.table(as.table(prcompobject$rotation[list_genes_first_comp, str_c("PC", as.character(i+1))]), file = table_genes_file, sep = "\t", row.names = F, col.names = F)
    
  }
  print(all_genes)
  return(all_genes)
}

```

```{r PCA_gene_selection_RNA_exp}
# starting object : project_after_diff.pca
library(ggfortify)
project_after_diff.pca <- prcomp(t(table_counts_after_diff))
pdf(file = "PCA_after_diff.pdf")
ggplot2::autoplot(project_after_diff.pca, colour = '')
dev.off()

color_patients <- c("#E8552D",  "#2EC021", "#EFBB00", "#1F445C", "#D48B57", "#824587", "#ADA49E", "#4A9AD3")

groups <- as.factor(c("B77", "B77", "B77", "B77","B94", "B94", "B94", "B94","C04", "C04", "C04", "C04","C17", "C17", "C17", "C17","D42", "D42", "D42", "D42","E69", "E69", "E69", "E69", "F59", "F59", "F59", "F59","G36", "G36", "G36", "G36"))

pdf(file = "PCA_counts_after_diff_exp.pdf")
PCA_plot("PCA_after_diff_exp", project_after_diff.pca, color_patients, groups, 7)
dev.off()

p <- fviz_pca_ind(project_after_diff.pca,
             axes = c(1,2), 
             col.ind = groups, # colorer par groupes
             palette = c("#E8552D",  "#2EC021", "#EFBB00", "#1F445C", "#D48B57", "#824587", "#ADA49E", "#4A9AD3"),
             addEllipses = TRUE, # Ellipse de concentration
             ellipse.type = "confidence",
             legend.title = "Groups",
             repel = TRUE
             )

pdf(file = "PCA_after_diff_exp_gene_selection.pdf")
genes_most_important_PCA <- PCA_gene_select(table_counts_after_diff, "PCA_after_diff_exp_gene_selection", project_after_diff.pca, color_patients, groups, 7, 100)
dev.off()

genes_PC1 <- names(head(sort(abs(as.table(project_after_diff.pca$rotation)[,1]),decreasing = T), 50)) 

umap_after_diff_genes_select <- umap(t(table_counts_after_diff[!genes_most_important_PCA %in% genes_PC1,]))

u <- prcomp(t(table_counts_after_diff[genes_most_important_PCA,]))
pdf(file="screeplot.pdf")
fviz_eig(u)
dev.off()

u <- umap(u$x)

u <- umap(project_after_diff.pca$x[,1:5])

umap_plot_df <- data.frame(u$layout) %>%
  # Turn sample IDs stored as row names into a column
  tibble::rownames_to_column("sample") %>%
  # Add the metadata into this data frame; match by sample IDs
  dplyr::inner_join(table_meta_data, by = "sample")

library(ggforce)
cairo_pdf(file = "umap_genes_selected_after_diff_exp_no_PC1.pdf")
ggplot(
  umap_plot_df,
  aes(
    x = X1,
    y = X2,
    color = Hum_Mou,# label points with different colors for each `subgroup`
    shape = Time
  )
) +
  geom_point() +
  #stat_ellipse(aes(x=X1, y=X2,color=Patient, group=Patient),type = "euclid")
  geom_mark_ellipse(aes(x=X1, y=X2,color=Patient, group=Patient, label=Patient))
dev.off()
```

```{r glmnet_multivariate_gaussian}
library(glmnet)
table_variables <- t(table_counts_after_diff)
table_explain <- project_after_diff.pca$x


# PC1
cv <- cv.glmnet(x = table_variables,y = table_explain[,"PC1"] ,alpha=1)
res_glmnet <- glmnet(x = table_variables, y = table_explain[,"PC1"], family = "mgaussian", alpha = 1, lambda = cv$lambda.1se)
predict_res_glmnet <- predict.glmnet(res_glmnet, newx = new_matrix, s = res_glmnet$lambda, type=c("coefficients"))
coeff_PC1 <- as.matrix(predict_res_glmnet)
coeff_PC1 <- as.data.frame(coeff_PC1)
coeff_PC1$Genes <- rownames(coeff_PC1)
coeff_PC1 <- coeff_PC1[which(abs(coeff_PC1[,1]) > 0),]
remove <- c("(Intercept)")
coeff_PC1 <- coeff_PC1[!(row.names(coeff_PC1) %in% remove), ]
write.table(coeff_PC1, file = "PC1_glmnet_genes_selection.csv", sep = "\t", quote = F, row.names = F)
coeff_PC1$PC <- "PC1"

#PC2
cv <- cv.glmnet(x = table_variables,y = table_explain[,"PC2"] ,alpha=1)
res_glmnet <- glmnet(x = table_variables, y = table_explain[,"PC2"], family = "mgaussian", alpha = 1, lambda = cv$lambda.1se)
predict_res_glmnet <- predict.glmnet(res_glmnet, newx = new_matrix, s = res_glmnet$lambda, type=c("coefficients"))
coeff_PC2 <- as.matrix(predict_res_glmnet)
coeff_PC2 <- as.data.frame(coeff_PC2)
coeff_PC2$Genes <- rownames(coeff_PC2)
coeff_PC2 <- coeff_PC2[which(abs(coeff_PC2[,1]) > 0),]
remove <- c("(Intercept)")
coeff_PC2 <- coeff_PC2[!(row.names(coeff_PC2) %in% remove), ]
write.table(coeff_PC2, file = "PC2_glmnet_genes_selection.csv", sep = "\t", quote = F, row.names = F)
coeff_PC2$PC <- "PC2"

#PC3
cv <- cv.glmnet(x = table_variables,y = table_explain[,"PC3"] ,alpha=1)
res_glmnet <- glmnet(x = table_variables, y = table_explain[,"PC3"], family = "mgaussian", alpha = 1, lambda = cv$lambda.1se)
predict_res_glmnet <- predict.glmnet(res_glmnet, newx = new_matrix, s = res_glmnet$lambda, type=c("coefficients"))
coeff_PC3 <- as.matrix(predict_res_glmnet)
coeff_PC3 <- as.data.frame(coeff_PC3)
coeff_PC3$Genes <- rownames(coeff_PC3)
coeff_PC3 <- coeff_PC3[which(abs(coeff_PC3[,1]) > 0),]
remove <- c("(Intercept)")
coeff_PC3 <- coeff_PC3[!(row.names(coeff_PC3) %in% remove), ]
write.table(coeff_PC3, file = "PC3_glmnet_genes_selection.csv", sep = "\t", quote = F, row.names = F)
coeff_PC3$PC <- "PC3"

#PC4
cv <- cv.glmnet(x = table_variables,y = table_explain[,"PC4"] ,alpha=1)
res_glmnet <- glmnet(x = table_variables, y = table_explain[,"PC4"], family = "mgaussian", alpha = 1, lambda = cv$lambda.1se)
predict_res_glmnet <- predict.glmnet(res_glmnet, newx = new_matrix, s = res_glmnet$lambda, type=c("coefficients"))
coeff_PC4 <- as.matrix(predict_res_glmnet)
coeff_PC4 <- as.data.frame(coeff_PC4)
coeff_PC4$Genes <- rownames(coeff_PC4)
coeff_PC4 <- coeff_PC4[which(abs(coeff_PC4[,1]) > 0),]
remove <- c("(Intercept)")
coeff_PC4 <- coeff_PC4[!(row.names(coeff_PC4) %in% remove), ]
write.table(coeff_PC4, file = "PC4_glmnet_genes_selection.csv", sep = "\t", quote = F, row.names = F)
coeff_PC4$PC <- "PC4"

#PC5
cv <- cv.glmnet(x = table_variables,y = table_explain[,"PC5"] ,alpha=1)
res_glmnet <- glmnet(x = table_variables, y = table_explain[,"PC5"], family = "mgaussian", alpha = 1, lambda = cv$lambda.1se)
predict_res_glmnet <- predict.glmnet(res_glmnet, newx = new_matrix, s = res_glmnet$lambda, type=c("coefficients"))
coeff_PC5 <- as.matrix(predict_res_glmnet)
coeff_PC5 <- as.data.frame(coeff_PC5)
coeff_PC5$Genes <- rownames(coeff_PC5)
coeff_PC5 <- coeff_PC5[which(abs(coeff_PC5[,1]) > 0),]
remove <- c("(Intercept)")
coeff_PC5 <- coeff_PC5[!(row.names(coeff_PC5) %in% remove), ]
write.table(coeff_PC5, file = "PC5_glmnet_genes_selection.csv", sep = "\t", quote = F, row.names = F)
coeff_PC5$PC <- "PC5"

#PC6
cv <- cv.glmnet(x = table_variables,y = table_explain[,"PC6"] ,alpha=1)
res_glmnet <- glmnet(x = table_variables, y = table_explain[,"PC6"], family = "mgaussian", alpha = 1, lambda = cv$lambda.1se)
predict_res_glmnet <- predict.glmnet(res_glmnet, newx = new_matrix, s = res_glmnet$lambda, type=c("coefficients"))
coeff_PC6 <- as.matrix(predict_res_glmnet)
coeff_PC6 <- as.data.frame(coeff_PC6)
coeff_PC6$Genes <- rownames(coeff_PC6)
coeff_PC6 <- coeff_PC6[which(abs(coeff_PC6[,1]) > 0),]
remove <- c("(Intercept)")
coeff_PC6 <- coeff_PC6[!(row.names(coeff_PC6) %in% remove), ]
write.table(coeff_PC6, file = "PC6_glmnet_genes_selection.csv", sep = "\t", quote = F, row.names = F)
coeff_PC6$PC <- "PC6"

#PC7
cv <- cv.glmnet(x = table_variables,y = table_explain[,"PC7"] ,alpha=1)
res_glmnet <- glmnet(x = table_variables, y = table_explain[,"PC7"], family = "mgaussian", alpha = 1, lambda = cv$lambda.1se)
predict_res_glmnet <- predict.glmnet(res_glmnet, newx = new_matrix, s = res_glmnet$lambda, type=c("coefficients"))
coeff_PC7 <- as.matrix(predict_res_glmnet)
coeff_PC7 <- as.data.frame(coeff_PC7)
coeff_PC7$Genes <- rownames(coeff_PC7)
coeff_PC7 <- coeff_PC7[which(abs(coeff_PC7[,1]) > 0),]
remove <- c("(Intercept)")
coeff_PC7 <- coeff_PC7[!(row.names(coeff_PC7) %in% remove), ]
write.table(coeff_PC7, file = "PC7_glmnet_genes_selection.csv", sep = "\t", quote = F, row.names = F)
coeff_PC7$PC <- "PC7"

genes_PC1_glmnet <- rownames(coeff_PC1)
genes_PC2_glmnet <- rownames(coeff_PC2)
genes_PC3_glmnet <- rownames(coeff_PC3)
genes_PC4_glmnet <- rownames(coeff_PC4)
genes_PC5_glmnet <- rownames(coeff_PC5)
genes_PC6_glmnet <- rownames(coeff_PC6)
genes_PC7_glmnet <- rownames(coeff_PC7)

enrichment_PC1 <- enrichr(genes_PC1_glmnet, dbs_all)
plot_enrichr(enrichment_PC1, dbs_all, "plot_enrichment_PC1_glmnet_expression.pdf")

enrichment_PC2 <- enrichr(genes_PC2_glmnet, dbs_all)
plot_enrichr(enrichment_PC2, dbs_all, "plot_enrichment_PC2_glmnet_expression.pdf")

enrichment_PC3 <- enrichr(genes_PC3_glmnet, dbs_all)
plot_enrichr(enrichment_PC3, dbs_all, "plot_enrichment_PC3_glmnet_expression.pdf")

enrichment_PC4 <- enrichr(genes_PC4_glmnet, dbs_all)
plot_enrichr(enrichment_PC4, dbs_all, "plot_enrichment_PC4_glmnet_expression.pdf")

enrichment_PC5 <- enrichr(genes_PC5_glmnet, dbs_all)
plot_enrichr(enrichment_PC5, dbs_all, "plot_enrichment_PC5_glmnet_expression.pdf")

enrichment_PC6 <- enrichr(genes_PC6_glmnet, dbs_all)
plot_enrichr(enrichment_PC6, dbs_all, "plot_enrichment_PC6_glmnet_expression.pdf")

enrichment_PC7 <- enrichr(genes_PC7_glmnet, dbs_all)
plot_enrichr(enrichment_PC7, dbs_all, "plot_enrichment_PC7_glmnet_expression.pdf")

cairo_pdf(file = "enrichment_PCA_glmnet_selected.pdf")
for(dbs in 1:length(enrichment_PC1)){
 enrich_dbs_PC1 <- enrichment_PC1[dbs]
 enrich_dbs_PC1 <- as.data.frame(as.matrix(enrich_dbs_PC1[[1]]))
 if(nrow(enrich_dbs_PC1) > 0){
 enrich_dbs_PC1$PCA <- "PC1"}else{enrich_dbs_PC1 <- data.frame()}
 
 enrich_dbs_PC3 <- enrichment_PC3[dbs]
 enrich_dbs_PC3 <- as.data.frame(as.matrix(enrich_dbs_PC3[[1]]))
 if(nrow(enrich_dbs_PC3) > 0){
 enrich_dbs_PC3$PCA <- "PC3"}else{enrich_dbs_PC3 <- data.frame()}
 
 enrich_dbs_PC4 <- enrichment_PC4[dbs]
 enrich_dbs_PC4 <- as.data.frame(as.matrix(enrich_dbs_PC4[[1]]))
 if(nrow(enrich_dbs_PC4) > 0){
 enrich_dbs_PC4$PCA <- "PC4"}else{enrich_dbs_PC4 <- data.frame()}
 
 enrich_dbs_PC5 <- enrichment_PC5[dbs]
 enrich_dbs_PC5 <- as.data.frame(as.matrix(enrich_dbs_PC5[[1]]))
 if(nrow(enrich_dbs_PC5) > 0){
 enrich_dbs_PC5$PCA <- "PC5"}else{enrich_dbs_PC5 <- data.frame()}
 
 enrich_dbs_PC6 <- enrichment_PC6[dbs]
 enrich_dbs_PC6 <- as.data.frame(as.matrix(enrich_dbs_PC6[[1]]))
 if(nrow(enrich_dbs_PC6) > 0){
 enrich_dbs_PC6$PCA <- "PC6"}else{enrich_dbs_PC6 <- data.frame()}
 
 enrich_dbs_PC7 <- enrichment_PC7[dbs]
 enrich_dbs_PC7 <- as.data.frame(as.matrix(enrich_dbs_PC7[[1]]))
 if(nrow(enrich_dbs_PC7) > 0){
 enrich_dbs_PC7$PCA <- "PC7"}else{enrich_dbs_PC7 <- data.frame()}
 
 table_enrich_PCA <- bind_rows(enrich_dbs_PC1, enrich_dbs_PC3, enrich_dbs_PC4, enrich_dbs_PC5, enrich_dbs_PC6, enrich_dbs_PC7)
 # table_enrich_PCA <- as.matrix(table_enrich_PCA[1])
 # column_select <- str_c(names(enrichment_PC1[dbs])[1],".Overlap")
 table_enrich_PCA$Adjusted.P.value <- as.numeric(table_enrich_PCA$Adjusted.P.value)
 table_enrich_PCA <- table_enrich_PCA[which(table_enrich_PCA[,"Adjusted.P.value"] <= 0.05),]
 table_enrich_PCA <- as.data.frame(table_enrich_PCA)
 
 if(nrow(table_enrich_PCA) > 0){
      table_enrich_PCA <- separate(table_enrich_PCA, col = "Overlap", sep = "/", into = c("nb_genes", "nb_genes_class"), remove = F)
      # print(table_enrich_PCA)
      table_enrich_PCA$nb_genes <- as.numeric(table_enrich_PCA$nb_genes)
      plot_enrich_pca <- ggplot(table_enrich_PCA, aes(x = PCA, y = Term)) + geom_point(aes(colour=Adjusted.P.value, size=nb_genes)) + labs(title = enrichment_PC1[dbs], x = "Component", y = "Term") + scale_color_gradient(high = "blue", low = "red")
      plot(ggplotGrob(plot_enrich_pca))
 }
}
dev.off()


genes_glmnet_selection <- bind_rows(coeff_PC1, coeff_PC3, coeff_PC4, coeff_PC5, coeff_PC6, coeff_PC7)
u_glmnet <- prcomp(t(table_counts_after_diff[genes_glmnet_selection$Genes,]))

u_glmnet <- umap(u_glmnet$x)

umap_plot_df <- data.frame(u_glmnet$layout) %>%
  # Turn sample IDs stored as row names into a column
  tibble::rownames_to_column("sample") %>%
  # Add the metadata into this data frame; match by sample IDs
  dplyr::inner_join(table_meta_data, by = "sample")

library(ggforce)
cairo_pdf(file = "umap_glmnet_selected.pdf")
ggplot(
  umap_plot_df,
  aes(
    x = X1,
    y = X2,
    color = Patient,# label points with different colors for each `subgroup`
    shape = Time,
    size = 3
  )
) +
  geom_point() +
  geom_mark_ellipse(aes(x=X1, y=X2,color=Patient, group=Patient, size = 0.2))
  #stat_ellipse(aes(x=X1, y=X2,color=Patient, group=Patient),type = "euclid")
dev.off()

# Heatmap genes PC glmnet selected
table_count_genes_glmnet <- table_counts_after_diff[genes_glmnet_selection$Genes,]

library(ggsci)
gene_annotation <- genes_glmnet_selection[,c("Genes", "PC")]
gene_annotation$Genes <- NULL
pheatmap::pheatmap(table_count_genes_glmnet, clustering_method = "ward.D", filename = "heatmap_counts_glmnet_selection_scale.pdf", cellheight = 8, scale = "row", cluster_rows = F, clustering_distance_cols = amap::Dist(t(table_count_genes_glmnet), method = "kendall"), annotation_row = gene_annotation, annotation_col = table_meta_data[,c("Patient", "Hum_Mou")], color = c("#327ded","#32edc1", "#faffb8", "#ffa196", "#e81f07"), breaks = c(-5, -3, -1.5, 1.5, 3, 5))

```

```{r top genes each component}

top_genes_PC1 <- names(head(sort(abs(as.table(project_after_diff.pca$rotation)[,1]),decreasing = T), 20))
top_genes_PC1 <- as.data.frame(top_genes_PC1)
top_genes_PC1$PC <- "PC1"
colnames(top_genes_PC1) <- c("Gene", "PC")

top_genes_PC3 <- names(head(sort(abs(as.table(project_after_diff.pca$rotation)[,3]),decreasing = T), 20))
top_genes_PC3 <- as.data.frame(top_genes_PC3)
top_genes_PC3$PC <- "PC3"
colnames(top_genes_PC3) <- c("Gene", "PC")

top_genes_PC4 <- names(head(sort(abs(as.table(project_after_diff.pca$rotation)[,4]),decreasing = T), 20))
top_genes_PC4 <- as.data.frame(top_genes_PC4)
top_genes_PC4$PC <- "PC4"
colnames(top_genes_PC4) <- c("Gene", "PC")

top_genes_PC5 <- names(head(sort(abs(as.table(project_after_diff.pca$rotation)[,5]),decreasing = T), 20))
top_genes_PC5 <- as.data.frame(top_genes_PC5)
top_genes_PC5$PC <- "PC5"
colnames(top_genes_PC5) <- c("Gene", "PC")

top_genes_PC6 <- names(head(sort(abs(as.table(project_after_diff.pca$rotation)[,6]),decreasing = T), 20))
top_genes_PC6 <- as.data.frame(top_genes_PC6)
top_genes_PC6$PC <- "PC6"
colnames(top_genes_PC6) <- c("Gene", "PC")

top_genes_PC7 <- names(head(sort(abs(as.table(project_after_diff.pca$rotation)[,7]),decreasing = T), 20))
top_genes_PC7 <- as.data.frame(top_genes_PC7)
top_genes_PC7$PC <- "PC7"
colnames(top_genes_PC7) <- c("Gene", "PC")

top_genes <- bind_rows(top_genes_PC1, top_genes_PC3, top_genes_PC4, top_genes_PC5, top_genes_PC6, top_genes_PC7)
top_genes <- top_genes[!duplicated(top_genes[,c('Gene')]),]

table_count_top_genes <- table_counts_after_diff[as.character(top_genes$Gene),]

library(ggsci)
library("RColorBrewer")
top_gene_annotation <- top_genes
rownames(top_gene_annotation) <- top_genes$Gene
table_count_top_genes <- table_count_top_genes[top_genes$Gene,]
top_gene_annotation$Gene <- NULL

pheatmap::pheatmap(table_count_top_genes, clustering_method = "ward.D", filename = "heatmap_counts_pca_top_genes_scale.pdf", cellheight = 8, scale = "row", cluster_rows = F, clustering_distance_cols = amap::Dist(t(table_count_top_genes), method = "kendall"), annotation_row = top_gene_annotation, annotation_col = table_meta_data[,c("Patient", "Hum_Mou")], color = c("#327ded","#32edc1", "#faffb8", "#ffa196", "#e81f07"), breaks = c(-5, -3, -1.5, 1.5, 3, 5))

# annotation_colors = brewer.pal(n = 6, name = 'Dark2')
```

```{r cnv_genes_annotation_function}
library(biomaRt)
mart <- useMart(biomart="ENSEMBL_MART_ENSEMBL", host="grch37.ensembl.org", path="/biomart/martservice", dataset="hsapiens_gene_ensembl")

cnv_gene_annot <- function(table_cnv_prep, sample_name, mart){
  
  cnv_Amp <- table_cnv_prep[which(table_cnv_prep$CNvalue == 2 & table_cnv_prep$length > 500000),]
  if(nrow(cnv_Amp) != 0){
    genes_Amp = getBM(attributes = c("ensembl_gene_id","hgnc_symbol","hpa_accession"), filters = 'chromosomal_region', values = cnv_Amp$query, mart = mart)
    genes_Amp <- as.data.frame(unique(genes_Amp$hgnc_symbol))
    colnames(genes_Amp) <- "symbol"
    genes_Amp$cnv_type <- "Amplification"
    genes_Amp$sample <- sample_name
  }
  
  cnv_Focal_Amp <- table_cnv_prep[which(table_cnv_prep$CNvalue == 2 & table_cnv_prep$length <= 500000),]
  if(nrow(cnv_Focal_Amp) != 0){
    genes_Focal_Amp = getBM(attributes = c("ensembl_gene_id","hgnc_symbol","hpa_accession"), filters = 'chromosomal_region', values = cnv_Focal_Amp$query, mart = mart)
    genes_Focal_Amp <- as.data.frame(unique(genes_Focal_Amp$hgnc_symbol))
    colnames(genes_Focal_Amp) <- "symbol"
    genes_Focal_Amp$cnv_type <- "Focal_Amplification"
    genes_Focal_Amp$sample <- sample_name
  }

  cnv_gain <- table_cnv_prep[which(table_cnv_prep$CNvalue == 1),]
  genes_gain = getBM(attributes = c("ensembl_gene_id","hgnc_symbol","hpa_accession"), filters = 'chromosomal_region', values = cnv_gain$query, mart = mart)
  genes_gain <- as.data.frame(unique(genes_gain$hgnc_symbol))
  colnames(genes_gain) <- "symbol"
  genes_gain$cnv_type <- "Gain"
  genes_gain$sample <- sample_name

  cnv_normal <- table_cnv_prep[which(table_cnv_prep$CNvalue == 0),]
  genes_normal = getBM(attributes = c("ensembl_gene_id","hgnc_symbol","hpa_accession"), filters = 'chromosomal_region', values = cnv_normal$query, mart = mart)
  genes_normal <- as.data.frame(unique(genes_normal$hgnc_symbol))
  colnames(genes_normal) <- "symbol"
  genes_normal$cnv_type <- "Normal"
  genes_normal$sample <- sample_name

  cnv_loss <- table_cnv_prep[which(table_cnv_prep$CNvalue == -1),]
  if(nrow(cnv_loss) != 0){
    genes_loss = getBM(attributes = c("ensembl_gene_id","hgnc_symbol","hpa_accession"), filters = 'chromosomal_region', values = cnv_loss$query, mart = mart)
    genes_loss <- as.data.frame(unique(genes_loss$hgnc_symbol))
    colnames(genes_loss) <- "symbol"
    genes_loss$cnv_type <- "Loss"
    genes_loss$sample <- sample_name
  }
  else{
    genes_loss <- data.frame()
  }

  cnv_del <- table_cnv_prep[which(table_cnv_prep$CNvalue == -2),]
  if(nrow(cnv_del) != 0){
    genes_del = getBM(attributes = c("ensembl_gene_id","hgnc_symbol","hpa_accession"), filters = 'chromosomal_region', values = cnv_del$query, mart = mart)
    if(nrow(genes_del) != 0){
      genes_del <- as.data.frame(unique(genes_del$hgnc_symbol))
      colnames(genes_del) <- "symbol"
      genes_del$cnv_type <- "Deletion"
      genes_del$sample <- sample_name
    }
    else{
      gene_del <- data.frame()
    }
  }
  else{
    gene_del <- data.frame()
  }
  
  if(nrow(cnv_Focal_Amp) != 0 & nrow(cnv_del) != 0 & nrow(cnv_Amp) != 0){
    table_cnv <- bind_rows(genes_Amp, genes_Focal_Amp, genes_gain, genes_normal, genes_loss, genes_del)
  }
  else if(nrow(cnv_Focal_Amp) == 0 & nrow(cnv_del) != 0 & nrow(cnv_Amp) != 0){
    table_cnv <- bind_rows(genes_Amp, genes_gain, genes_normal, genes_loss, genes_del)
  }
  else if(nrow(cnv_Focal_Amp) != 0 & nrow(cnv_del) == 0 & nrow(cnv_Amp) != 0){
    table_cnv <- bind_rows(genes_Amp, genes_Focal_Amp, genes_gain, genes_normal, genes_loss)
  }
  else if(nrow(cnv_Focal_Amp) != 0 & nrow(cnv_del) != 0 & nrow(cnv_Amp) == 0){
    table_cnv <- bind_rows(genes_Focal_Amp, genes_gain, genes_normal, genes_loss, genes_del)
  }
  else if(nrow(cnv_Focal_Amp) != 0 & nrow(cnv_del) == 0 & nrow(cnv_Amp) == 0){
    table_cnv <- bind_rows(genes_Focal_Amp, genes_gain, genes_normal, genes_loss)
  }
  else if(nrow(cnv_Focal_Amp) == 0 & nrow(cnv_del) == 0 & nrow(cnv_Amp) != 0){
    table_cnv <- bind_rows(genes_Amp, genes_gain, genes_normal, genes_loss)
  }
  else if(nrow(cnv_Focal_Amp) == 0 & nrow(cnv_del) != 0 & nrow(cnv_Amp) == 0){
    table_cnv <- bind_rows(genes_gain, genes_normal, genes_loss, genes_del)
  }
  else{
    table_cnv <- bind_rows(genes_gain, genes_normal, genes_loss)
  }
  return(table_cnv)
}

```

```{r gene_annotation_cnv_B77}
# MAP177
cnv_B77_Diag <- CNV_prep("../../facets/MAP177_Diag_cncf.txt")
cnv_B77_Diag <- separate(data = cnv_B77_Diag, col = Chromosome, into = c("str", "chr"), sep = "r")
cnv_B77_Diag$query <- str_c(cnv_B77_Diag$chr, ":", cnv_B77_Diag$chromStart, ":", cnv_B77_Diag$chromEnd)
cnv_B77_Diag_genes <- cnv_gene_annot(cnv_B77_Diag, "B77_Diag", mart)

cnv_B77_Relapse <- CNV_prep("../../facets/MAP177_Relapse_cncf.txt")
cnv_B77_Relapse <- separate(data = cnv_B77_Relapse, col = Chromosome, into = c("str", "chr"), sep = "r")
cnv_B77_Relapse$query <- str_c(cnv_B77_Relapse$chr, ":", cnv_B77_Relapse$chromStart, ":", cnv_B77_Relapse$chromEnd)
cnv_B77_Relapse_genes <- cnv_gene_annot(cnv_B77_Relapse, "B77_Relapse", mart)

cnv_B77_Ortho <- CNV_prep("../../facets/MAP177_Ortho_cncf.txt")
cnv_B77_Ortho <- separate(data = cnv_B77_Ortho, col = Chromosome, into = c("str", "chr"), sep = "r")
cnv_B77_Ortho$query <- str_c(cnv_B77_Ortho$chr, ":", cnv_B77_Ortho$chromStart, ":", cnv_B77_Ortho$chromEnd)
cnv_B77_Ortho_genes <- cnv_gene_annot(cnv_B77_Ortho, "B77_Ortho", mart)

cnv_B77_Sc <- CNV_prep("../../facets/MAP177_Sc_cncf.txt")
cnv_B77_Sc <- separate(data = cnv_B77_Sc, col = Chromosome, into = c("str", "chr"), sep = "r")
cnv_B77_Sc$query <- str_c(cnv_B77_Sc$chr, ":", cnv_B77_Sc$chromStart, ":", cnv_B77_Sc$chromEnd)
cnv_B77_Sc_genes <- cnv_gene_annot(cnv_B77_Sc, "B77_Sc", mart)

cnv_B77_genes <- bind_rows(cnv_B77_Diag_genes, cnv_B77_Relapse_genes, cnv_B77_Ortho_genes, cnv_B77_Sc_genes)
```

```{r gene_annotation_cnv_B94}
# MAP194
cnv_B94_Diag <- CNV_prep("../../facets/MAP194_Diag_cncf.txt")
cnv_B94_Diag <- separate(data = cnv_B94_Diag, col = Chromosome, into = c("str", "chr"), sep = "r")
cnv_B94_Diag$query <- str_c(cnv_B94_Diag$chr, ":", cnv_B94_Diag$chromStart, ":", cnv_B94_Diag$chromEnd)
cnv_B94_Diag_genes <- cnv_gene_annot(cnv_B94_Diag, "B94_Diag", mart)

cnv_B94_Relapse <- CNV_prep("../../facets/MAP194_Relapse_cncf.txt")
cnv_B94_Relapse <- separate(data = cnv_B94_Relapse, col = Chromosome, into = c("str", "chr"), sep = "r")
cnv_B94_Relapse$query <- str_c(cnv_B94_Relapse$chr, ":", cnv_B94_Relapse$chromStart, ":", cnv_B94_Relapse$chromEnd)
cnv_B94_Relapse_genes <- cnv_gene_annot(cnv_B94_Relapse, "B94_Relapse", mart)

cnv_B94_Ortho <- CNV_prep("../../facets/MAP194_Ortho_cncf.txt")
cnv_B94_Ortho <- separate(data = cnv_B94_Ortho, col = Chromosome, into = c("str", "chr"), sep = "r")
cnv_B94_Ortho$query <- str_c(cnv_B94_Ortho$chr, ":", cnv_B94_Ortho$chromStart, ":", cnv_B94_Ortho$chromEnd)
cnv_B94_Ortho_genes <- cnv_gene_annot(cnv_B94_Ortho, "B94_Ortho", mart)

cnv_B94_Sc <- CNV_prep("../../facets/MAP194_Sc_cncf.txt")
cnv_B94_Sc <- separate(data = cnv_B94_Sc, col = Chromosome, into = c("str", "chr"), sep = "r")
cnv_B94_Sc$query <- str_c(cnv_B94_Sc$chr, ":", cnv_B94_Sc$chromStart, ":", cnv_B94_Sc$chromEnd)
cnv_B94_Sc_genes <- cnv_gene_annot(cnv_B94_Sc, "B94_Sc", mart)

cnv_B94_genes <- bind_rows(cnv_B94_Diag_genes, cnv_B94_Relapse_genes, cnv_B94_Ortho_genes, cnv_B94_Sc_genes)
```

```{r gene_annotation_cnv_C17}
# MAP217
cnv_C17_Diag <- CNV_prep("../../facets/MAP217_Diag_cncf.txt")
cnv_C17_Diag <- separate(data = cnv_C17_Diag, col = Chromosome, into = c("str", "chr"), sep = "r")
cnv_C17_Diag$query <- str_c(cnv_C17_Diag$chr, ":", cnv_C17_Diag$chromStart, ":", cnv_C17_Diag$chromEnd)
cnv_C17_Diag_genes <- cnv_gene_annot(cnv_C17_Diag, "C17_Diag", mart)

cnv_C17_Relapse <- CNV_prep("../../facets/MAP217_Relapse_cncf.txt")
cnv_C17_Relapse <- separate(data = cnv_C17_Relapse, col = Chromosome, into = c("str", "chr"), sep = "r")
cnv_C17_Relapse$query <- str_c(cnv_C17_Relapse$chr, ":", cnv_C17_Relapse$chromStart, ":", cnv_C17_Relapse$chromEnd)
cnv_C17_Relapse_genes <- cnv_gene_annot(cnv_C17_Relapse, "C17_Relapse", mart)

cnv_C17_Ortho <- CNV_prep("../../facets/MAP217_Ortho_cncf.txt")
cnv_C17_Ortho <- separate(data = cnv_C17_Ortho, col = Chromosome, into = c("str", "chr"), sep = "r")
cnv_C17_Ortho$query <- str_c(cnv_C17_Ortho$chr, ":", cnv_C17_Ortho$chromStart, ":", cnv_C17_Ortho$chromEnd)
cnv_C17_Ortho_genes <- cnv_gene_annot(cnv_C17_Ortho, "C17_Ortho", mart)

cnv_C17_Sc <- CNV_prep("../../facets/MAP217_Sc_cncf.txt")
cnv_C17_Sc <- separate(data = cnv_C17_Sc, col = Chromosome, into = c("str", "chr"), sep = "r")
cnv_C17_Sc$query <- str_c(cnv_C17_Sc$chr, ":", cnv_C17_Sc$chromStart, ":", cnv_C17_Sc$chromEnd)
cnv_C17_Sc_genes <- cnv_gene_annot(cnv_C17_Sc, "C17_Sc", mart)

cnv_C17_genes <- bind_rows(cnv_C17_Diag_genes, cnv_C17_Relapse_genes, cnv_C17_Ortho_genes, cnv_C17_Sc_genes)
```

```{r gene_annotation_cnv_D42}
# MAP342
cnv_D42_Diag <- CNV_prep("../../facets/MAP342_Diag_cncf.txt")
cnv_D42_Diag <- separate(data = cnv_D42_Diag, col = Chromosome, into = c("str", "chr"), sep = "r")
cnv_D42_Diag$query <- str_c(cnv_D42_Diag$chr, ":", cnv_D42_Diag$chromStart, ":", cnv_D42_Diag$chromEnd)
cnv_D42_Diag_genes <- cnv_gene_annot(cnv_D42_Diag, "D42_Diag", mart)

cnv_D42_Relapse <- CNV_prep("../../facets/MAP342_Relapse_cncf.txt")
cnv_D42_Relapse <- separate(data = cnv_D42_Relapse, col = Chromosome, into = c("str", "chr"), sep = "r")
cnv_D42_Relapse$query <- str_c(cnv_D42_Relapse$chr, ":", cnv_D42_Relapse$chromStart, ":", cnv_D42_Relapse$chromEnd)
cnv_D42_Relapse_genes <- cnv_gene_annot(cnv_D42_Relapse, "D42_Relapse", mart)

cnv_D42_Ortho <- CNV_prep("../../facets/MAP342_Ortho_cncf.txt")
cnv_D42_Ortho <- separate(data = cnv_D42_Ortho, col = Chromosome, into = c("str", "chr"), sep = "r")
cnv_D42_Ortho$query <- str_c(cnv_D42_Ortho$chr, ":", cnv_D42_Ortho$chromStart, ":", cnv_D42_Ortho$chromEnd)
cnv_D42_Ortho_genes <- cnv_gene_annot(cnv_D42_Ortho, "D42_Ortho", mart)

cnv_D42_Sc <- CNV_prep("../../facets/MAP342_Sc_Ortho-meta_cncf.txt")
cnv_D42_Sc <- separate(data = cnv_D42_Sc, col = Chromosome, into = c("str", "chr"), sep = "r")
cnv_D42_Sc$query <- str_c(cnv_D42_Sc$chr, ":", cnv_D42_Sc$chromStart, ":", cnv_D42_Sc$chromEnd)
cnv_D42_Sc_genes <- cnv_gene_annot(cnv_D42_Sc, "D42_Sc", mart)

cnv_D42_genes <- bind_rows(cnv_D42_Diag_genes, cnv_D42_Relapse_genes, cnv_D42_Ortho_genes, cnv_D42_Sc_genes)
```

```{r gene_annotation_cnv_E69}
# MAP194
cnv_E69_Diag <- CNV_prep("../../facets/MAP469_Diag_cncf.txt")
cnv_E69_Diag <- separate(data = cnv_E69_Diag, col = Chromosome, into = c("str", "chr"), sep = "r")
cnv_E69_Diag$query <- str_c(cnv_E69_Diag$chr, ":", cnv_E69_Diag$chromStart, ":", cnv_E69_Diag$chromEnd)
cnv_E69_Diag_genes <- cnv_gene_annot(cnv_E69_Diag, "E69_Diag", mart)

cnv_E69_Relapse <- CNV_prep("../../facets/MAP469_Relapse_cncf.txt")
cnv_E69_Relapse <- separate(data = cnv_E69_Relapse, col = Chromosome, into = c("str", "chr"), sep = "r")
cnv_E69_Relapse$query <- str_c(cnv_E69_Relapse$chr, ":", cnv_E69_Relapse$chromStart, ":", cnv_E69_Relapse$chromEnd)
cnv_E69_Relapse_genes <- cnv_gene_annot(cnv_E69_Relapse, "E69_Relapse", mart)

cnv_E69_Ortho <- CNV_prep("../../facets/MAP469_Ortho_cncf.txt")
cnv_E69_Ortho <- separate(data = cnv_E69_Ortho, col = Chromosome, into = c("str", "chr"), sep = "r")
cnv_E69_Ortho$query <- str_c(cnv_E69_Ortho$chr, ":", cnv_E69_Ortho$chromStart, ":", cnv_E69_Ortho$chromEnd)
cnv_E69_Ortho_genes <- cnv_gene_annot(cnv_E69_Ortho, "E69_Ortho", mart)

cnv_E69_Sc <- CNV_prep("../../facets/MAP469_Sc_cncf.txt")
cnv_E69_Sc <- separate(data = cnv_E69_Sc, col = Chromosome, into = c("str", "chr"), sep = "r")
cnv_E69_Sc$query <- str_c(cnv_E69_Sc$chr, ":", cnv_E69_Sc$chromStart, ":", cnv_E69_Sc$chromEnd)
cnv_E69_Sc_genes <- cnv_gene_annot(cnv_E69_Sc, "E69_Sc", mart)

cnv_E69_genes <- bind_rows(cnv_E69_Diag_genes, cnv_E69_Relapse_genes, cnv_E69_Ortho_genes, cnv_E69_Sc_genes)
```

```{r gene_annotation_cnv_F59}
# MAP559
cnv_F59_Relapse <- CNV_prep("../../facets/MAP559_Relapse_cncf.txt")
cnv_F59_Relapse <- separate(data = cnv_F59_Relapse, col = Chromosome, into = c("str", "chr"), sep = "r")
cnv_F59_Relapse$query <- str_c(cnv_F59_Relapse$chr, ":", cnv_F59_Relapse$chromStart, ":", cnv_F59_Relapse$chromEnd)
cnv_F59_Relapse_genes <- cnv_gene_annot(cnv_F59_Relapse, "F59_Relapse", mart)

cnv_F59_Ortho <- CNV_prep("../../facets/MAP559_Ortho_cncf.txt")
cnv_F59_Ortho <- separate(data = cnv_F59_Ortho, col = Chromosome, into = c("str", "chr"), sep = "r")
cnv_F59_Ortho$query <- str_c(cnv_F59_Ortho$chr, ":", cnv_F59_Ortho$chromStart, ":", cnv_F59_Ortho$chromEnd)
cnv_F59_Ortho_genes <- cnv_gene_annot(cnv_F59_Ortho, "F59_Ortho", mart)

cnv_F59_Sc <- CNV_prep("../../facets/MAP559_Sc_cncf.txt")
cnv_F59_Sc <- separate(data = cnv_F59_Sc, col = Chromosome, into = c("str", "chr"), sep = "r")
cnv_F59_Sc$query <- str_c(cnv_F59_Sc$chr, ":", cnv_F59_Sc$chromStart, ":", cnv_F59_Sc$chromEnd)
cnv_F59_Sc_genes <- cnv_gene_annot(cnv_F59_Sc, "F59_Sc", mart)

cnv_F59_genes <- bind_rows(cnv_F59_Relapse_genes, cnv_F59_Ortho_genes, cnv_F59_Sc_genes)
```

```{r gene_annotation_cnv_G36}
# MAP194
cnv_G36_Diag <- CNV_prep("../../facets/MAP636_Diag_cncf.txt")
cnv_G36_Diag <- separate(data = cnv_G36_Diag, col = Chromosome, into = c("str", "chr"), sep = "r")
cnv_G36_Diag$query <- str_c(cnv_G36_Diag$chr, ":", cnv_G36_Diag$chromStart, ":", cnv_G36_Diag$chromEnd)
cnv_G36_Diag_genes <- cnv_gene_annot(cnv_G36_Diag, "G36_Diag", mart)

cnv_G36_Relapse <- CNV_prep("../../facets/MAP636_Relapse_cncf.txt")
cnv_G36_Relapse <- separate(data = cnv_G36_Relapse, col = Chromosome, into = c("str", "chr"), sep = "r")
cnv_G36_Relapse$query <- str_c(cnv_G36_Relapse$chr, ":", cnv_G36_Relapse$chromStart, ":", cnv_G36_Relapse$chromEnd)
cnv_G36_Relapse_genes <- cnv_gene_annot(cnv_G36_Relapse, "G36_Relapse", mart)

cnv_G36_Ortho <- CNV_prep("../../facets/MAP636_Ortho_cncf.txt")
cnv_G36_Ortho <- separate(data = cnv_G36_Ortho, col = Chromosome, into = c("str", "chr"), sep = "r")
cnv_G36_Ortho$query <- str_c(cnv_G36_Ortho$chr, ":", cnv_G36_Ortho$chromStart, ":", cnv_G36_Ortho$chromEnd)
cnv_G36_Ortho_genes <- cnv_gene_annot(cnv_G36_Ortho, "G36_Ortho", mart)

cnv_G36_Sc <- CNV_prep("../../facets/MAP636_Sc_cncf.txt")
cnv_G36_Sc <- separate(data = cnv_G36_Sc, col = Chromosome, into = c("str", "chr"), sep = "r")
cnv_G36_Sc$query <- str_c(cnv_G36_Sc$chr, ":", cnv_G36_Sc$chromStart, ":", cnv_G36_Sc$chromEnd)
cnv_G36_Sc_genes <- cnv_gene_annot(cnv_G36_Sc, "G36_Sc", mart)

cnv_G36_genes <- bind_rows(cnv_G36_Diag_genes, cnv_G36_Relapse_genes, cnv_G36_Ortho_genes, cnv_G36_Sc_genes)
```

```{r cnv_genes_annot_fusions_associated}
table_all_genes_cnv <- bind_rows(cnv_B77_genes, cnv_B94_genes, cnv_C17_genes, cnv_D42_genes, cnv_E69_genes, cnv_F59_genes, cnv_G36_genes)
table_all_genes_cnv <- table_all_genes_cnv[,c("symbol", "cnv_type", "sample")]
write.table(x = table_all_genes_cnv, file = "table_genes_cnv_all_samples.csv", quote = F, row.names = F)

table_fusions <- table_Fus_Mut[which(table_Fus_Mut$Alteration == "Fusion"),]
table_fusions$sample <- str_c(table_fusions$Patient, "_", table_fusions$Time)

table_genes_cnv_in_fusion <- table_all_genes_cnv[which(table_all_genes_cnv$symbol %in% c(table_fusions$Gene1, table_fusions$Gene2)),]
table_genes_cnv_in_fusion <- table_genes_cnv_in_fusion[!duplicated(table_genes_cnv_in_fusion[,c('symbol', 'sample')]),]
rownames(table_genes_cnv_in_fusion) <- str_c(table_genes_cnv_in_fusion$sample, ":", table_genes_cnv_in_fusion$symbol)

table_fusions$Gene1_cnv <- table_genes_cnv_in_fusion[str_c(table_fusions$sample, ":", table_fusions$Gene1),"cnv_type"]
table_fusions$Gene2_cnv <- table_genes_cnv_in_fusion[str_c(table_fusions$sample, ":", table_fusions$Gene2),"cnv_type"]
write.table(table_fusions, file = "table_fusions_cnv_associated.csv", quote = F, row.names = F)

table_fusions$Time <- factor(table_fusions$Time , levels=c("Diag", "Relapse", "Ortho", "Sc"))
```

```{r cnv_genes_annot_fusions_ associated_plot}
table_fusions_plot <- table_fusions[, c("Gene1", "Patient", "Time", "sample", "Gene1_cnv")]
colnames(table_fusions_plot) <- c("Gene", "Patient", "Time", "sample", "Gene_cnv")
table_fusions_plot_2 <- table_fusions[, c("Gene2", "Patient", "Time", "sample", "Gene2_cnv")]
colnames(table_fusions_plot_2) <- c("Gene", "Patient", "Time", "sample", "Gene_cnv")
table_to_plot <- bind_rows(table_fusions_plot, table_fusions_plot_2)

# each gene
pdf(file = "sample_fusions_cnv_associated.pdf", width = 10)
g <- ggplot(table_to_plot, aes(x = Time, y = Gene_cnv ,fill = Gene_cnv)) +
  geom_col(position = "stack", width = 0.6, )
g + facet_wrap(~ Patient, ncol=4)
plot(g) 
dev.off()

# both sides
table_both_sides <- table_fusions
table_both_sides$both_sides_cnv <- str_c(table_both_sides$Gene1_cnv, "-", table_both_sides$Gene2_cnv)
pdf(file = "sample_fusions_cnv_associated_both_sides.pdf", width = 17)
g <- ggplot(table_both_sides, aes(x = Time, y = both_sides_cnv ,fill = both_sides_cnv)) +
  geom_col(position = "stack", width = 0.6, )
g + facet_wrap(~ Patient, ncol=4)
plot(g) 
dev.off()

# only with focal amplification
table_both_sides_foc_amp <- table_both_sides[which(table_both_sides$Gene1_cnv == "Focal_Amplification"),]
table_both_sides_foc_amp_2 <- table_both_sides[which(table_both_sides$Gene2_cnv == "Focal_Amplification"),]

table_both_sides_foc_amp <- unique(bind_rows(table_both_sides_foc_amp, table_both_sides_foc_amp_2))
pdf(file = "sample_fusions_cnv_associated_both_sides_focal_amplification.pdf", width = 17)
g <- ggplot(table_both_sides_foc_amp, aes(x = Time, y = both_sides_cnv ,fill = both_sides_cnv)) +
  geom_col(position = "stack", width = 0.6, )
g + facet_wrap(~ Patient, ncol=4)
plot(g)
dev.off()
table_write_foc_amp <- table_both_sides_foc_amp[which(table_both_sides_foc_amp$Gene1_cnv == "Focal_Amplification" & table_both_sides_foc_amp$Gene2_cnv == "Focal_Amplification"),]
write.table(table_write_foc_amp, file = "fusion_double_amplification_focale.tsv", quote = F, sep = "\t", col.names = T, row.names = F)

# all amplifications
table_both_sides_amp <- table_both_sides[which(table_both_sides$Gene1_cnv == "Amplification"),]
table_both_sides_amp_2 <- table_both_sides[which(table_both_sides$Gene2_cnv == "Amplification"),]
table_tot_amp <- unique(bind_rows(table_both_sides_amp, table_both_sides_amp_2)) 

table_both_sides_amp <- bind_rows(table_both_sides_foc_amp, table_tot_amp)
test_cor <- cor(table_both_sides_amp$Gene1_cnv, table_both_sides_amp$Gene2_cnv)
pdf(file = "sample_fusions_cnv_associated_both_sides_amplification.pdf", width = 17)
g <- ggplot(table_both_sides_amp, aes(x = Time, y = both_sides_cnv ,fill = both_sides_cnv)) +
  geom_col(position = "stack", width = 0.6, )
g + facet_wrap(~ Patient, ncol=4)
plot(g)
dev.off()
```

```{r all_fusions_in_arriba_files}
folder <- "../Fusions/arriba_data/Arriba_files/"
pattern <- "MAP*"

data_all <- list.files(path = folder, pattern = pattern, full.names = TRUE)
data_all <- lapply(data_all, function(i){read.table(i, header = FALSE)}) %>% bind_rows
data_arriba <- unique(data_all)

data_arriba$ID <- str_c(data_arriba$V1, "--", data_arriba$V2)

table_selected_fusions <- table_fusions
table_selected_fusions$ID_arriba <- str_c(table_selected_fusions$Gene1, "--", table_selected_fusions$Gene2)
selected_arriba <- data_arriba[which(data_arriba$ID %in% table_selected_fusions$ID_arriba),]
selected_arriba <- selected_arriba[which(selected_arriba$ID %in% list_fusions),]
selected_arriba$ID <- NULL

write.table(selected_arriba, file = "selected_mutations_visualisation_table_arriba.tsv", sep = "\t", quote = F, row.names = F, col.names = F)
```

```{r ponderated_expression}
# creating the expression table from the matrix
library(reshape2)

list_cnv_order <- c("MAP177_Diag", "MAP177_Relapse", "MAP177_Ortho", "MAP177_Sc","MAP194_Diag", "MAP194_Relapse", "MAP194_Ortho", "MAP194_Sc","MAP217_Diag", "MAP217_Relapse", "MAP217_Ortho", "MAP217_Sc","MAP342_Diag", "MAP342_Relapse", "MAP342_Ortho", "MAP342_Ortho.meta","MAP469_Diag", "MAP469_Relapse", "MAP469_Ortho", "MAP469_Sc", "MAP559_Relapse", "MAP559_Ortho", "MAP559_Sc", "MAP636_Diag", "MAP636_Relapse", "MAP636_Ortho", "MAP636_Sc")

list_cnv_order_rename <- c("B77_Diag", "B77_Relapse", "B77_Ortho", "B77_Sc","B94_Diag", "B94_Relapse", "B94_Ortho", "B94_Sc", "C17_Diag", "C17_Relapse", "C17_Ortho", "C17_Sc","D42_Diag", "D42_Relapse", "D42_Ortho", "D42_Sc","E69_Diag", "E69_Relapse", "E69_Ortho", "E69_Sc", "F59_Relapse", "F59_Ortho", "F59_Sc","G36_Diag", "G36_Relapse", "G36_Ortho", "G36_Sc")


table_expression <- exp_count[,list_cnv_order_rename]

table_cnv <- read.table(file = "../../scripts_facets/out_seg_file_continue.csv", header = T)
table_cnv <- table_cnv[which(table_cnv$Hugo_Symbol %in% rownames(table_expression)),]
table_expression <- table_expression[which(rownames(table_expression) %in% table_cnv$Hugo_Symbol),]
rownames(table_cnv) <- table_cnv$Hugo_Symbol
table_cnv$Hugo_Symbol <- NULL
table_cnv <- table_cnv[rownames(table_expression),list_cnv_order]
colnames(table_cnv) <- list_cnv_order_rename
table_cnv <- as.matrix(table_cnv)

table_ponderated_expression <- table_expression / table_cnv
table_ponderated_expression[table_ponderated_expression %in% c("NaN", "Inf", "-Inf", "Na")] <- 0
to_write <- as.data.frame(table_ponderated_expression)
to_write$Hugo_symbol <- rownames(to_write)
to_write <- to_write[,c("Hugo_symbol", list_cnv_order_rename)]
write.table(to_write, file = "ponderated_expression_table.csv", quote = F, sep = "\t", row.names = F, col.names = T)

u_ponder <- prcomp(t(table_ponderated_expression))

u_ponder <- umap(u_ponder$x)

umap_plot_df <- data.frame(u_ponder$layout) %>%
  # Turn sample IDs stored as row names into a column
  tibble::rownames_to_column("sample") %>%
  # Add the metadata into this data frame; match by sample IDs
  dplyr::inner_join(table_meta_data, by = "sample")

library(ggforce)
cairo_pdf(file = "umap_cnv_ponder_gene_expression.pdf")
ggplot(
  umap_plot_df,
  aes(
    x = X1,
    y = X2,
    color = Patient,# label points with different colors for each `subgroup`
    shape = Time,
    size = 3
  )
) +
  geom_point()
  #stat_ellipse(aes(x=X1, y=X2,color=Patient, group=Patient),type = "euclid")
dev.off()

```

```{r analyse_differentielle_exp_ponder}
library(DESeq2)
library(factoextra)
library(tidyr)
# analyse differentielle

####### data matrix
table_meta_data <- table_meta_data[list_cnv_order_rename,]


####### differential expression
ddsMat <- DESeqDataSetFromMatrix(countData = round(table_ponderated_expression),
                                 colData = table_meta_data,
                                 design = ~ Hum_Mou)

keep <- rowSums(counts(ddsMat)) > 1
dds <- ddsMat[keep,]
vsd <- vst(dds, blind = FALSE)
matrix_vsd <- assay(vsd)

sampleDists <- dist(t(assay(vsd)))

sampleDistMatrix <- as.matrix( sampleDists )
pheatmap::pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         filename = "sample_distance_matrix_ponderated.pdf")

pcaData <- plotPCA(vsd, intgroup = c("Hum_Mou"), returnData = TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))

dds <- DESeq(dds)
res <- results(dds, contrast = c("Hum_Mou", "Hum", "Mou"))


####volcano
alpha <- 0.05 # Threshold on the adjusted p-value
cols <- densCols(res$log2FoldChange, -log10(res$pvalue))

png("volcano_plot_human_mou_ponderated.png")
plot(res$log2FoldChange, -log10(res$padj), col=cols, panel.first=grid(),
     main="Volcano plot", xlab="Effect size: log2(fold-change)", ylab="-log10(adjusted p-value)",
     pch=20, cex=0.6)
abline(v=0)
abline(v=c(-1.5,1.5), col="brown")
abline(h=-log10(alpha), col="brown")

dev.off()

res_select <- res[-which(abs(res$log2FoldChange) > 1.5 & res$padj < alpha),]
res_removed <- res[which(abs(res$log2FoldChange) > 1.5 & res$padj < alpha),]
res_removed_data <- as.data.frame(res_removed[,c("log2FoldChange", "padj", "baseMean")])
res_removed_data <- res_removed_data[order(abs(res_removed_data$log2FoldChange), decreasing = T),]
write.table(res_removed_data, file = "table_foldChange_removed_genes_ponderated.csv", sep = "\t", quote = F)

png("volcano_plot_human_mou_after_selection_ponderated.png")
plot(res_select$log2FoldChange, -log10(res_select$padj), col=cols, panel.first=grid(),
     main="Volcano plot", xlab="Effect size: log2(fold-change)", ylab="-log10(adjusted p-value)",
     pch=20, cex=0.6)
abline(v=0)
abline(v=c(-1.5,1.5), col="brown")
abline(h=-log10(alpha), col="brown")
dev.off()

list_genes_keep <- rownames(res_select)

table_counts_removed <- matrix_vsd[which(!rownames(res) %in% list_genes_keep),]
table_counts_after_diff <- matrix_vsd[list_genes_keep,]

genes_removed <- rownames(table_counts_removed)

project_removed.pca <- prcomp(t(table_counts_removed))
project_after_diff.pca <- prcomp(t(table_counts_after_diff))

table_explained_variance_after_diff <- summary(project_after_diff.pca)
table_explained_variance_after_diff <- table_explained_variance_after_diff$importance
write.table(table_explained_variance_after_diff, file = "table_explained_variance_after_diff_ponderated.csv", sep = "\t", quote = F, row.names = T)

table_explained_variance_removed <- summary(project_removed.pca)
table_explained_variance_removed <- table_explained_variance_removed$importance
write.table(table_explained_variance_removed, file = "table_explained_variance_removed_ponderated.csv", sep = "\t", quote = F, row.names = T)


cairo_pdf(file = "explained_variance_after_diff_exp_ponderated.pdf")
fviz_eig(project_after_diff.pca)
dev.off()

cairo_pdf(file = "explained_variance_removed_exp_ponderated.pdf")
fviz_eig(project_removed.pca)
dev.off()

# explaining why we remove the second component
table_components <- as.data.frame(abs(project_removed.pca$x[,"PC2"]))
colnames(table_components) <- "PC2"
table_components$sample <- rownames(table_components)
table_components <- separate(table_components, col = "sample", into = c("Patient", "Time"))
table_components$Hum_Mou <- "Human"
table_components[which(table_components$Time %in% c("Ortho", "Sc")), "Hum_Mou"] <- "Mouse"

# boxplot
cairo_pdf(file = "second_component_boxplot_Human_Mouse_ponderated.pdf")
table_components %>%
  ggplot(aes(x = Hum_Mou,y = PC2)) +
  geom_boxplot() +
  geom_point(aes(fill=Patient,group=Patient),size=4,shape=21, position = position_dodge(0.2)) +
  theme(legend.position = "right") +
  labs(title = "PC2 values between human and mouse")
dev.off()

#umap without selection
#library(umap)
umap_after_diff <- umap(project_after_diff.pca$x[,c(1,3,4,5,6,7)])

umap_plot_df <- data.frame(umap_after_diff$layout) %>%
  # Turn sample IDs stored as row names into a column
  tibble::rownames_to_column("sample") %>%
  # Add the metadata into this data frame; match by sample IDs
  dplyr::inner_join(table_meta_data, by = "sample")

library(ggforce)
cairo_pdf(file = "umap_all_genes_after_diff_exp_ponderated.pdf")
ggplot(
  umap_plot_df,
  aes(
    x = X1,
    y = X2,
    color = Patient,# label points with different colors for each `subgroup`
    shape = Time,
    size = 3
  )
) +
  geom_point()
  #stat_ellipse(aes(x=X1, y=X2,color=Patient, group=Patient),type = "euclid")
dev.off()

umap_removed <- umap(project_removed.pca$x)

umap_removed_plot_df <- data.frame(umap_removed$layout) %>%
  # Turn sample IDs stored as row names into a column
  tibble::rownames_to_column("sample") %>%
  # Add the metadata into this data frame; match by sample IDs
  dplyr::inner_join(table_meta_data, by = "sample")

library(ggforce)
cairo_pdf(file = "umap_removed_after_diff_exp_ponderated.pdf")
ggplot(
  umap_removed_plot_df,
  aes(
    x = X1,
    y = X2,
    color = Patient,# label points with different colors for each `subgroup`
    shape = Time,
    size = 3
  )
) +
  geom_point()
  #stat_ellipse(aes(x=X1, y=X2,color=Patient, group=Patient),type = "euclid")
dev.off()

# heatmap correlation ACP
cor_acp_mat <- cor(t(project_after_diff.pca$x[,c(1,3,4,5,6,7)]))

pheatmap::pheatmap(cor_acp_mat, clustering_method = "ward.D", filename = "heatmap_ACP_correlation_ponderated.pdf", cellheight = 8, clustering_distance_rows = amap::Dist(cor_acp_mat, method="euclidean"), clustering_distance_cols = amap::Dist(t(cor_acp_mat), method = "kendall"))

cor_acp_mat_removed <- cor(t(project_removed.pca$x))

pheatmap::pheatmap(cor_acp_mat_removed, clustering_method = "ward.D", filename = "heatmap_ACP_removed_genes_correlation_ponderated.pdf", cellheight = 8, clustering_distance_rows = amap::Dist(cor_acp_mat_removed, method="euclidean"), clustering_distance_cols = amap::Dist(t(cor_acp_mat_removed), method = "kendall"))

# GSEA enrichment
library(dplyr)
gene_list=as.data.frame(res$log2FoldChange, res@rownames)
gene_list <- order(abs(gene_list), decreasing = T)
gene_list[list_genes_keep,]
names(gene_list) <- res_test$row
gene_list=gene_list[!is.na(gene_list)]
GSEA(gene=gene_list,TERM2GENE=msigdb$c5_bp,verbose=FALSE,pvalueCutoff=0.01,pAdjustMethod="BH",minGSSize=5,maxGSSize=500) -> gsea_res
GSEA(gene=gene_list,TERM2GENE=msigdb$h,verbose=FALSE,pvalueCutoff=0.01,pAdjustMethod="BH",minGSSize=5,maxGSSize=500) -> gsea_res
GSEA(gene=gene_list,TERM2GENE=msigdb$c2_kegg,verbose=FALSE,pvalueCutoff=0.01,pAdjustMethod="BH",minGSSize=5,maxGSSize=500) -> gsea_res


# wilcox test
res_wilcox_PC2 <- wilcox.test(x = as.vector(table_components[which(table_components$Hum_Mou == "Human"),"PC2"]), y = as.vector(table_components[which(table_components$Hum_Mou == "Mouse"),"PC2"]), alternative = "two.sided")



library(enrichR)
# function based on test
PCA_enrichment <- function(prcomp_object, dbs, components, name_file){
  
  pdf(file = name_file, width = 14, height = 18)
  for(database in dbs){
    first <- T
    for(component in components){
      if(first == T){
        first <- F
        table_enrich_PCA <- names(head(sort(abs(as.table(prcomp_object$rotation)[,component]), decreasing = T),20))
        enrichr_PCA <- enrichr(table_enrich_PCA, database)
        table_enrich_PCA <- enrichr_PCA[1]
        table_enrich_PCA$PCA <- str_c("PC", as.character(component))
      }
      else{
        table_component_sup <- names(head(sort(abs(as.table(prcomp_object$rotation)[,component]), decreasing = T),20))
        enrichr_comp <- enrichr(table_component_sup, database)
        table_component_sup <- enrichr_comp[1]
        table_component_sup$PCA <- str_c("PC", as.character(component))
        table_enrich_PCA <- bind_rows(table_enrich_PCA, table_component_sup)
      }
      
    }
    # print(table_enrich_PCA)
    table_enrich_PCA <- as.data.frame(table_enrich_PCA)
    table_temp <- table_enrich_PCA[,database]
    table_temp$PCA <- table_enrich_PCA[,"PCA"]
    table_enrich_PCA <- table_temp
    #print(table_enrich_PCA[,database])
    print(colnames(table_enrich_PCA))
    #print(str_c(database, '.', 'Adjusted.P.value'))
    table_enrich_PCA <- table_enrich_PCA[which(table_enrich_PCA[,'Adjusted.P.value'] <= 0.01),]
    # print(table_enrich_PCA)
    if(nrow(table_enrich_PCA) > 0){
      table_enrich_PCA <- separate(table_enrich_PCA, col = "Overlap", sep = "/", into = c("nb_genes", "nb_genes_class"), remove = F)
      # print(table_enrich_PCA)
      table_enrich_PCA$nb_genes <- as.numeric(table_enrich_PCA$nb_genes)
      plot_enrich_pca <- ggplot(table_enrich_PCA, aes(x = PCA, y = Term)) + geom_point(aes(colour=Adjusted.P.value, size=nb_genes)) + labs(title = database, x = "Component", y = "Term", size = 6) + scale_color_gradient(high = "blue", low = "red")
      plot(ggplotGrob(plot_enrich_pca))
    }
    
  }
  dev.off()
}
library(stringr)
PCA_enrichment(project_after_diff.pca, dbs_all, c(1,3,4,5,6,7), "all_components_enrichment_expression_ponderated.pdf")
dev.off()
PCA_enrichment(project_removed.pca, dbs_all, c(1,2,3,4,5,6,7,8,9), "all_components_enrichment_gene_removed_expression_ponderated.pdf")
dev.off()
# plot PCA contribution

# function table prep
pca_sample_contrib <- function(prcomp_x_matrix_per_cent, list_composant){ 
  return_table <- data.frame()
  
  table_sample_contribution <- as.data.frame(abs(t(prcomp_x_matrix_per_cent)))
  table_sample_contribution <- table_sample_contribution[list_composant,]
  #print(table_sample_contribution)
  table_sample_contribution <- apply(table_sample_contribution,2,function(x){x*100/sum(x)})
  #print(rowSums(table_sample_contribution))
  table_sample_contribution <- as.data.frame(t(table_sample_contribution))
  # print(rowSums(table_sample_contribution))

  for(composant in list_composant){
    table_comp <- as.data.frame(table_sample_contribution[,composant])
    rownames(table_comp) <- rownames(table_sample_contribution)
    colnames(table_comp) <- "value"
    table_comp$sample <- rownames(table_comp)
    #print(table_comp)
    table_comp <- separate(table_comp, col = sample, sep = "_", into = c("Patient", "Time"))
    table_comp$PC <- composant
    
    return_table <- bind_rows(return_table, table_comp)
  }
  return_table$Time <- factor(return_table$Time , levels=c("Diag", "Relapse", "Ortho", "Sc"))
  return_table$Patient <- factor(return_table$Patient , levels=c("B77", "B94", "C17", "D42", "E69", "F59", "G36"))
  return(return_table)
}



#
table_sample_contribution <- pca_sample_contrib(as.data.frame(abs(project_after_diff.pca$x)), c("PC1", "PC3", "PC4", "PC5", "PC6", "PC7"))

cairo_pdf(file = "sample_componant_participation_PCA_noPC2_ponderated.pdf", width = 10)
g <- ggplot(table_sample_contribution, aes(x = Time,y = value ,fill = PC)) +
  geom_col(position = "stack", width = 0.6)
g + facet_wrap(~ Patient, ncol=4)
plot(g)
dev.off()

# removed part
table_sample_contribution <- pca_sample_contrib(as.data.frame(abs(project_removed.pca$x)), c("PC1", "PC2", "PC3", "PC4", "PC5", "PC6", "PC7"))

pdf(file = "sample_componant_participation_PCA_removed_genes_ponderated.pdf", width = 10)
g <- ggplot(table_sample_contribution, aes(x = Time,y = value ,fill = PC)) +
  geom_col(position = "stack", width = 0.6)
g + facet_wrap(~ Patient, ncol=4)
plot(g) 
dev.off()

```

```{r genes_component_glmnet}
table_genes <- project_after_diff.pca$rotation

table_PC1 <- as.data.frame(table_genes[head(names(sort(abs(table_genes[,"PC1"]), decreasing = T)),100),"PC1"])
colnames(table_PC1) <- "PC1"
write.table(table_PC1, file = "Genes_PC1_ponderated.csv", sep = "\t", quote = F, col.names = F, row.names = T)

table_PC2 <- as.data.frame(table_genes[head(names(sort(abs(table_genes[,"PC2"]), decreasing = T)),100),"PC2"])
colnames(table_PC2) <- "PC2"
write.table(table_PC2, file = "Genes_PC2_ponderated.csv", sep = "\t", quote = F, col.names = F, row.names = T)

table_PC3 <- as.data.frame(table_genes[head(names(sort(abs(table_genes[,"PC3"]), decreasing = T)),100),"PC3"])
colnames(table_PC3) <- "PC3"
write.table(table_PC3, file = "Genes_PC3_ponderated.csv", sep = "\t", quote = F, col.names = F, row.names = T)

table_PC4 <- as.data.frame(table_genes[head(names(sort(abs(table_genes[,"PC4"]), decreasing = T)),100),"PC4"])
colnames(table_PC4) <- "PC4"
write.table(table_PC4, file = "Genes_PC4_ponderated.csv", sep = "\t", quote = F, col.names = F, row.names = T)

table_PC5 <- as.data.frame(table_genes[head(names(sort(abs(table_genes[,"PC5"]), decreasing = T)),100),"PC5"])
colnames(table_PC5) <- "PC5"
write.table(table_PC5, file = "Genes_PC5_ponderated.csv", sep = "\t", quote = F, col.names = F, row.names = T)

table_PC6 <- as.data.frame(table_genes[head(names(sort(abs(table_genes[,"PC6"]), decreasing = T)),100),"PC6"])
colnames(table_PC6) <- "PC6"
write.table(table_PC6, file = "Genes_PC6_ponderated.csv", sep = "\t", quote = F, col.names = F, row.names = T)

table_PC7 <- as.data.frame(table_genes[head(names(sort(abs(table_genes[,"PC7"]), decreasing = T)),100),"PC7"])
colnames(table_PC7) <- "PC7"
write.table(table_PC7, file = "Genes_PC7_ponderated.csv", sep = "\t", quote = F, col.names = F, row.names = T)
```

```{r gsea}
organism = "org.Hs.eg.db"
# BiocManager::install(organism, character.only = TRUE)
library(organism, character.only = TRUE)
library(clusterProfiler)
library(gtools)
library(dplyr)
list_dbs <- keytypes(org.Hs.eg.db)
TERM2GENE <- msigdb$c3_tft

table_PC1$genes <- rownames(table_PC1)
rowid <- order(table_PC1[, "PC1"], decreasing = T)
table_PC1 <- table_PC1[rowid,]
table_PC1$genes <- NULL
gene_PC1 <- table_PC1$PC1
names(gene_PC1) <- rownames(table_PC1)
foldchange_PC1 <- as.data.frame(foldchange(gene_PC1, as.table(table_genes[rownames(table_PC1), c("PC3", "PC4", "PC5","PC6","PC7")])))

foldchange_PC1$mean <- rowMeans(foldchange_PC1)
rowid <- order(foldchange_PC1[, "mean"], decreasing = T)
foldchange_PC1 <- foldchange_PC1[rowid,]
gene_PC1 <- foldchange_PC1$mean
names(gene_PC1) <- rownames(foldchange_PC1)

gse_PC1 <- GSEA(gene_PC1,
     exponent = 1,
     nPerm = 1000,
     minGSSize = 2,
     maxGSSize = 200,
     pvalueCutoff = 0.05,
     pAdjustMethod = "none",
     TERM2GENE,
     TERM2NAME = NA,
     verbose = TRUE,
     seed = FALSE,
     by = "DOSE")

library(DOSE)
cairo_pdf(file = "enrichment_PC1_ponderated.pdf")
#emapplot(gse, showCategory = 10)
cnetplot(gse_PC1, categorySize="pvalue", foldChange=gene_PC1, showCategory = 10)
#ridgeplot(gse) + labs(x = "enrichment distribution")
dev.off()


# PC2

table_PC2$genes <- rownames(table_PC2)
rowid <- order(table_PC2[, "PC2"], decreasing = T)
table_PC2 <- table_PC2[rowid,]
table_PC2$genes <- NULL
gene_PC2 <- table_PC2$PC2
names(gene_PC2) <- rownames(table_PC2)
foldchange_PC2 <- as.data.frame(foldchange(gene_PC2, as.table(table_genes[rownames(table_PC2), c("PC3", "PC4", "PC5","PC6","PC7")])))

foldchange_PC2$mean <- rowMeans(foldchange_PC2)
rowid <- order(foldchange_PC2[, "mean"], decreasing = T)
foldchange_PC2 <- foldchange_PC2[rowid,]
gene_PC2 <- foldchange_PC2$mean
names(gene_PC2) <- rownames(foldchange_PC2)

gse_PC2 <- GSEA(gene_PC2,
     exponent = 1,
     nPerm = 1000,
     minGSSize = 2,
     maxGSSize = 200,
     pvalueCutoff = 0.05,
     pAdjustMethod = "none",
     TERM2GENE,
     TERM2NAME = NA,
     verbose = TRUE,
     seed = FALSE,
     by = "DOSE")

library(DOSE)
cairo_pdf(file = "enrichment_PC2_ponderated.pdf")
#emapplot(gse, showCategory = 10)
cnetplot(gse_PC2, categorySize="pvalue", foldChange=gene_PC2, showCategory = 10)
#ridgeplot(gse) + labs(x = "enrichment distribution")
dev.off()


# PC3

table_PC3$genes <- rownames(table_PC3)
rowid <- order(table_PC3[, "PC3"], decreasing = T)
table_PC3 <- table_PC3[rowid,]
table_PC3$genes <- NULL
gene_PC3 <- table_PC3$PC3
names(gene_PC3) <- rownames(table_PC3)
foldchange_PC3 <- as.data.frame(foldchange(gene_PC3, as.table(table_genes[rownames(table_PC3), c("PC3", "PC4", "PC5","PC6","PC7")])))

foldchange_PC3$mean <- rowMeans(foldchange_PC3)
rowid <- order(foldchange_PC3[, "mean"], decreasing = T)
foldchange_PC3 <- foldchange_PC3[rowid,]
gene_PC3 <- foldchange_PC3$mean
names(gene_PC3) <- rownames(foldchange_PC3)

gse_PC3 <- GSEA(gene_PC3,
     exponent = 1,
     nPerm = 1000,
     minGSSize = 2,
     maxGSSize = 200,
     pvalueCutoff = 0.05,
     pAdjustMethod = "none",
     TERM2GENE,
     TERM2NAME = NA,
     verbose = TRUE,
     seed = FALSE,
     by = "DOSE")

library(DOSE)
cairo_pdf(file = "enrichment_PC3_ponderated.pdf")
#emapplot(gse, showCategory = 10)
cnetplot(gse_PC3, categorySize="pvalue", foldChange=gene_PC3, showCategory = 10)
#ridgeplot(gse) + labs(x = "enrichment distribution")
dev.off()

# PC4

table_PC4$genes <- rownames(table_PC4)
rowid <- order(table_PC4[, "PC4"], decreasing = T)
table_PC4 <- table_PC4[rowid,]
table_PC4$genes <- NULL
gene_PC4 <- table_PC4$PC4
names(gene_PC4) <- rownames(table_PC4)
foldchange_PC4 <- as.data.frame(foldchange(gene_PC4, as.table(table_genes[rownames(table_PC4), c("PC3", "PC4", "PC5","PC6","PC7")])))

foldchange_PC4$mean <- rowMeans(foldchange_PC4)
rowid <- order(foldchange_PC4[, "mean"], decreasing = T)
foldchange_PC4 <- foldchange_PC4[rowid,]
gene_PC4 <- foldchange_PC4$mean
names(gene_PC4) <- rownames(foldchange_PC4)

gse_PC4 <- GSEA(gene_PC4,
     exponent = 1,
     nPerm = 1000,
     minGSSize = 2,
     maxGSSize = 200,
     pvalueCutoff = 0.05,
     pAdjustMethod = "none",
     TERM2GENE,
     TERM2NAME = NA,
     verbose = TRUE,
     seed = FALSE,
     by = "DOSE")

library(DOSE)
cairo_pdf(file = "enrichment_PC4_ponderated.pdf")
#emapplot(gse, showCategory = 10)
cnetplot(gse_PC4, categorySize="pvalue", foldChange=gene_PC4, showCategory = 10)
#ridgeplot(gse) + labs(x = "enrichment distribution")
dev.off()

# PC5

table_PC5$genes <- rownames(table_PC5)
rowid <- order(table_PC5[, "PC5"], decreasing = T)
table_PC5 <- table_PC5[rowid,]
table_PC5$genes <- NULL
gene_PC5 <- table_PC5$PC5
names(gene_PC5) <- rownames(table_PC5)
foldchange_PC5 <- as.data.frame(foldchange(gene_PC5, as.table(table_genes[rownames(table_PC5), c("PC3", "PC5", "PC5","PC6","PC7")])))

foldchange_PC5$mean <- rowMeans(foldchange_PC5)
rowid <- order(foldchange_PC5[, "mean"], decreasing = T)
foldchange_PC5 <- foldchange_PC5[rowid,]
gene_PC5 <- foldchange_PC5$mean
names(gene_PC5) <- rownames(foldchange_PC5)

gse_PC5 <- GSEA(gene_PC5,
     exponent = 1,
     nPerm = 1000,
     minGSSize = 2,
     maxGSSize = 200,
     pvalueCutoff = 0.05,
     pAdjustMethod = "none",
     TERM2GENE,
     TERM2NAME = NA,
     verbose = TRUE,
     seed = FALSE,
     by = "DOSE")

library(DOSE)
cairo_pdf(file = "enrichment_PC5_ponderated.pdf")
#emapplot(gse, showCategory = 10)
cnetplot(gse_PC5, categorySize="pvalue", foldChange=gene_PC5, showCategory = 10)
#ridgeplot(gse) + labs(x = "enrichment distribution")
dev.off()


# PC6

table_PC6$genes <- rownames(table_PC6)
rowid <- order(table_PC6[, "PC6"], decreasing = T)
table_PC6 <- table_PC6[rowid,]
table_PC6$genes <- NULL
gene_PC6 <- table_PC6$PC6
names(gene_PC6) <- rownames(table_PC6)
foldchange_PC6 <- as.data.frame(foldchange(gene_PC6, as.table(table_genes[rownames(table_PC6), c("PC3", "PC6", "PC5","PC6","PC7")])))

foldchange_PC6$mean <- rowMeans(foldchange_PC6)
rowid <- order(foldchange_PC6[, "mean"], decreasing = T)
foldchange_PC6 <- foldchange_PC6[rowid,]
gene_PC6 <- foldchange_PC6$mean
names(gene_PC6) <- rownames(foldchange_PC6)

gse_PC6 <- GSEA(gene_PC6,
     exponent = 1,
     nPerm = 1000,
     minGSSize = 2,
     maxGSSize = 200,
     pvalueCutoff = 0.05,
     pAdjustMethod = "none",
     TERM2GENE,
     TERM2NAME = NA,
     verbose = TRUE,
     seed = FALSE,
     by = "DOSE")

library(DOSE)
cairo_pdf(file = "enrichment_PC6_ponderated.pdf")
#emapplot(gse, showCategory = 10)
cnetplot(gse_PC6, categorySize="pvalue", foldChange=gene_PC6, showCategory = 10)
#ridgeplot(gse) + labs(x = "enrichment distribution")
dev.off()


# PC7

table_PC7$genes <- rownames(table_PC7)
rowid <- order(table_PC7[, "PC7"], decreasing = T)
table_PC7 <- table_PC7[rowid,]
table_PC7$genes <- NULL
gene_PC7 <- table_PC7$PC7
names(gene_PC7) <- rownames(table_PC7)
foldchange_PC7 <- as.data.frame(foldchange(gene_PC7, as.table(table_genes[rownames(table_PC7), c("PC3", "PC7", "PC5","PC6","PC7")])))

foldchange_PC7$mean <- rowMeans(foldchange_PC7)
rowid <- order(foldchange_PC7[, "mean"], decreasing = T)
foldchange_PC7 <- foldchange_PC7[rowid,]
gene_PC7 <- foldchange_PC7$mean
names(gene_PC7) <- rownames(foldchange_PC7)

gse_PC7 <- GSEA(gene_PC7,
     exponent = 1,
     nPerm = 1000,
     minGSSize = 2,
     maxGSSize = 200,
     pvalueCutoff = 0.05,
     pAdjustMethod = "none",
     TERM2GENE,
     TERM2NAME = NA,
     verbose = TRUE,
     seed = FALSE,
     by = "DOSE")

library(DOSE)
cairo_pdf(file = "enrichment_PC7_ponderated.pdf")
#emapplot(gse, showCategory = 10)
cnetplot(gse_PC7, categorySize="pvalue", foldChange=gene_PC7, showCategory = 10)
#ridgeplot(gse) + labs(x = "enrichment distribution")
dev.off()


PCA_enrichment(project_after_diff.pca, dbs_all, c(1,2,3,4,5,6,7), "enrichr_PCA_ponderated_expression.pdf")
dev.off()
dbs_all
```

```{r glmnet_multivariate_gaussian}
library(glmnet)
table_variables <- t(table_counts_after_diff)
table_explain <- project_after_diff.pca$x


# PC1
cv <- cv.glmnet(x = table_variables,y = table_explain[,"PC1"] ,alpha=1)
res_glmnet <- glmnet(x = table_variables, y = table_explain[,"PC1"], family = "mgaussian", alpha = 1, lambda = cv$lambda.1se)
predict_res_glmnet <- predict.glmnet(res_glmnet, newx = new_matrix, s = res_glmnet$lambda, type=c("coefficients"))
coeff_PC1 <- as.matrix(predict_res_glmnet)
coeff_PC1 <- as.data.frame(coeff_PC1)
coeff_PC1$Genes <- rownames(coeff_PC1)
coeff_PC1 <- coeff_PC1[which(abs(coeff_PC1[,1]) > 0),]
remove <- c("(Intercept)")
coeff_PC1 <- coeff_PC1[!(row.names(coeff_PC1) %in% remove), ]
write.table(coeff_PC1, file = "PC1_glmnet_genes_selection_ponderated.csv", sep = "\t", quote = F, row.names = F)
coeff_PC1$PC <- "PC1"

#PC2
cv <- cv.glmnet(x = table_variables,y = table_explain[,"PC2"] ,alpha=1)
res_glmnet <- glmnet(x = table_variables, y = table_explain[,"PC2"], family = "mgaussian", alpha = 1, lambda = cv$lambda.1se)
predict_res_glmnet <- predict.glmnet(res_glmnet, newx = new_matrix, s = res_glmnet$lambda, type=c("coefficients"))
coeff_PC2 <- as.matrix(predict_res_glmnet)
coeff_PC2 <- as.data.frame(coeff_PC2)
coeff_PC2$Genes <- rownames(coeff_PC2)
coeff_PC2 <- coeff_PC2[which(abs(coeff_PC2[,1]) > 0),]
remove <- c("(Intercept)")
coeff_PC2 <- coeff_PC2[!(row.names(coeff_PC2) %in% remove), ]
write.table(coeff_PC2, file = "PC2_glmnet_genes_selection_ponderated.csv", sep = "\t", quote = F, row.names = F)
coeff_PC2$PC <- "PC2"

#PC3
cv <- cv.glmnet(x = table_variables,y = table_explain[,"PC3"] ,alpha=1)
res_glmnet <- glmnet(x = table_variables, y = table_explain[,"PC3"], family = "mgaussian", alpha = 1, lambda = cv$lambda.1se)
predict_res_glmnet <- predict.glmnet(res_glmnet, newx = new_matrix, s = res_glmnet$lambda, type=c("coefficients"))
coeff_PC3 <- as.matrix(predict_res_glmnet)
coeff_PC3 <- as.data.frame(coeff_PC3)
coeff_PC3$Genes <- rownames(coeff_PC3)
coeff_PC3 <- coeff_PC3[which(abs(coeff_PC3[,1]) > 0),]
remove <- c("(Intercept)")
coeff_PC3 <- coeff_PC3[!(row.names(coeff_PC3) %in% remove), ]
write.table(coeff_PC3, file = "PC3_glmnet_genes_selection_ponderated.csv", sep = "\t", quote = F, row.names = F)
coeff_PC3$PC <- "PC3"

#PC4
cv <- cv.glmnet(x = table_variables,y = table_explain[,"PC4"] ,alpha=1)
res_glmnet <- glmnet(x = table_variables, y = table_explain[,"PC4"], family = "mgaussian", alpha = 1, lambda = cv$lambda.1se)
predict_res_glmnet <- predict.glmnet(res_glmnet, newx = new_matrix, s = res_glmnet$lambda, type=c("coefficients"))
coeff_PC4 <- as.matrix(predict_res_glmnet)
coeff_PC4 <- as.data.frame(coeff_PC4)
coeff_PC4$Genes <- rownames(coeff_PC4)
coeff_PC4 <- coeff_PC4[which(abs(coeff_PC4[,1]) > 0),]
remove <- c("(Intercept)")
coeff_PC4 <- coeff_PC4[!(row.names(coeff_PC4) %in% remove), ]
write.table(coeff_PC4, file = "PC4_glmnet_genes_selection_ponderated.csv", sep = "\t", quote = F, row.names = F)
coeff_PC4$PC <- "PC4"

#PC5
cv <- cv.glmnet(x = table_variables,y = table_explain[,"PC5"] ,alpha=1)
res_glmnet <- glmnet(x = table_variables, y = table_explain[,"PC5"], family = "mgaussian", alpha = 1, lambda = cv$lambda.1se)
predict_res_glmnet <- predict.glmnet(res_glmnet, newx = new_matrix, s = res_glmnet$lambda, type=c("coefficients"))
coeff_PC5 <- as.matrix(predict_res_glmnet)
coeff_PC5 <- as.data.frame(coeff_PC5)
coeff_PC5$Genes <- rownames(coeff_PC5)
coeff_PC5 <- coeff_PC5[which(abs(coeff_PC5[,1]) > 0),]
remove <- c("(Intercept)")
coeff_PC5 <- coeff_PC5[!(row.names(coeff_PC5) %in% remove), ]
write.table(coeff_PC5, file = "PC5_glmnet_genes_selection_ponderated.csv", sep = "\t", quote = F, row.names = F)
coeff_PC5$PC <- "PC5"

#PC6
cv <- cv.glmnet(x = table_variables,y = table_explain[,"PC6"] ,alpha=1)
res_glmnet <- glmnet(x = table_variables, y = table_explain[,"PC6"], family = "mgaussian", alpha = 1, lambda = cv$lambda.1se)
predict_res_glmnet <- predict.glmnet(res_glmnet, newx = new_matrix, s = res_glmnet$lambda, type=c("coefficients"))
coeff_PC6 <- as.matrix(predict_res_glmnet)
coeff_PC6 <- as.data.frame(coeff_PC6)
coeff_PC6$Genes <- rownames(coeff_PC6)
coeff_PC6 <- coeff_PC6[which(abs(coeff_PC6[,1]) > 0),]
remove <- c("(Intercept)")
coeff_PC6 <- coeff_PC6[!(row.names(coeff_PC6) %in% remove), ]
write.table(coeff_PC6, file = "PC6_glmnet_genes_selection_ponderated.csv", sep = "\t", quote = F, row.names = F)
coeff_PC6$PC <- "PC6"

#PC7
cv <- cv.glmnet(x = table_variables,y = table_explain[,"PC7"] ,alpha=1)
res_glmnet <- glmnet(x = table_variables, y = table_explain[,"PC7"], family = "mgaussian", alpha = 1, lambda = cv$lambda.1se)
predict_res_glmnet <- predict.glmnet(res_glmnet, newx = new_matrix, s = res_glmnet$lambda, type=c("coefficients"))
coeff_PC7 <- as.matrix(predict_res_glmnet)
coeff_PC7 <- as.data.frame(coeff_PC7)
coeff_PC7$Genes <- rownames(coeff_PC7)
coeff_PC7 <- coeff_PC7[which(abs(coeff_PC7[,1]) > 0),]
remove <- c("(Intercept)")
coeff_PC7 <- coeff_PC7[!(row.names(coeff_PC7) %in% remove), ]
write.table(coeff_PC7, file = "PC7_glmnet_genes_selection_ponderated.csv", sep = "\t", quote = F, row.names = F)
coeff_PC7$PC <- "PC7"


genes_glmnet_selection <- bind_rows(coeff_PC1, coeff_PC3, coeff_PC4, coeff_PC5, coeff_PC6, coeff_PC7)
u_glmnet <- prcomp(t(table_counts_after_diff[genes_glmnet_selection$Genes,]))

u_glmnet <- umap(u_glmnet$x)

umap_plot_df <- data.frame(u_glmnet$layout) %>%
  # Turn sample IDs stored as row names into a column
  tibble::rownames_to_column("sample") %>%
  # Add the metadata into this data frame; match by sample IDs
  dplyr::inner_join(table_meta_data, by = "sample")

library(ggforce)
cairo_pdf(file = "umap_glmnet_selected_ponderated.pdf")
ggplot(
  umap_plot_df,
  aes(
    x = X1,
    y = X2,
    color = Patient,# label points with different colors for each `subgroup`
    shape = Time,
    size = 3
  )
) +
  geom_point()
  #stat_ellipse(aes(x=X1, y=X2,color=Patient, group=Patient),type = "euclid")
dev.off()

# Heatmap genes PC glmnet selected
table_count_genes_glmnet <- log10(table_counts_after_diff[genes_glmnet_selection$Genes,] + 1)
gene_annotation <- genes_glmnet_selection[,c("Genes", "PC")]
gene_annotation$Genes <- NULL
pheatmap::pheatmap(table_count_genes_glmnet, clustering_method = "ward.D", filename = "heatmap_counts_glmnet_selection_ponderated.pdf", cellheight = 8, clustering_distance_rows = amap::Dist(table_count_genes_glmnet, method="euclidean"), clustering_distance_cols = amap::Dist(t(table_count_genes_glmnet), method = "kendall"), annotation_row = gene_annotation, annotation_col = table_meta_data[,c("Patient", "Hum_Mou")])
```

```{r converting_function}
convertMouseGeneList <- function(x){
require("biomaRt")
human = useMart("ensembl", dataset = "hsapiens_gene_ensembl")
mouse = useMart("ensembl", dataset = "mmusculus_gene_ensembl")
genesV2 = getLDS(attributes = c("mgi_symbol"), filters = "mgi_symbol", values = x , mart = mouse, attributesL = c("hgnc_symbol"), martL = human, uniqueRows=T)
humanx <- unique(genesV2[, 2])
# Print the first 6 genes found to the screen
print(head(humanx))
return(humanx)
}
```

```{r microenvironnment analysis}
expression_mouse_part <- read.table("../expression/file_salmon_tpm_pdx_mouse_part.csv", header = T, sep = "\t")
convert_table <- read.table("../expression/hs_mm_orthologs.txt", header = T, sep = "\t", fill = F)
convert_table$Mouse_symbol <- bitr(convert_table$Other_gene_id,fromType="ENTREZID",toType="SYMBOL",OrgDb="org.Mm.eg.db",drop=F)
convert_table$Human_symbol <- bitr(convert_table$Gene_id,fromType="ENTREZID",toType="SYMBOL",OrgDb="org.Hs.eg.db",drop=F)
convert_table <- convert_table[,c("Mouse_symbol", "Human_symbol")]
convert_table <- data.frame(convert_table$Mouse_symbol$SYMBOL, convert_table$Human_symbol$SYMBOL)
convert_table <- convert_table[complete.cases(convert_table),]

rownames(convert_table) <- convert_table$convert_table.Mouse_symbol.SYMBOL

rownames(expression_mouse_part) <- expression_mouse_part$Hugo_symbol
expression_mouse_part <- expression_mouse_part[which(expression_mouse_part$Hugo_symbol %in% convert_table$convert_table.Mouse_symbol.SYMBOL),]
expression_mouse_part <- expression_mouse_part[match(convert_table$convert_table.Mouse_symbol.SYMBOL, expression_mouse_part$Hugo_symbol),]
expression_mouse_part$Human_symbol <- convert_table[rownames(expression_mouse_part), "convert_table.Human_symbol.SYMBOL"]
expression_mouse_part <- expression_mouse_part[!duplicated(expression_mouse_part[,c('Human_symbol')]),]
expression_mouse_part <- expression_mouse_part[complete.cases(expression_mouse_part),]
rownames(expression_mouse_part) <- expression_mouse_part$Human_symbol
expression_mouse_part$Hugo_symbol <- NULL
expression_mouse_part$Human_symbol <- NULL
expression_mouse_part <- expression_mouse_part[rowSums(expression_mouse_part)>17*3,]

```

```{r pca_expression_mouse_part}
expression_mouse_part <- expression_mouse_part %>% mutate_if(is.factor, as.numeric)
expression_mouse_part_log <- log10(expression_mouse_part + 1)
acp_all <- prcomp(t(expression_mouse_part_log))

pdf(file = "explained_variance_mouse_part.pdf")
fviz_eig(acp_all)
dev.off()

# meta data
meta_data_mouse_part <- data.frame(colnames(expression_mouse_part))
colnames(meta_data_mouse_part) <- "sample"
meta_data_mouse_part <- separate(data = meta_data_mouse_part, col = "sample", into = c("patient", "time"), sep = "_", remove = F)

# Umap
umap_naiv <- umap(acp_all$x[,c(1,2,3,4)])

umap_plot_df <- data.frame(umap_naiv$layout) %>%
  # Turn sample IDs stored as row names into a column
  tibble::rownames_to_column("sample") %>%
  # Add the metadata into this data frame; match by sample IDs
  dplyr::inner_join(meta_data_mouse_part, by = "sample")

library(ggforce)
cairo_pdf(file = "umap_all_genes_mouse_part.pdf")
ggplot(
  umap_plot_df,
  aes(
    x = X1,
    y = X2,
    color = patient,# label points with different colors for each `subgroup`
    shape = time,
    size = 3
  )
) +
  geom_point()
dev.off()
```

```{r expression_removed_genes}
removed_genes_expression <- rownames(res_removed_data)

table_xenome_removed <- expression_mouse_part[which(rownames(expression_mouse_part) %in% removed_genes_expression),]

removed_genes_expression_not_in_mouse <- removed_genes_expression[!removed_genes_expression %in% rownames(table_xenome_removed)]
removed_genes_expression_in_mouse <- rownames(table_xenome_removed)

### removed part not found in mouse (supplemental)
table_missing_in_mouse <- table_expression[removed_genes_expression_not_in_mouse,]
table_missing_in_mouse <- table_missing_in_mouse[complete.cases(table_missing_in_mouse),]
write.table(table_missing_in_mouse, file = "table_expression_removed_not_in_mouse.tsv", sep = "\t", row.names = T, col.names = T)

enrichment <- enrichr(removed_genes_expression_not_in_mouse, dbs_all)
plot_enrichr(enrichment, dbs_all, "plot_enrichment_removed_genes_expression_not_in_mouse.pdf")

write.table(as.data.frame(removed_genes_expression_not_in_mouse), file = "table_genes_removed_expression_not_in_mouse.csv", sep = "\t", col.names = F, row.names = F, quote = F)
write.table(as.data.frame(removed_genes_expression_in_mouse), file = "table_genes_removed_expression_in_mouse.csv", sep = "\t", col.names = F, row.names = F, quote = F)
write.table(as.data.frame(removed_genes_expression), file = "table_genes_removed_expression.csv", sep = "\t", col.names = F, row.names = F, quote = F)

correlation_missing_in_mouse <- cor(table_missing_in_mouse)

cairo_pdf(file = "correlation_expression_removed_not_in_mouse.pdf")
corrplot::corrplot(correlation_missing_in_mouse, method = "circle", order = "hclust")
dev.off()

log_table_missing_in_mouse <- log(table_missing_in_mouse + 1)

data_long_expression_removed_not_in_mouse <- as.data.frame(log_table_missing_in_mouse) %>%
  pivot_longer(colnames(log_table_missing_in_mouse)) %>% 
  as.data.frame()

cairo_pdf(filename = "density_expression_removed_not_in_mouse.pdf")
ggplot(data_long_expression_removed_not_in_mouse, aes(x = value)) +    # Draw each column as density
  geom_density() + 
  facet_wrap(~ name, scales = "free")
dev.off()

data_long_expression_removed_not_in_mouse <- separate(data_long_expression_removed_not_in_mouse, name, into = c("patient", "category"), sep = "_", remove = F)

cairo_pdf(filename = "density_expression_removed_not_in_mouse_by_time.pdf")
ggplot(data_long_expression_removed_not_in_mouse, aes(x = value)) +    # Draw each column as density
  geom_density() + 
  facet_wrap(~ category, scales = "free")
dev.off()
#


enrichment_in_mouse <- enrichr(removed_genes_expression_in_mouse, dbs_all)
plot_enrichr(enrichment_in_mouse, dbs_all, "plot_enrichment_removed_genes_expression_in_mouse.pdf")

write.table(as.data.frame(removed_genes_expression_in_mouse), file = "table_genes_removed_expression_in_mouse.csv", sep = "\t", col.names = F, row.names = F, quote = F)
```

```{r umap_genes_found_in_xenome_mouse}
table_xenome_removed_log <- log10(table_xenome_removed + 1)
acp_all <- prcomp(t(table_xenome_removed_log))

cairo_pdf(file = "explained_variance_mouse_part_xenome_removed.pdf")
fviz_eig(acp_all)
dev.off()

# Umap
umap_naiv <- umap(acp_all$x[,c(1,2,3,4,5)])

umap_plot_df <- data.frame(umap_naiv$layout) %>%
  # Turn sample IDs stored as row names into a column
  tibble::rownames_to_column("sample") %>%
  # Add the metadata into this data frame; match by sample IDs
  dplyr::inner_join(meta_data_mouse_part, by = "sample")

library(ggforce)
cairo_pdf(file = "umap_genes_found_in_xenome_mouse_part.pdf")
ggplot(
  umap_plot_df,
  aes(
    x = X1,
    y = X2,
    color = patient,# label points with different colors for each `subgroup`
    shape = time,
    size = 3
  )
) +
  geom_point()
dev.off()
```

```{r microenvironment}
table_combined_expression_me <- table_counts_removed[which(rownames(table_counts_removed) %in% removed_genes_expression_in_mouse),]
colnames(table_combined_expression_me) <- list_exp_order

list_columns <- colnames(table_combined_expression_me[,which(colnames(table_combined_expression_me) %in% colnames(table_xenome_removed))])
table_xenome_removed <- table_xenome_removed[,list_columns]

table_combined_expression_me <- table_combined_expression_me[removed_genes_expression_in_mouse,]
table_combined_expression_me_mouse_part <- table_combined_expression_me[removed_genes_expression_in_mouse,list_columns]
table_combined_expression_me_mouse_part <- table_combined_expression_me + table_xenome_removed
table_combined_expression_me_human_part <- table_combined_expression_me[,-which(colnames(table_combined_expression_me) %in% list_columns)]

table_combined_expression_me <- cbind(table_combined_expression_me_human_part, table_combined_expression_me_mouse_part)

# plotting
sample <- data.frame(colnames(table_combined_expression_me))
table_meta_data <- separate(data = sample, col = "colnames.table_combined_expression_me.", into = c("Patient", "time"), sep = "_", remove = F)
colnames(table_meta_data) <- c("sample", "Patient", "Time")
rownames(table_meta_data) <- table_meta_data$sample

table_combined_expression_me_log <- log10(table_combined_expression_me + 1)
acp_all <- prcomp(t(table_combined_expression_me_log))

cairo_pdf(file = "explained_variance_combined_expression_me.pdf")
fviz_eig(acp_all)
dev.off()

# Umap
umap_naiv <- umap(acp_all$x[,c(1,2,3)])

umap_plot_df <- data.frame(umap_naiv$layout) %>%
  # Turn sample IDs stored as row names into a column
  tibble::rownames_to_column("sample") %>%
  # Add the metadata into this data frame; match by sample IDs
  dplyr::inner_join(table_meta_data, by = "sample")

library(ggforce)
cairo_pdf(file = "umap_combined_expression_me.pdf")
ggplot(
  umap_plot_df,
  aes(
    x = X1,
    y = X2,
    color = Patient,# label points with different colors for each `subgroup`
    shape = time,
    size = 3
  )
) +
  geom_point()
dev.off()
```

```{r separated analysis}
table_combined_expression_me_scaled <- t(scale(t(table_combined_expression_me)))


acp_all <- prcomp(t(table_combined_expression_me_scaled))

cairo_pdf(file = "explained_variance_combined_expression_me_scaled.pdf")
fviz_eig(acp_all)
dev.off()

# Umap
umap_naiv <- umap(acp_all$x[,c(1,2,3)])

umap_plot_df <- data.frame(umap_naiv$layout) %>%
  # Turn sample IDs stored as row names into a column
  tibble::rownames_to_column("sample") %>%
  # Add the metadata into this data frame; match by sample IDs
  dplyr::inner_join(table_meta_data, by = "sample")

library(ggforce)
cairo_pdf(file = "umap_combined_expression_me_scaled.pdf")
ggplot(
  umap_plot_df,
  aes(
    x = X1,
    y = X2,
    colour = Patient,
    shape = time,
    size = 3
  )
) +
  geom_point()
dev.off()



# separate scale
table_combined_expression_me_separate_scale <- cbind(t(scale(t(table_combined_expression_me_human_part))), t(scale(t(table_combined_expression_me_mouse_part))))

data_long <- as.data.frame(table_combined_expression_me_log) %>%
  pivot_longer(colnames(table_combined_expression_me_log)) %>% 
  as.data.frame()

cairo_pdf(filename = "density_expression_each_sample_me_log.pdf")
ggplot(data_long, aes(x = value)) +    # Draw each column as density
  geom_density() + 
  facet_wrap(~ name, scales = "free")
dev.off()

acp_all <- prcomp(t(table_combined_expression_me_separate_scale))

cairo_pdf(file = "explained_variance_combined_expression_me_separate_scaled.pdf")
fviz_eig(acp_all)
dev.off()

# Umap
umap_naiv <- umap(acp_all$x[,c(1,2,3,4,5)])

umap_plot_df <- data.frame(umap_naiv$layout) %>%
  # Turn sample IDs stored as row names into a column
  tibble::rownames_to_column("sample") %>%
  # Add the metadata into this data frame; match by sample IDs
  dplyr::inner_join(table_meta_data, by = "sample")

umap_plot_df[umap_plot_df[,"sample"] == "MAP342_Ortho.meta", "sample"] <- "MAP342_Sc"
umap_plot_df[umap_plot_df[,"Time"] == "Ortho.meta", "Time"] <- "Sc"

library(ggforce)
cairo_pdf(file = "umap_combined_expression_me_separate_scaled.pdf")
ggplot(
  umap_plot_df,
  aes(
    x = X1,
    y = X2,
    colour = Patient,
    shape = Time,
    size = 3
  )
) +
  geom_point()
dev.off()

# correlation
cor_table <- cor(table_combined_expression_me_separate_scale)
cairo_pdf(file = "correlation_combined_expression_me_separate_scaled.pdf")
corrplot::corrplot(cor_table, method = "circle", order = "hclust")
dev.off()

PCA_enrichment(acp_all, dbs_all, c(1,2,3,4,5), "all_components_enrichment_me_separate_scale.pdf")
dev.off()

table_test <- as.data.frame(abs(acp_all$x))
rownames(table_test)[rownames(table_test) == "MAP342_Ortho.meta"] <- "MAP342_Sc"
table_sample_contribution <- pca_sample_contrib(table_test, c("PC1", "PC2", "PC3", "PC4"))

table_values_pca <- as.data.frame(acp_all$x)
rownames(table_values_pca)[rownames(table_values_pca) == "MAP342_Ortho.meta"] <- "MAP342_Sc"
table_values_pca <- pca_sample_values(table_values_pca, c("PC1", "PC2", "PC3", "PC4"))

pdf(file = "sample_values_participation_PCA_combined_expression_me_separate_scaled.pdf", width = 14, height = 10)
g <- ggplot(table_values_pca, aes(x = Time,y = value ,fill = PC)) +
  geom_bar(stat = "identity") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))

g + facet_wrap(~ PC + Patient, ncol=8, labeller = label_wrap_gen(multi_line=FALSE))
plot(g)
dev.off()

table_values_pca_umap <- merge(table_values_pca, umap_plot_df, by=c("Patient", "Time"))
table_values_pca_umap_PC5 <- table_values_pca_umap[which(table_values_pca_umap$PC == "PC5"),]

cairo_pdf(file = "umap_combined_expression_me_separate_scaled_PCA.pdf", height = 12, width = 14)
g <- ggplot(
  table_values_pca_umap,
  aes(
    x = X1,
    y = X2,
    shape = Time,
    size = 1
  )
) +
  geom_point(aes(color = scale(value))) +
  #geom_mark_ellipse(aes(x=X1, y=X2, group=Patient, size = 0.02), linetype = 2) +
  scale_color_gradient2(low="blue", mid="white", high="red") +
  theme_dark()
g + facet_wrap(~ PC, ncol = 2)
dev.off()

pdf(file = "sample_componant_participation_PCA_me_separate_scale.pdf", width = 10)
g <- ggplot(table_sample_contribution, aes(x = Time,y = value ,fill = PC)) +
  geom_col(position = "stack", width = 0.6)
g + facet_wrap(~ Patient, ncol=4)
plot(g)
dev.off()

PCA_enrichment(acp_all, "GO_Biological_Process_2021", c(1,2,3,4), "all_components_enrichment_me_separate_scale_GO_biological.pdf")

table_PC1 <- as.data.frame(acp_all$rotation[head(names(sort(abs(acp_all$rotation[,"PC1"]), decreasing = T)),100),"PC1"])
colnames(table_PC1) <- "PC1"
write.table(table_PC1, file = "Genes_PC1_ME_separate_scale.csv", sep = "\t", quote = F, col.names = F, row.names = T)

table_PC2 <- as.data.frame(acp_all$rotation[head(names(sort(abs(acp_all$rotation[,"PC2"]), decreasing = T)),100),"PC2"])
colnames(table_PC2) <- "PC2"
write.table(table_PC2, file = "Genes_PC2_ME_separate_scale.csv", sep = "\t", quote = F, col.names = F, row.names = T)

table_PC3 <- as.data.frame(acp_all$rotation[head(names(sort(abs(acp_all$rotation[,"PC3"]), decreasing = T)),100),"PC3"])
colnames(table_PC3) <- "PC3"
write.table(table_PC3, file = "Genes_PC3_ME_separate_scale.csv", sep = "\t", quote = F, col.names = F, row.names = T)

table_PC4 <- as.data.frame(acp_all$rotation[head(names(sort(abs(acp_all$rotation[,"PC4"]), decreasing = T)),100),"PC4"])
colnames(table_PC4) <- "PC4"
write.table(table_PC4, file = "Genes_PC4_ME_separate_scale.csv", sep = "\t", quote = F, col.names = F, row.names = T)

table_PC5 <- as.data.frame(acp_all$rotation[head(names(sort(abs(acp_all$rotation[,"PC5"]), decreasing = T)),100),"PC5"])
colnames(table_PC5) <- "PC5"
write.table(table_PC5, file = "Genes_PC5_ME_separate_scale.csv", sep = "\t", quote = F, col.names = F, row.names = T)
```

```{r enrichissement par PC ME}
read_tsv("Genes_PC1_ME_separate_scale.csv",col_names=c("Genes","Contrib"))  %>% arrange(desc(Contrib)) %>%as.data.frame %>%remove_rownames %>% column_to_rownames("Genes")-> PC1 ; tmp=PC1[,1];   names(tmp) <- rownames(PC1) ; PC1=tmp

read_tsv("Genes_PC2_ME_separate_scale.csv",col_names=c("Genes","Contrib"))  %>% arrange(desc(Contrib)) %>%as.data.frame %>%remove_rownames %>% column_to_rownames("Genes")-> PC2 ; tmp=PC2[,1];   names(tmp) <- rownames(PC2) ; PC2=tmp

read_tsv("Genes_PC3_ME_separate_scale.csv",col_names=c("Genes","Contrib"))  %>% arrange(desc(Contrib)) %>%as.data.frame %>%remove_rownames %>% column_to_rownames("Genes")-> PC3 ; tmp=PC3[,1];   names(tmp) <- rownames(PC3) ; PC3=tmp

read_tsv("Genes_PC4_ME_separate_scale.csv",col_names=c("Genes","Contrib"))  %>% arrange(desc(Contrib)) %>%as.data.frame %>%remove_rownames %>% column_to_rownames("Genes")-> PC4 ; tmp=PC4[,1];   names(tmp) <- rownames(PC4) ; PC4=tmp

read_tsv("Genes_PC5_ME_separate_scale.csv",col_names=c("Genes","Contrib"))  %>% arrange(desc(Contrib)) %>%as.data.frame %>%remove_rownames %>% column_to_rownames("Genes")-> PC5 ; tmp=PC5[,1];   names(tmp) <- rownames(PC5) ; PC5=tmp


Gost_by_sign=function(x,y,Type="GO:BP"){
  flag1=0;
  flag2=0;
  
  if(length(names(x[which(x>0)])) > 0){
  gost(names(x[which(x>0)]),organism = "hsapiens", significant = FALSE) ->res
  res$result %>% filter(p_value<0.05 & source==Type) %>% mutate(Score=-log(p_value))%>% add_column(PC=y) ->res_pos
  }else{
  flag1=1
};
  
  if(length(names(x[which(x<0)])) > 0){
  gost(names(x[which(x<0)]),organism = "hsapiens", significant = FALSE) ->res
  res$result %>% filter(p_value<0.05 & source==Type) %>% mutate(Score=log(p_value))%>% add_column(PC=y) ->res_neg
  }else{
  flag2=1
  };
  
if(flag1==0 & flag2==0){
rbind(res_pos,res_neg) -> res
}else{
  if(flag1==0){
    res_pos -> res
  }else{
if(flag2==0){
res_neg -> res
}
  }
}
return(res)
}
library(gprofiler2)

list("PC1"=PC1,"PC2"=PC2,"PC3"=PC3,"PC4"=PC4) ->LPC

map2(LPC,names(LPC),~Gost_by_sign(.x,.y,Type="GO:BP")) -> All_5GOBP

All_5GOBP %>% bind_rows %>% filter((PC %in% c("PC1","PC2","PC3","PC4")) )  %>% mutate(sign=ifelse(Score>0,"P","M")) %>% group_by(PC,sign) %>% top_n(20,wt=abs(Score))%>% ungroup %>%  count((term_name)) %>%  arrange(desc(n)) %>% filter(n>2) %>% dplyr::select(Name=1)-> Num_All_5GOBP

pdf("enrichment_all_components_by_sign_ME.pdf", width = 12, height = 13)
All_5GOBP_to_plot <- All_5GOBP %>% bind_rows %>% filter((PC %in% c("PC1","PC2","PC3","PC4"))) %>% filter(!(term_name %in% Num_All_5GOBP$Name)) %>% mutate(sign=ifelse(Score>0,"P","M")) %>% group_by(PC,sign) %>% top_n(20,wt=abs(Score))%>% ungroup %>% mutate(term_PC=fct_reorder(fct_reorder(paste0(term_name,"",PC),desc(Score)),PC))
  ggplot(All_5GOBP_to_plot[order(All_5GOBP_to_plot$PC, decreasing = T),],aes(term_PC,Score,fill=PC)) + geom_bar(stat="identity") + coord_flip() + geom_hline(yintercept = 0,colour="black", linetype = "longdash") + scale_x_discrete(limits = rev(levels(All_5GOBP_to_plot$term_PC)))
dev.off()
```

```{r differnetial_analysis_ME}
ddsMat <- DESeqDataSetFromMatrix(countData = round(table_combined_expression_me_separate_scale),
                                 colData = table_meta_data,
                                 design = ~ Hum_Mou)

keep <- rowSums(counts(ddsMat)) > 1
dds <- ddsMat[keep,]
vsd <- vst(dds, blind = FALSE)
matrix_vsd <- assay(vsd)

sampleDists <- dist(t(assay(vsd)))

sampleDistMatrix <- as.matrix( sampleDists )
pheatmap::pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         filename = "sample_distance_matrix_ponderated.pdf")

pcaData <- plotPCA(vsd, intgroup = c("Hum_Mou"), returnData = TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))

dds <- DESeq(dds)
res <- results(dds, contrast = c("Hum_Mou", "Hum", "Mou"))

""
####volcano
alpha <- 0.05 # Threshold on the adjusted p-value
cols <- densCols(res$log2FoldChange, -log10(res$pvalue))

png("volcano_plot_human_mou_ponderated.png")
plot(res$log2FoldChange, -log10(res$padj), col=cols, panel.first=grid(),
     main="Volcano plot", xlab="Effect size: log2(fold-change)", ylab="-log10(adjusted p-value)",
     pch=20, cex=0.6)
abline(v=0)
abline(v=c(-1.5,1.5), col="brown")
abline(h=-log10(alpha), col="brown")

dev.off()

res_select <- res[-which(abs(res$log2FoldChange) > 1.5 & res$padj < alpha),]
res_removed <- res[which(abs(res$log2FoldChange) > 1.5 & res$padj < alpha),]
res_removed_data <- as.data.frame(res_removed[,c("log2FoldChange", "padj", "baseMean")])
res_removed_data <- res_removed_data[order(abs(res_removed_data$log2FoldChange), decreasing = T),]
write.table(res_removed_data, file = "table_foldChange_removed_genes_ponderated.csv", sep = "\t", quote = F)

png("volcano_plot_human_mou_after_selection_ponderated.png")
plot(res_select$log2FoldChange, -log10(res_select$padj), col=cols, panel.first=grid(),
     main="Volcano plot", xlab="Effect size: log2(fold-change)", ylab="-log10(adjusted p-value)",
     pch=20, cex=0.6)
abline(v=0)
abline(v=c(-1.5,1.5), col="brown")
abline(h=-log10(alpha), col="brown")
dev.off()

list_genes_keep <- rownames(res_select)

table_counts_removed <- matrix_vsd[which(!rownames(res) %in% list_genes_keep),]
table_counts_after_diff <- matrix_vsd[list_genes_keep,]

genes_removed <- rownames(table_counts_removed)

project_removed.pca <- prcomp(t(table_counts_removed))
project_after_diff.pca <- prcomp(t(table_counts_after_diff))

table_explained_variance_after_diff <- summary(project_after_diff.pca)
table_explained_variance_after_diff <- table_explained_variance_after_diff$importance
write.table(table_explained_variance_after_diff, file = "table_explained_variance_after_diff_ponderated.csv", sep = "\t", quote = F, row.names = T)

table_explained_variance_removed <- summary(project_removed.pca)
table_explained_variance_removed <- table_explained_variance_removed$importance
write.table(table_explained_variance_removed, file = "table_explained_variance_removed_ponderated.csv", sep = "\t", quote = F, row.names = T)


cairo_pdf(file = "explained_variance_after_diff_exp_ponderated.pdf")
fviz_eig(project_after_diff.pca)
dev.off()

cairo_pdf(file = "explained_variance_removed_exp_ponderated.pdf")
fviz_eig(project_removed.pca)
dev.off()

```

```{r cnv_mutations_association}
vect_samples <- c("B77_Diag", "B77_Relapse", "B77_Ortho", "B77_Sc","B94_Diag", "B94_Relapse", "B94_Ortho", "B94_Sc", "C17_Diag", "C17_Relapse", "C17_Ortho", "C17_Sc","D42_Diag", "D42_Relapse", "D42_Ortho", "D42_Sc","E69_Diag", "E69_Relapse", "E69_Ortho", "E69_Sc", "F59_Relapse", "F59_Ortho", "F59_Sc","G36_Diag", "G36_Relapse", "G36_Ortho", "G36_Sc")

vect_patients <- c("B77", "B77", "B77", "B77","B94", "B94", "B94", "B94", "C17", "C17", "C17", "C17","D42", "D42", "D42", "D42","E69", "E69", "E69", "E69", "F59", "F59", "F59","G36", "G36", "G36", "G36")

vect_times <- c("Diag", "Relapse", "Ortho", "Sc","Diag", "Relapse", "Ortho", "Sc", "Diag", "Relapse", "Ortho", "Sc","Diag", "Relapse", "Ortho", "Sc","Diag", "Relapse", "Ortho", "Sc", "Relapse", "Ortho", "Sc","Diag", "Relapse", "Ortho", "Sc")

table_nb_mut_cnv <- data.frame(vect_samples, vect_patients, vect_times)

table_all_mut <- table_mut
table_all_mut$sample <- str_c(table_all_mut$Patient, "_",table_all_mut$Time)

table_nb_mut_cnv$nb_mutations <- table(table_all_mut$sample)

table_nb_mut_cnv$nb_segments <- c(nrow(cnv_B77_Diag), nrow(cnv_B77_Relapse), nrow(cnv_B77_Ortho), nrow(cnv_B77_Sc), nrow(cnv_B94_Diag), nrow(cnv_B94_Relapse), nrow(cnv_B94_Ortho), nrow(cnv_B94_Sc), nrow(cnv_C17_Diag), nrow(cnv_C17_Relapse), nrow(cnv_C17_Ortho), nrow(cnv_C17_Sc), nrow(cnv_D42_Diag), nrow(cnv_D42_Relapse), nrow(cnv_D42_Ortho), nrow(cnv_D42_Sc), nrow(cnv_E69_Diag), nrow(cnv_E69_Relapse), nrow(cnv_E69_Ortho), nrow(cnv_E69_Sc), nrow(cnv_F59_Relapse), nrow(cnv_F59_Ortho), nrow(cnv_F59_Sc), nrow(cnv_G36_Diag), nrow(cnv_G36_Relapse), nrow(cnv_G36_Ortho), nrow(cnv_G36_Sc))

table_nb_mut_cnv$nb_cnv <- c(nrow(cnv_B77_Diag[which(cnv_B77_Diag$CNvalue != 0),]), nrow(cnv_B77_Relapse[which(cnv_B77_Relapse$CNvalue != 0),]), nrow(cnv_B77_Ortho[which(cnv_B77_Ortho$CNvalue != 0),]), nrow(cnv_B77_Sc[which(cnv_B77_Sc$CNvalue != 0),]), nrow(cnv_B94_Diag[which(cnv_B94_Diag$CNvalue != 0),]), nrow(cnv_B94_Relapse[which(cnv_B94_Relapse$CNvalue != 0),]), nrow(cnv_B94_Ortho[which(cnv_B94_Ortho$CNvalue != 0),]), nrow(cnv_B94_Sc[which(cnv_B94_Sc$CNvalue != 0),]), nrow(cnv_C17_Diag[which(cnv_C17_Diag$CNvalue != 0),]), nrow(cnv_C17_Relapse[which(cnv_C17_Relapse$CNvalue != 0),]), nrow(cnv_C17_Ortho[which(cnv_C17_Ortho$CNvalue != 0),]), nrow(cnv_C17_Sc[which(cnv_C17_Sc$CNvalue != 0),]), nrow(cnv_D42_Diag[which(cnv_D42_Diag$CNvalue != 0),]), nrow(cnv_D42_Relapse[which(cnv_D42_Relapse$CNvalue != 0),]), nrow(cnv_D42_Ortho[which(cnv_D42_Ortho$CNvalue != 0),]), nrow(cnv_D42_Sc[which(cnv_D42_Sc$CNvalue != 0),]), nrow(cnv_E69_Diag[which(cnv_E69_Diag$CNvalue != 0),]), nrow(cnv_E69_Relapse[which(cnv_E69_Relapse$CNvalue != 0),]), nrow(cnv_E69_Ortho[which(cnv_E69_Ortho$CNvalue != 0),]), nrow(cnv_E69_Sc[which(cnv_E69_Sc$CNvalue != 0),]), nrow(cnv_F59_Relapse[which(cnv_F59_Relapse$CNvalue != 0),]), nrow(cnv_F59_Ortho[which(cnv_F59_Ortho$CNvalue != 0),]), nrow(cnv_F59_Sc[which(cnv_F59_Sc$CNvalue != 0),]), nrow(cnv_G36_Diag[which(cnv_G36_Diag$CNvalue != 0),]), nrow(cnv_G36_Relapse[which(cnv_G36_Relapse$CNvalue != 0),]), nrow(cnv_G36_Ortho[which(cnv_G36_Ortho$CNvalue != 0),]), nrow(cnv_G36_Sc[which(cnv_G36_Sc$CNvalue != 0),]))

table_nb_mut_cnv$nb_mutations <- as.numeric(table_nb_mut_cnv$nb_mutations)
# plotting

lm_eqn <- function(df){
    m <- lm(y ~ x, df);
    eq <- substitute(italic(y) == a + b %.% italic(x)*","~~italic(r)^2~"="~r2, 
         list(a = format(unname(coef(m)[1]), digits = 2),
              b = format(unname(coef(m)[2]), digits = 2),
             r2 = format(summary(m)$r.squared, digits = 3)))
    as.character(as.expression(eq));
}

cairo_pdf(filename = "plot_nb_mut_nb_segment.pdf")
ggplot(table_nb_mut_cnv, aes(nb_mutations, nb_segments)) +
       geom_point(aes(nb_mutations, nb_segments, colour = factor(vect_patients), shape = factor(vect_times), size=2)) +
       geom_smooth(aes(x = nb_mutations, y = nb_segments),method="lm")
dev.off()

cairo_pdf(filename = "plot_nb_mut_nb_cnv.pdf")
ggplot(table_nb_mut_cnv, aes(nb_mutations, nb_cnv)) +
       geom_point(aes(nb_mutations, nb_cnv, colour = factor(vect_patients), shape = factor(vect_times), size=2)) +
       geom_smooth(aes(x = nb_mutations, y = nb_cnv),method="lm")
dev.off()

coefficient_corr <- cor.test(table_nb_mut_cnv$nb_mutations, table_nb_mut_cnv$nb_segments, method = "spearman")
```

```{r flatten_coor_matrix}
# ++++++++++++++++++++++++++++
# flattenCorrMatrix
# ++++++++++++++++++++++++++++
# cormat : matrix of the correlation coefficients
flattenCorrMatrix <- function(cormat) {
  ut <- upper.tri(cormat)
  data.frame(
    row = rownames(cormat)[row(cormat)[ut]],
    column = rownames(cormat)[col(cormat)[ut]],
    cor  =(cormat)[ut]
    )
}

```

```{r SC vs Ortho ME}
# first the table we use
table_counts_SC_Ortho_ME <- table_combined_expression_me
table_counts_SC_Ortho_ME$MAP342_Sc <- table_counts_SC_Ortho_ME$MAP342_Ortho.meta
table_counts_SC_Ortho_ME$MAP342_Ortho.meta <- NULL

# then, the metadata
sample <- c("MAP177_Relapse", "MAP177_Ortho", "MAP177_Sc", "MAP194_Relapse", "MAP194_Ortho", "MAP194_Sc", "MAP204_Relapse", "MAP204_Ortho", "MAP204_Sc", "MAP217_Relapse", "MAP217_Ortho", "MAP217_Sc", "MAP342_Relapse", "MAP342_Ortho", "MAP342_Sc", "MAP469_Relapse", "MAP469_Ortho", "MAP469_Sc", "MAP559_Relapse", "MAP559_Ortho", "MAP559_Sc", "MAP636_Relapse", "MAP636_Ortho", "MAP636_Sc")
table_counts_SC_Ortho_ME <- table_counts_SC_Ortho_ME[,sample]

patient <- c("MAP177", "MAP177", "MAP177", "MAP194", "MAP194", "MAP194", "MAP204", "MAP204", "MAP204", "MAP217", "MAP217", "MAP217", "MAP342", "MAP342", "MAP342", "MAP469", "MAP469", "MAP469", "MAP559", "MAP559", "MAP559", "MAP636", "MAP636", "MAP636")
time <- c("Relapse", "Ortho", "Sc", "Relapse", "Ortho", "Sc", "Relapse", "Ortho", "Sc", "Relapse", "Ortho", "Sc", "Relapse", "Ortho", "Sc", "Relapse", "Ortho", "Sc", "Relapse", "Ortho", "Sc", "Relapse", "Ortho", "Sc")
table_meta_data_SC_Ortho_ME <- data.frame(sample, patient, time)

table_counts_SC_Ortho_ME <- log10(table_counts_SC_Ortho_ME + 1)

cor_table_SC_Ortho_table <- cor(table_counts_SC_Ortho_ME, method = "pearson")
cairo_pdf("correlation_relapse_SC_ortho.pdf")
corrplot::corrplot(cor_table_SC_Ortho_table, method = "circle", )
dev.off()

pheatmap::pheatmap(cor_table_SC_Ortho_table, clustering_method = "ward.D", filename = "correlation_pheatmap_SC_Ortho_relapse.pdf", cellheight = 8, clustering_distance_rows = amap::Dist(cor_table_SC_Ortho_table, method="euclidean"), clustering_distance_cols = amap::Dist(t(cor_table_SC_Ortho_table), method = "kendall"))

flat_corr <- flattenCorrMatrix(cor_table_SC_Ortho_table)
flat_corr <- separate(data = flat_corr, col = "row", into = c("patient1", "time1"), sep = "_")
flat_corr <- separate(data = flat_corr, col = "column", into = c("patient2", "time2"), sep = "_")

flat_corr <- flat_corr[which(flat_corr$time1 %in% c("Relapse")), ]
flat_corr <- flat_corr[-which(flat_corr$time2 %in% c("Relapse")), ]
flat_corr <- flat_corr[which(flat_corr$patient1 == flat_corr$patient2), ]


flat_sc_ortho <- flat_corr[,c("patient1", "time2", "cor")]

# boxplot
cairo_pdf(file = "boxplot_relapse_sc_ortho_corr.pdf")
flat_sc_ortho %>%
  ggplot(aes(x = time2,y = cor)) +
  geom_boxplot() +
  geom_point(aes(fill=patient1,group=patient1),size=4,shape=21, position = position_dodge(0.2)) +
  theme(legend.position = "right") +
  labs(title = "correlation between SC or Ortho samples with Relapse")
dev.off()
```

```{r SC vs Ortho tumor}
# first the table we use
table_counts_SC_Ortho_tumor <- exp_table[which(rownames(exp_table) %in% list_genes_keep),c("MAP177_Diag", "MAP177_Relapse", "MAP177_Ortho", "MAP177_Sc","MAP194_Diag", "MAP194_Relapse", "MAP194_Ortho", "MAP194_Sc","MAP204_Diag", "MAP204_Relapse", "MAP204_Ortho", "MAP204_Sc","MAP217_Diag", "MAP217_Relapse", "MAP217_Ortho", "MAP217_Sc","MAP342_Diag", "MAP342_Relapse", "MAP342_Ortho", "MAP342_Ortho.meta","MAP469_Diag", "MAP469_Relapse", "MAP469_Ortho", "MAP469_Sc", "MAP559_Diag", "MAP559_Relapse", "MAP559_Ortho", "MAP559_Sc", "MAP636_Diag", "MAP636_Relapse", "MAP636_Ortho", "MAP636_Sc")]
table_counts_SC_Ortho_tumor$MAP342_Sc <- table_counts_SC_Ortho_tumor$MAP342_Ortho.meta  
table_counts_SC_Ortho_tumor$MAP342_Ortho.meta <- NULL

# then, the metadata
sample <- c("MAP177_Relapse", "MAP177_Ortho", "MAP177_Sc", "MAP194_Relapse", "MAP194_Ortho", "MAP194_Sc", "MAP204_Relapse", "MAP204_Ortho", "MAP204_Sc", "MAP217_Relapse", "MAP217_Ortho", "MAP217_Sc", "MAP342_Relapse", "MAP342_Ortho", "MAP342_Sc", "MAP469_Relapse", "MAP469_Ortho", "MAP469_Sc", "MAP559_Relapse", "MAP559_Ortho", "MAP559_Sc", "MAP636_Relapse", "MAP636_Ortho", "MAP636_Sc")
table_counts_SC_Ortho_tumor <- table_counts_SC_Ortho_tumor[,sample]

patient <- c("MAP177", "MAP177", "MAP177", "MAP194", "MAP194", "MAP194", "MAP204", "MAP204", "MAP204", "MAP217", "MAP217", "MAP217", "MAP342", "MAP342", "MAP342", "MAP469", "MAP469", "MAP469", "MAP559", "MAP559", "MAP559", "MAP636", "MAP636", "MAP636")
time <- c("Relapse", "Ortho", "Sc", "Relapse", "Ortho", "Sc", "Relapse", "Ortho", "Sc", "Relapse", "Ortho", "Sc", "Relapse", "Ortho", "Sc", "Relapse", "Ortho", "Sc", "Relapse", "Ortho", "Sc", "Relapse", "Ortho", "Sc")
table_meta_data_SC_Ortho_tumor <- data.frame(sample, patient, time)
table_meta_data_SC_Ortho_tumor$time <- as.factor(table_meta_data_SC_Ortho_tumor$time)

# correlation analysis
table_counts_SC_Ortho_tumor <- log10(table_counts_SC_Ortho_tumor + 1)

cor_table_SC_Ortho_table <- cor(table_counts_SC_Ortho_tumor, method = "pearson")
cairo_pdf("correlation_relapse_SC_ortho_tumor.pdf")
corrplot::corrplot(cor_table_SC_Ortho_table, method = "circle", )
dev.off()

pheatmap::pheatmap(cor_table_SC_Ortho_table, clustering_method = "ward.D", filename = "correlation_pheatmap_SC_Ortho_relapse_tumor.pdf", cellheight = 8, clustering_distance_rows = amap::Dist(cor_table_SC_Ortho_table, method="euclidean"), clustering_distance_cols = amap::Dist(t(cor_table_SC_Ortho_table), method = "kendall"))

flat_corr <- flattenCorrMatrix(cor_table_SC_Ortho_table)
flat_corr <- separate(data = flat_corr, col = "row", into = c("patient1", "time1"), sep = "_")
flat_corr <- separate(data = flat_corr, col = "column", into = c("patient2", "time2"), sep = "_")

flat_corr <- flat_corr[which(flat_corr$time1 %in% c("Relapse")), ]
flat_corr <- flat_corr[-which(flat_corr$time2 %in% c("Relapse")), ]
flat_corr <- flat_corr[which(flat_corr$patient1 == flat_corr$patient2), ]


flat_sc_ortho <- flat_corr[,c("patient1", "time2", "cor")]

# boxplot
cairo_pdf(file = "boxplot_relapse_sc_ortho_corr_tumor.pdf")
flat_sc_ortho %>%
  ggplot(aes(x = time2,y = cor)) +
  geom_boxplot() +
  geom_point(aes(fill=patient1,group=patient1),size=4,shape=21, position = position_dodge(0.2)) +
  theme(legend.position = "right") +
  labs(title = "correlation between SC or Ortho samples with Relapse")
dev.off()
```

```{r quantification sequences xenograft samples}
table_quantification <- read.table(file = "../expression/table_xenograft_sequence_quantification.tsv", header = TRUE)
table_quantification <- table_quantification[-17,]
cairo_pdf(file = "boxplot_proportion_mouse_reads_in_xenograft.pdf")
table_quantification %>%
  ggplot(aes(x = Time,y = prop_mouse)) +
  geom_boxplot() +
  geom_point(aes(fill=Patient,group=Patient),size=4,shape=21, position = position_dodge(0.2)) +
  theme(legend.position = "right") +
  labs(title = "proportion of mouse reads in xenograft samples")
dev.off()
```